// @huaweicloud/huaweicloud-sdk-vpc/v2/model/BatchDeleteVpcTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchDeleteVpcTagsRequestBodyActionEnum = exports.BatchDeleteVpcTagsRequestBody = void 0;
var BatchDeleteVpcTagsRequestBody = /** @class */ (function () {
    function BatchDeleteVpcTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchDeleteVpcTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchDeleteVpcTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchDeleteVpcTagsRequestBody;
}());
exports.BatchDeleteVpcTagsRequestBody = BatchDeleteVpcTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchDeleteVpcTagsRequestBodyActionEnum;
(function (BatchDeleteVpcTagsRequestBodyActionEnum) {
    BatchDeleteVpcTagsRequestBodyActionEnum["DELETE"] = "delete";
})(BatchDeleteVpcTagsRequestBodyActionEnum = exports.BatchDeleteVpcTagsRequestBodyActionEnum || (exports.BatchDeleteVpcTagsRequestBodyActionEnum = {}));
