// @huaweicloud/huaweicloud-sdk-vpc/v2/model/BatchCreateSubnetTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateSubnetTagsRequestBodyActionEnum = exports.BatchCreateSubnetTagsRequestBody = void 0;
var BatchCreateSubnetTagsRequestBody = /** @class */ (function () {
    function BatchCreateSubnetTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchCreateSubnetTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchCreateSubnetTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchCreateSubnetTagsRequestBody;
}());
exports.BatchCreateSubnetTagsRequestBody = BatchCreateSubnetTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchCreateSubnetTagsRequestBodyActionEnum;
(function (BatchCreateSubnetTagsRequestBodyActionEnum) {
    BatchCreateSubnetTagsRequestBodyActionEnum["CREATE"] = "create";
})(BatchCreateSubnetTagsRequestBodyActionEnum = exports.BatchCreateSubnetTagsRequestBodyActionEnum || (exports.BatchCreateSubnetTagsRequestBodyActionEnum = {}));
