// @huaweicloud/huaweicloud-sdk-vpc/v2/model/BatchDeleteSubnetTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchDeleteSubnetTagsRequestBodyActionEnum = exports.BatchDeleteSubnetTagsRequestBody = void 0;
var BatchDeleteSubnetTagsRequestBody = /** @class */ (function () {
    function BatchDeleteSubnetTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchDeleteSubnetTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchDeleteSubnetTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchDeleteSubnetTagsRequestBody;
}());
exports.BatchDeleteSubnetTagsRequestBody = BatchDeleteSubnetTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchDeleteSubnetTagsRequestBodyActionEnum;
(function (BatchDeleteSubnetTagsRequestBodyActionEnum) {
    BatchDeleteSubnetTagsRequestBodyActionEnum["DELETE"] = "delete";
})(BatchDeleteSubnetTagsRequestBodyActionEnum = exports.BatchDeleteSubnetTagsRequestBodyActionEnum || (exports.BatchDeleteSubnetTagsRequestBodyActionEnum = {}));
