// @huaweicloud/huaweicloud-sdk-vpc/v2/model/BatchCreateVpcTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateVpcTagsRequestBodyActionEnum = exports.BatchCreateVpcTagsRequestBody = void 0;
var BatchCreateVpcTagsRequestBody = /** @class */ (function () {
    function BatchCreateVpcTagsRequestBody(action, tags) {
        this['action'] = action;
        this['tags'] = tags;
    }
    BatchCreateVpcTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    BatchCreateVpcTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BatchCreateVpcTagsRequestBody;
}());
exports.BatchCreateVpcTagsRequestBody = BatchCreateVpcTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var BatchCreateVpcTagsRequestBodyActionEnum;
(function (BatchCreateVpcTagsRequestBodyActionEnum) {
    BatchCreateVpcTagsRequestBodyActionEnum["CREATE"] = "create";
})(BatchCreateVpcTagsRequestBodyActionEnum = exports.BatchCreateVpcTagsRequestBodyActionEnum || (exports.BatchCreateVpcTagsRequestBodyActionEnum = {}));
