// @huaweicloud/huaweicloud-sdk-vpc/v2/model/BindingVifDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BindingVifDetails = void 0;
var BindingVifDetails = /** @class */ (function () {
    function BindingVifDetails() {
    }
    BindingVifDetails.prototype.withPrimaryInterface = function (primaryInterface) {
        this['primary_interface'] = primaryInterface;
        return this;
    };
    Object.defineProperty(BindingVifDetails.prototype, "primaryInterface", {
        get: function () {
            return this['primary_interface'];
        },
        set: function (primaryInterface) {
            this['primary_interface'] = primaryInterface;
        },
        enumerable: false,
        configurable: true
    });
    BindingVifDetails.prototype.withPortFilter = function (portFilter) {
        this['port_filter'] = portFilter;
        return this;
    };
    Object.defineProperty(BindingVifDetails.prototype, "portFilter", {
        get: function () {
            return this['port_filter'];
        },
        set: function (portFilter) {
            this['port_filter'] = portFilter;
        },
        enumerable: false,
        configurable: true
    });
    BindingVifDetails.prototype.withOvsHybridPlug = function (ovsHybridPlug) {
        this['ovs_hybrid_plug'] = ovsHybridPlug;
        return this;
    };
    Object.defineProperty(BindingVifDetails.prototype, "ovsHybridPlug", {
        get: function () {
            return this['ovs_hybrid_plug'];
        },
        set: function (ovsHybridPlug) {
            this['ovs_hybrid_plug'] = ovsHybridPlug;
        },
        enumerable: false,
        configurable: true
    });
    return BindingVifDetails;
}());
exports.BindingVifDetails = BindingVifDetails;
