// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateVpcPeeringOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVpcPeeringOption = void 0;
var CreateVpcPeeringOption = /** @class */ (function () {
    function CreateVpcPeeringOption(name, requestVpcInfo, acceptVpcInfo) {
        this['name'] = name;
        this['request_vpc_info'] = requestVpcInfo;
        this['accept_vpc_info'] = acceptVpcInfo;
    }
    CreateVpcPeeringOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateVpcPeeringOption.prototype.withRequestVpcInfo = function (requestVpcInfo) {
        this['request_vpc_info'] = requestVpcInfo;
        return this;
    };
    Object.defineProperty(CreateVpcPeeringOption.prototype, "requestVpcInfo", {
        get: function () {
            return this['request_vpc_info'];
        },
        set: function (requestVpcInfo) {
            this['request_vpc_info'] = requestVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    CreateVpcPeeringOption.prototype.withAcceptVpcInfo = function (acceptVpcInfo) {
        this['accept_vpc_info'] = acceptVpcInfo;
        return this;
    };
    Object.defineProperty(CreateVpcPeeringOption.prototype, "acceptVpcInfo", {
        get: function () {
            return this['accept_vpc_info'];
        },
        set: function (acceptVpcInfo) {
            this['accept_vpc_info'] = acceptVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    return CreateVpcPeeringOption;
}());
exports.CreateVpcPeeringOption = CreateVpcPeeringOption;
