// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateVpcRouteOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVpcRouteOptionTypeEnum = exports.CreateVpcRouteOption = void 0;
var CreateVpcRouteOption = /** @class */ (function () {
    function CreateVpcRouteOption(destination, nexthop, type, vpcId) {
        this['destination'] = destination;
        this['nexthop'] = nexthop;
        this['type'] = type;
        this['vpc_id'] = vpcId;
    }
    CreateVpcRouteOption.prototype.withDestination = function (destination) {
        this['destination'] = destination;
        return this;
    };
    CreateVpcRouteOption.prototype.withNexthop = function (nexthop) {
        this['nexthop'] = nexthop;
        return this;
    };
    CreateVpcRouteOption.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    CreateVpcRouteOption.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(CreateVpcRouteOption.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateVpcRouteOption;
}());
exports.CreateVpcRouteOption = CreateVpcRouteOption;
/**
    * @export
    * @enum {string}
    */
var CreateVpcRouteOptionTypeEnum;
(function (CreateVpcRouteOptionTypeEnum) {
    CreateVpcRouteOptionTypeEnum["PEERING"] = "peering";
})(CreateVpcRouteOptionTypeEnum = exports.CreateVpcRouteOptionTypeEnum || (exports.CreateVpcRouteOptionTypeEnum = {}));
