// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateSecurityGroupRuleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSecurityGroupRuleOption = void 0;
var CreateSecurityGroupRuleOption = /** @class */ (function () {
    function CreateSecurityGroupRuleOption(securityGroupId, direction) {
        this['security_group_id'] = securityGroupId;
        this['direction'] = direction;
    }
    CreateSecurityGroupRuleOption.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withPortRangeMin = function (portRangeMin) {
        this['port_range_min'] = portRangeMin;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "portRangeMin", {
        get: function () {
            return this['port_range_min'];
        },
        set: function (portRangeMin) {
            this['port_range_min'] = portRangeMin;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withPortRangeMax = function (portRangeMax) {
        this['port_range_max'] = portRangeMax;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "portRangeMax", {
        get: function () {
            return this['port_range_max'];
        },
        set: function (portRangeMax) {
            this['port_range_max'] = portRangeMax;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateSecurityGroupRuleOption;
}());
exports.CreateSecurityGroupRuleOption = CreateSecurityGroupRuleOption;
