// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreatePortOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePortOption = void 0;
var CreatePortOption = /** @class */ (function () {
    function CreatePortOption(networkId) {
        this['network_id'] = networkId;
    }
    CreatePortOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreatePortOption.prototype.withNetworkId = function (networkId) {
        this['network_id'] = networkId;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "networkId", {
        get: function () {
            return this['network_id'];
        },
        set: function (networkId) {
            this['network_id'] = networkId;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withFixedIps = function (fixedIps) {
        this['fixed_ips'] = fixedIps;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "fixedIps", {
        get: function () {
            return this['fixed_ips'];
        },
        set: function (fixedIps) {
            this['fixed_ips'] = fixedIps;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withDeviceOwner = function (deviceOwner) {
        this['device_owner'] = deviceOwner;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "deviceOwner", {
        get: function () {
            return this['device_owner'];
        },
        set: function (deviceOwner) {
            this['device_owner'] = deviceOwner;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withAdminStateUp = function (adminStateUp) {
        this['admin_state_up'] = adminStateUp;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "adminStateUp", {
        get: function () {
            return this['admin_state_up'];
        },
        set: function (adminStateUp) {
            this['admin_state_up'] = adminStateUp;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withAllowedAddressPairs = function (allowedAddressPairs) {
        this['allowed_address_pairs'] = allowedAddressPairs;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "allowedAddressPairs", {
        get: function () {
            return this['allowed_address_pairs'];
        },
        set: function (allowedAddressPairs) {
            this['allowed_address_pairs'] = allowedAddressPairs;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withExtraDhcpOpts = function (extraDhcpOpts) {
        this['extra_dhcp_opts'] = extraDhcpOpts;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "extraDhcpOpts", {
        get: function () {
            return this['extra_dhcp_opts'];
        },
        set: function (extraDhcpOpts) {
            this['extra_dhcp_opts'] = extraDhcpOpts;
        },
        enumerable: false,
        configurable: true
    });
    CreatePortOption.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(CreatePortOption.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return CreatePortOption;
}());
exports.CreatePortOption = CreatePortOption;
