// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateSecurityGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSecurityGroupOption = void 0;
var CreateSecurityGroupOption = /** @class */ (function () {
    function CreateSecurityGroupOption(name) {
        this['name'] = name;
    }
    CreateSecurityGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateSecurityGroupOption.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupOption.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupOption.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupOption.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateSecurityGroupOption;
}());
exports.CreateSecurityGroupOption = CreateSecurityGroupOption;
