// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateFlowLogReq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFlowLogReqTrafficTypeEnum = exports.CreateFlowLogReqResourceTypeEnum = exports.CreateFlowLogReq = void 0;
var CreateFlowLogReq = /** @class */ (function () {
    function CreateFlowLogReq(resourceType, resourceId, trafficType, logGroupId, logTopicId) {
        this['resource_type'] = resourceType;
        this['resource_id'] = resourceId;
        this['traffic_type'] = trafficType;
        this['log_group_id'] = logGroupId;
        this['log_topic_id'] = logTopicId;
    }
    CreateFlowLogReq.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateFlowLogReq.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateFlowLogReq.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    CreateFlowLogReq.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFlowLogReq.prototype.withTrafficType = function (trafficType) {
        this['traffic_type'] = trafficType;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "trafficType", {
        get: function () {
            return this['traffic_type'];
        },
        set: function (trafficType) {
            this['traffic_type'] = trafficType;
        },
        enumerable: false,
        configurable: true
    });
    CreateFlowLogReq.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFlowLogReq.prototype.withLogTopicId = function (logTopicId) {
        this['log_topic_id'] = logTopicId;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "logTopicId", {
        get: function () {
            return this['log_topic_id'];
        },
        set: function (logTopicId) {
            this['log_topic_id'] = logTopicId;
        },
        enumerable: false,
        configurable: true
    });
    CreateFlowLogReq.prototype.withIndexEnabled = function (indexEnabled) {
        this['index_enabled'] = indexEnabled;
        return this;
    };
    Object.defineProperty(CreateFlowLogReq.prototype, "indexEnabled", {
        get: function () {
            return this['index_enabled'];
        },
        set: function (indexEnabled) {
            this['index_enabled'] = indexEnabled;
        },
        enumerable: false,
        configurable: true
    });
    return CreateFlowLogReq;
}());
exports.CreateFlowLogReq = CreateFlowLogReq;
/**
    * @export
    * @enum {string}
    */
var CreateFlowLogReqResourceTypeEnum;
(function (CreateFlowLogReqResourceTypeEnum) {
    CreateFlowLogReqResourceTypeEnum["PORT"] = "port";
    CreateFlowLogReqResourceTypeEnum["NETWORK"] = "network";
    CreateFlowLogReqResourceTypeEnum["VPC"] = "vpc";
})(CreateFlowLogReqResourceTypeEnum = exports.CreateFlowLogReqResourceTypeEnum || (exports.CreateFlowLogReqResourceTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var CreateFlowLogReqTrafficTypeEnum;
(function (CreateFlowLogReqTrafficTypeEnum) {
    CreateFlowLogReqTrafficTypeEnum["ALL"] = "all";
    CreateFlowLogReqTrafficTypeEnum["ACCEPT"] = "accept";
    CreateFlowLogReqTrafficTypeEnum["REJECT"] = "reject";
})(CreateFlowLogReqTrafficTypeEnum = exports.CreateFlowLogReqTrafficTypeEnum || (exports.CreateFlowLogReqTrafficTypeEnum = {}));
