// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateRouteTableReq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRouteTableReq = void 0;
var CreateRouteTableReq = /** @class */ (function () {
    function CreateRouteTableReq(vpcId) {
        this['vpc_id'] = vpcId;
    }
    CreateRouteTableReq.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateRouteTableReq.prototype.withRoutes = function (routes) {
        this['routes'] = routes;
        return this;
    };
    CreateRouteTableReq.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(CreateRouteTableReq.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    CreateRouteTableReq.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return CreateRouteTableReq;
}());
exports.CreateRouteTableReq = CreateRouteTableReq;
