// @huaweicloud/huaweicloud-sdk-vpc/v2/model/CreateVpcOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVpcOption = void 0;
var CreateVpcOption = /** @class */ (function () {
    function CreateVpcOption() {
    }
    CreateVpcOption.prototype.withCidr = function (cidr) {
        this['cidr'] = cidr;
        return this;
    };
    CreateVpcOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateVpcOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateVpcOption.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateVpcOption.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateVpcOption;
}());
exports.CreateVpcOption = CreateVpcOption;
