// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ExtraDhcpOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExtraDhcpOptionOptNameEnum = exports.ExtraDhcpOption = void 0;
var ExtraDhcpOption = /** @class */ (function () {
    function ExtraDhcpOption(optName) {
        this['opt_name'] = optName;
    }
    ExtraDhcpOption.prototype.withOptName = function (optName) {
        this['opt_name'] = optName;
        return this;
    };
    Object.defineProperty(ExtraDhcpOption.prototype, "optName", {
        get: function () {
            return this['opt_name'];
        },
        set: function (optName) {
            this['opt_name'] = optName;
        },
        enumerable: false,
        configurable: true
    });
    ExtraDhcpOption.prototype.withOptValue = function (optValue) {
        this['opt_value'] = optValue;
        return this;
    };
    Object.defineProperty(ExtraDhcpOption.prototype, "optValue", {
        get: function () {
            return this['opt_value'];
        },
        set: function (optValue) {
            this['opt_value'] = optValue;
        },
        enumerable: false,
        configurable: true
    });
    return ExtraDhcpOption;
}());
exports.ExtraDhcpOption = ExtraDhcpOption;
/**
    * @export
    * @enum {string}
    */
var ExtraDhcpOptionOptNameEnum;
(function (ExtraDhcpOptionOptNameEnum) {
    ExtraDhcpOptionOptNameEnum["NTP"] = "ntp";
    ExtraDhcpOptionOptNameEnum["ADDRESSTIME"] = "addresstime";
})(ExtraDhcpOptionOptNameEnum = exports.ExtraDhcpOptionOptNameEnum || (exports.ExtraDhcpOptionOptNameEnum = {}));
