// @huaweicloud/huaweicloud-sdk-vpc/v2/model/FlowLogResp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlowLogRespStatusEnum = exports.FlowLogRespLogStoreTypeEnum = exports.FlowLogRespTrafficTypeEnum = exports.FlowLogRespResourceTypeEnum = exports.FlowLogResp = void 0;
var FlowLogResp = /** @class */ (function () {
    function FlowLogResp(id, name, tenantId, description, resourceType, resourceId, trafficType, logGroupId, logTopicId, logStoreType, createdAt, updatedAt, adminState, status) {
        this['id'] = id;
        this['name'] = name;
        this['tenant_id'] = tenantId;
        this['description'] = description;
        this['resource_type'] = resourceType;
        this['resource_id'] = resourceId;
        this['traffic_type'] = trafficType;
        this['log_group_id'] = logGroupId;
        this['log_topic_id'] = logTopicId;
        this['log_store_type'] = logStoreType;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['admin_state'] = adminState;
        this['status'] = status;
    }
    FlowLogResp.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    FlowLogResp.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    FlowLogResp.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    FlowLogResp.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withTrafficType = function (trafficType) {
        this['traffic_type'] = trafficType;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "trafficType", {
        get: function () {
            return this['traffic_type'];
        },
        set: function (trafficType) {
            this['traffic_type'] = trafficType;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withLogTopicId = function (logTopicId) {
        this['log_topic_id'] = logTopicId;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "logTopicId", {
        get: function () {
            return this['log_topic_id'];
        },
        set: function (logTopicId) {
            this['log_topic_id'] = logTopicId;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withLogStoreType = function (logStoreType) {
        this['log_store_type'] = logStoreType;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "logStoreType", {
        get: function () {
            return this['log_store_type'];
        },
        set: function (logStoreType) {
            this['log_store_type'] = logStoreType;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withAdminState = function (adminState) {
        this['admin_state'] = adminState;
        return this;
    };
    Object.defineProperty(FlowLogResp.prototype, "adminState", {
        get: function () {
            return this['admin_state'];
        },
        set: function (adminState) {
            this['admin_state'] = adminState;
        },
        enumerable: false,
        configurable: true
    });
    FlowLogResp.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    return FlowLogResp;
}());
exports.FlowLogResp = FlowLogResp;
/**
    * @export
    * @enum {string}
    */
var FlowLogRespResourceTypeEnum;
(function (FlowLogRespResourceTypeEnum) {
    FlowLogRespResourceTypeEnum["PORT"] = "port";
    FlowLogRespResourceTypeEnum["NETWORK"] = "network";
    FlowLogRespResourceTypeEnum["VPC"] = "vpc";
})(FlowLogRespResourceTypeEnum = exports.FlowLogRespResourceTypeEnum || (exports.FlowLogRespResourceTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var FlowLogRespTrafficTypeEnum;
(function (FlowLogRespTrafficTypeEnum) {
    FlowLogRespTrafficTypeEnum["ALL"] = "all";
    FlowLogRespTrafficTypeEnum["ACCEPT"] = "accept";
    FlowLogRespTrafficTypeEnum["REJECT"] = "reject";
})(FlowLogRespTrafficTypeEnum = exports.FlowLogRespTrafficTypeEnum || (exports.FlowLogRespTrafficTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var FlowLogRespLogStoreTypeEnum;
(function (FlowLogRespLogStoreTypeEnum) {
    FlowLogRespLogStoreTypeEnum["LTS"] = "lts";
})(FlowLogRespLogStoreTypeEnum = exports.FlowLogRespLogStoreTypeEnum || (exports.FlowLogRespLogStoreTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var FlowLogRespStatusEnum;
(function (FlowLogRespStatusEnum) {
    FlowLogRespStatusEnum["ACTIVE"] = "ACTIVE";
    FlowLogRespStatusEnum["DOWN"] = "DOWN";
    FlowLogRespStatusEnum["ERROR"] = "ERROR";
})(FlowLogRespStatusEnum = exports.FlowLogRespStatusEnum || (exports.FlowLogRespStatusEnum = {}));
