// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListVpcRoutesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcRoutesRequestTypeEnum = exports.ListVpcRoutesRequest = void 0;
var ListVpcRoutesRequest = /** @class */ (function () {
    function ListVpcRoutesRequest() {
    }
    ListVpcRoutesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListVpcRoutesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListVpcRoutesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListVpcRoutesRequest.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    ListVpcRoutesRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(ListVpcRoutesRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListVpcRoutesRequest.prototype.withDestination = function (destination) {
        this['destination'] = destination;
        return this;
    };
    ListVpcRoutesRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(ListVpcRoutesRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return ListVpcRoutesRequest;
}());
exports.ListVpcRoutesRequest = ListVpcRoutesRequest;
/**
    * @export
    * @enum {string}
    */
var ListVpcRoutesRequestTypeEnum;
(function (ListVpcRoutesRequestTypeEnum) {
    ListVpcRoutesRequestTypeEnum["PEERING"] = "peering";
})(ListVpcRoutesRequestTypeEnum = exports.ListVpcRoutesRequestTypeEnum || (exports.ListVpcRoutesRequestTypeEnum = {}));
