// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListVpcPeeringsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcPeeringsRequestStatusEnum = exports.ListVpcPeeringsRequest = void 0;
var ListVpcPeeringsRequest = /** @class */ (function () {
    function ListVpcPeeringsRequest() {
    }
    ListVpcPeeringsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListVpcPeeringsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListVpcPeeringsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListVpcPeeringsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListVpcPeeringsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListVpcPeeringsRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(ListVpcPeeringsRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    ListVpcPeeringsRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(ListVpcPeeringsRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    return ListVpcPeeringsRequest;
}());
exports.ListVpcPeeringsRequest = ListVpcPeeringsRequest;
/**
    * @export
    * @enum {string}
    */
var ListVpcPeeringsRequestStatusEnum;
(function (ListVpcPeeringsRequestStatusEnum) {
    ListVpcPeeringsRequestStatusEnum["PENDING_ACCEPTANCE"] = "PENDING_ACCEPTANCE";
    ListVpcPeeringsRequestStatusEnum["REJECTED"] = "REJECTED";
    ListVpcPeeringsRequestStatusEnum["EXPIRED"] = "EXPIRED";
    ListVpcPeeringsRequestStatusEnum["DELETED"] = "DELETED";
    ListVpcPeeringsRequestStatusEnum["ACTIVE"] = "ACTIVE";
})(ListVpcPeeringsRequestStatusEnum = exports.ListVpcPeeringsRequestStatusEnum || (exports.ListVpcPeeringsRequestStatusEnum = {}));
