// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListSecurityGroupsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityGroupsRequest = void 0;
var ListSecurityGroupsRequest = /** @class */ (function () {
    function ListSecurityGroupsRequest() {
    }
    ListSecurityGroupsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListSecurityGroupsRequest.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsRequest.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ListSecurityGroupsRequest;
}());
exports.ListSecurityGroupsRequest = ListSecurityGroupsRequest;
