// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListRouteTablesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRouteTablesRequest = void 0;
var ListRouteTablesRequest = /** @class */ (function () {
    function ListRouteTablesRequest() {
    }
    ListRouteTablesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListRouteTablesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListRouteTablesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListRouteTablesRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(ListRouteTablesRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListRouteTablesRequest.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(ListRouteTablesRequest.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    return ListRouteTablesRequest;
}());
exports.ListRouteTablesRequest = ListRouteTablesRequest;
