// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListVpcsByTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcsByTagsRequestBodyActionEnum = exports.ListVpcsByTagsRequestBody = void 0;
var ListVpcsByTagsRequestBody = /** @class */ (function () {
    function ListVpcsByTagsRequestBody(action) {
        this['action'] = action;
    }
    ListVpcsByTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    ListVpcsByTagsRequestBody.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListVpcsByTagsRequestBody.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListVpcsByTagsRequestBody.prototype.withMatches = function (matches) {
        this['matches'] = matches;
        return this;
    };
    ListVpcsByTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return ListVpcsByTagsRequestBody;
}());
exports.ListVpcsByTagsRequestBody = ListVpcsByTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var ListVpcsByTagsRequestBodyActionEnum;
(function (ListVpcsByTagsRequestBodyActionEnum) {
    ListVpcsByTagsRequestBodyActionEnum["FILTER"] = "filter";
    ListVpcsByTagsRequestBodyActionEnum["COUNT"] = "count";
})(ListVpcsByTagsRequestBodyActionEnum = exports.ListVpcsByTagsRequestBodyActionEnum || (exports.ListVpcsByTagsRequestBodyActionEnum = {}));
