// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListPrivateipsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPrivateipsRequest = void 0;
var ListPrivateipsRequest = /** @class */ (function () {
    function ListPrivateipsRequest(subnetId) {
        this['subnet_id'] = subnetId;
    }
    ListPrivateipsRequest.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(ListPrivateipsRequest.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    ListPrivateipsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListPrivateipsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    return ListPrivateipsRequest;
}());
exports.ListPrivateipsRequest = ListPrivateipsRequest;
