// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListResourceResp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourceResp = void 0;
var ListResourceResp = /** @class */ (function () {
    function ListResourceResp(resourceDetail, resourceId, resourceName) {
        this['resource_detail'] = resourceDetail;
        this['resource_id'] = resourceId;
        this['resource_name'] = resourceName;
    }
    ListResourceResp.prototype.withResourceDetail = function (resourceDetail) {
        this['resource_detail'] = resourceDetail;
        return this;
    };
    Object.defineProperty(ListResourceResp.prototype, "resourceDetail", {
        get: function () {
            return this['resource_detail'];
        },
        set: function (resourceDetail) {
            this['resource_detail'] = resourceDetail;
        },
        enumerable: false,
        configurable: true
    });
    ListResourceResp.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(ListResourceResp.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    ListResourceResp.prototype.withResourceName = function (resourceName) {
        this['resource_name'] = resourceName;
        return this;
    };
    Object.defineProperty(ListResourceResp.prototype, "resourceName", {
        get: function () {
            return this['resource_name'];
        },
        set: function (resourceName) {
            this['resource_name'] = resourceName;
        },
        enumerable: false,
        configurable: true
    });
    ListResourceResp.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return ListResourceResp;
}());
exports.ListResourceResp = ListResourceResp;
