// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListFlowLogsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListFlowLogsRequestStatusEnum = exports.ListFlowLogsRequestLogStoreTypeEnum = exports.ListFlowLogsRequestTrafficTypeEnum = exports.ListFlowLogsRequestResourceTypeEnum = exports.ListFlowLogsRequest = void 0;
var ListFlowLogsRequest = /** @class */ (function () {
    function ListFlowLogsRequest() {
    }
    ListFlowLogsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListFlowLogsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListFlowLogsRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListFlowLogsRequest.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withTrafficType = function (trafficType) {
        this['traffic_type'] = trafficType;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "trafficType", {
        get: function () {
            return this['traffic_type'];
        },
        set: function (trafficType) {
            this['traffic_type'] = trafficType;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withLogGroupId = function (logGroupId) {
        this['log_group_id'] = logGroupId;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "logGroupId", {
        get: function () {
            return this['log_group_id'];
        },
        set: function (logGroupId) {
            this['log_group_id'] = logGroupId;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withLogTopicId = function (logTopicId) {
        this['log_topic_id'] = logTopicId;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "logTopicId", {
        get: function () {
            return this['log_topic_id'];
        },
        set: function (logTopicId) {
            this['log_topic_id'] = logTopicId;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withLogStoreType = function (logStoreType) {
        this['log_store_type'] = logStoreType;
        return this;
    };
    Object.defineProperty(ListFlowLogsRequest.prototype, "logStoreType", {
        get: function () {
            return this['log_store_type'];
        },
        set: function (logStoreType) {
            this['log_store_type'] = logStoreType;
        },
        enumerable: false,
        configurable: true
    });
    ListFlowLogsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListFlowLogsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListFlowLogsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    return ListFlowLogsRequest;
}());
exports.ListFlowLogsRequest = ListFlowLogsRequest;
/**
    * @export
    * @enum {string}
    */
var ListFlowLogsRequestResourceTypeEnum;
(function (ListFlowLogsRequestResourceTypeEnum) {
    ListFlowLogsRequestResourceTypeEnum["PORT"] = "port";
    ListFlowLogsRequestResourceTypeEnum["NETWORK"] = "network";
    ListFlowLogsRequestResourceTypeEnum["VPC"] = "vpc";
})(ListFlowLogsRequestResourceTypeEnum = exports.ListFlowLogsRequestResourceTypeEnum || (exports.ListFlowLogsRequestResourceTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFlowLogsRequestTrafficTypeEnum;
(function (ListFlowLogsRequestTrafficTypeEnum) {
    ListFlowLogsRequestTrafficTypeEnum["ALL"] = "all";
    ListFlowLogsRequestTrafficTypeEnum["REJECT"] = "reject";
    ListFlowLogsRequestTrafficTypeEnum["ACCEPT"] = "accept";
})(ListFlowLogsRequestTrafficTypeEnum = exports.ListFlowLogsRequestTrafficTypeEnum || (exports.ListFlowLogsRequestTrafficTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFlowLogsRequestLogStoreTypeEnum;
(function (ListFlowLogsRequestLogStoreTypeEnum) {
    ListFlowLogsRequestLogStoreTypeEnum["LTS"] = "lts";
})(ListFlowLogsRequestLogStoreTypeEnum = exports.ListFlowLogsRequestLogStoreTypeEnum || (exports.ListFlowLogsRequestLogStoreTypeEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListFlowLogsRequestStatusEnum;
(function (ListFlowLogsRequestStatusEnum) {
    ListFlowLogsRequestStatusEnum["ACTIVE"] = "ACTIVE";
    ListFlowLogsRequestStatusEnum["DOWN"] = "DOWN";
    ListFlowLogsRequestStatusEnum["ERROR"] = "ERROR";
})(ListFlowLogsRequestStatusEnum = exports.ListFlowLogsRequestStatusEnum || (exports.ListFlowLogsRequestStatusEnum = {}));
