// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListPortsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPortsRequestStatusEnum = exports.ListPortsRequestDeviceOwnerEnum = exports.ListPortsRequest = void 0;
var ListPortsRequest = /** @class */ (function () {
    function ListPortsRequest() {
    }
    ListPortsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListPortsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListPortsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListPortsRequest.prototype.withAdminStateUp = function (adminStateUp) {
        this['admin_state_up'] = adminStateUp;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "adminStateUp", {
        get: function () {
            return this['admin_state_up'];
        },
        set: function (adminStateUp) {
            this['admin_state_up'] = adminStateUp;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withNetworkId = function (networkId) {
        this['network_id'] = networkId;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "networkId", {
        get: function () {
            return this['network_id'];
        },
        set: function (networkId) {
            this['network_id'] = networkId;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withMacAddress = function (macAddress) {
        this['mac_address'] = macAddress;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "macAddress", {
        get: function () {
            return this['mac_address'];
        },
        set: function (macAddress) {
            this['mac_address'] = macAddress;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withDeviceOwner = function (deviceOwner) {
        this['device_owner'] = deviceOwner;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "deviceOwner", {
        get: function () {
            return this['device_owner'];
        },
        set: function (deviceOwner) {
            this['device_owner'] = deviceOwner;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListPortsRequest.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListPortsRequest.prototype.withFixedIps = function (fixedIps) {
        this['fixed_ips'] = fixedIps;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "fixedIps", {
        get: function () {
            return this['fixed_ips'];
        },
        set: function (fixedIps) {
            this['fixed_ips'] = fixedIps;
        },
        enumerable: false,
        configurable: true
    });
    ListPortsRequest.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListPortsRequest.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ListPortsRequest;
}());
exports.ListPortsRequest = ListPortsRequest;
/**
    * @export
    * @enum {string}
    */
var ListPortsRequestDeviceOwnerEnum;
(function (ListPortsRequestDeviceOwnerEnum) {
    ListPortsRequestDeviceOwnerEnum["NETWORKDHCP"] = "network:dhcp";
    ListPortsRequestDeviceOwnerEnum["NEUTRONVIP_PORT"] = "neutron:VIP_PORT";
    ListPortsRequestDeviceOwnerEnum["NETWORKROUTER_INTERFACE_DISTRIBUTED"] = "network:router_interface_distributed";
    ListPortsRequestDeviceOwnerEnum["NETWORKROUTER_CENTRALIZED_SNAT"] = "network:router_centralized_snat";
})(ListPortsRequestDeviceOwnerEnum = exports.ListPortsRequestDeviceOwnerEnum || (exports.ListPortsRequestDeviceOwnerEnum = {}));
/**
    * @export
    * @enum {string}
    */
var ListPortsRequestStatusEnum;
(function (ListPortsRequestStatusEnum) {
    ListPortsRequestStatusEnum["ACTIVE"] = "ACTIVE";
    ListPortsRequestStatusEnum["BUILD"] = "BUILD";
    ListPortsRequestStatusEnum["DOWN"] = "DOWN";
})(ListPortsRequestStatusEnum = exports.ListPortsRequestStatusEnum || (exports.ListPortsRequestStatusEnum = {}));
