// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListSubnetsByTagsRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSubnetsByTagsRequestBodyActionEnum = exports.ListSubnetsByTagsRequestBody = void 0;
var ListSubnetsByTagsRequestBody = /** @class */ (function () {
    function ListSubnetsByTagsRequestBody(action) {
        this['action'] = action;
    }
    ListSubnetsByTagsRequestBody.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    ListSubnetsByTagsRequestBody.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSubnetsByTagsRequestBody.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListSubnetsByTagsRequestBody.prototype.withMatches = function (matches) {
        this['matches'] = matches;
        return this;
    };
    ListSubnetsByTagsRequestBody.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return ListSubnetsByTagsRequestBody;
}());
exports.ListSubnetsByTagsRequestBody = ListSubnetsByTagsRequestBody;
/**
    * @export
    * @enum {string}
    */
var ListSubnetsByTagsRequestBodyActionEnum;
(function (ListSubnetsByTagsRequestBodyActionEnum) {
    ListSubnetsByTagsRequestBodyActionEnum["FILTER"] = "filter";
    ListSubnetsByTagsRequestBodyActionEnum["COUNT"] = "count";
})(ListSubnetsByTagsRequestBodyActionEnum = exports.ListSubnetsByTagsRequestBodyActionEnum || (exports.ListSubnetsByTagsRequestBodyActionEnum = {}));
