// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListVpcsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcsRequest = void 0;
var ListVpcsRequest = /** @class */ (function () {
    function ListVpcsRequest() {
    }
    ListVpcsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListVpcsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListVpcsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListVpcsRequest.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListVpcsRequest.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ListVpcsRequest;
}());
exports.ListVpcsRequest = ListVpcsRequest;
