// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ListSecurityGroupRulesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityGroupRulesRequest = void 0;
var ListSecurityGroupRulesRequest = /** @class */ (function () {
    function ListSecurityGroupRulesRequest() {
    }
    ListSecurityGroupRulesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupRulesRequest.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    return ListSecurityGroupRulesRequest;
}());
exports.ListSecurityGroupRulesRequest = ListSecurityGroupRulesRequest;
