// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronListSecurityGroupRulesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronListSecurityGroupRulesRequest = void 0;
var NeutronListSecurityGroupRulesRequest = /** @class */ (function () {
    function NeutronListSecurityGroupRulesRequest() {
    }
    NeutronListSecurityGroupRulesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronListSecurityGroupRulesRequest.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListSecurityGroupRulesRequest.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListSecurityGroupRulesRequest.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListSecurityGroupRulesRequest.prototype.withPortRangeMax = function (portRangeMax) {
        this['port_range_max'] = portRangeMax;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "portRangeMax", {
        get: function () {
            return this['port_range_max'];
        },
        set: function (portRangeMax) {
            this['port_range_max'] = portRangeMax;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListSecurityGroupRulesRequest.prototype.withPortRangeMin = function (portRangeMin) {
        this['port_range_min'] = portRangeMin;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "portRangeMin", {
        get: function () {
            return this['port_range_min'];
        },
        set: function (portRangeMin) {
            this['port_range_min'] = portRangeMin;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListSecurityGroupRulesRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupRulesRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronListSecurityGroupRulesRequest;
}());
exports.NeutronListSecurityGroupRulesRequest = NeutronListSecurityGroupRulesRequest;
