// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronCreateFirewallPolicyOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronCreateFirewallPolicyOption = void 0;
var NeutronCreateFirewallPolicyOption = /** @class */ (function () {
    function NeutronCreateFirewallPolicyOption() {
    }
    NeutronCreateFirewallPolicyOption.prototype.withAudited = function (audited) {
        this['audited'] = audited;
        return this;
    };
    NeutronCreateFirewallPolicyOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronCreateFirewallPolicyOption.prototype.withFirewallRules = function (firewallRules) {
        this['firewall_rules'] = firewallRules;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallPolicyOption.prototype, "firewallRules", {
        get: function () {
            return this['firewall_rules'];
        },
        set: function (firewallRules) {
            this['firewall_rules'] = firewallRules;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallPolicyOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    return NeutronCreateFirewallPolicyOption;
}());
exports.NeutronCreateFirewallPolicyOption = NeutronCreateFirewallPolicyOption;
