// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronCreateFirewallRuleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronCreateFirewallRuleOptionActionEnum = exports.NeutronCreateFirewallRuleOption = void 0;
var NeutronCreateFirewallRuleOption = /** @class */ (function () {
    function NeutronCreateFirewallRuleOption() {
    }
    NeutronCreateFirewallRuleOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronCreateFirewallRuleOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronCreateFirewallRuleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronCreateFirewallRuleOption.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    NeutronCreateFirewallRuleOption.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallRuleOption.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallRuleOption.prototype.withDestinationIpAddress = function (destinationIpAddress) {
        this['destination_ip_address'] = destinationIpAddress;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallRuleOption.prototype, "destinationIpAddress", {
        get: function () {
            return this['destination_ip_address'];
        },
        set: function (destinationIpAddress) {
            this['destination_ip_address'] = destinationIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallRuleOption.prototype.withDestinationPort = function (destinationPort) {
        this['destination_port'] = destinationPort;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallRuleOption.prototype, "destinationPort", {
        get: function () {
            return this['destination_port'];
        },
        set: function (destinationPort) {
            this['destination_port'] = destinationPort;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallRuleOption.prototype.withSourceIpAddress = function (sourceIpAddress) {
        this['source_ip_address'] = sourceIpAddress;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallRuleOption.prototype, "sourceIpAddress", {
        get: function () {
            return this['source_ip_address'];
        },
        set: function (sourceIpAddress) {
            this['source_ip_address'] = sourceIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallRuleOption.prototype.withSourcePort = function (sourcePort) {
        this['source_port'] = sourcePort;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallRuleOption.prototype, "sourcePort", {
        get: function () {
            return this['source_port'];
        },
        set: function (sourcePort) {
            this['source_port'] = sourcePort;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallRuleOption.prototype.withEnabled = function (enabled) {
        this['enabled'] = enabled;
        return this;
    };
    return NeutronCreateFirewallRuleOption;
}());
exports.NeutronCreateFirewallRuleOption = NeutronCreateFirewallRuleOption;
/**
    * @export
    * @enum {string}
    */
var NeutronCreateFirewallRuleOptionActionEnum;
(function (NeutronCreateFirewallRuleOptionActionEnum) {
    NeutronCreateFirewallRuleOptionActionEnum["DENY"] = "DENY";
    NeutronCreateFirewallRuleOptionActionEnum["ALLOW"] = "ALLOW";
})(NeutronCreateFirewallRuleOptionActionEnum = exports.NeutronCreateFirewallRuleOptionActionEnum || (exports.NeutronCreateFirewallRuleOptionActionEnum = {}));
