// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronListFirewallRulesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronListFirewallRulesRequest = void 0;
var NeutronListFirewallRulesRequest = /** @class */ (function () {
    function NeutronListFirewallRulesRequest() {
    }
    NeutronListFirewallRulesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    NeutronListFirewallRulesRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronListFirewallRulesRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronListFirewallRulesRequest;
}());
exports.NeutronListFirewallRulesRequest = NeutronListFirewallRulesRequest;
