// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronUpdateFirewallRuleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronUpdateFirewallRuleOptionActionEnum = exports.NeutronUpdateFirewallRuleOption = void 0;
var NeutronUpdateFirewallRuleOption = /** @class */ (function () {
    function NeutronUpdateFirewallRuleOption() {
    }
    NeutronUpdateFirewallRuleOption.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    NeutronUpdateFirewallRuleOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronUpdateFirewallRuleOption.prototype.withDestinationIpAddress = function (destinationIpAddress) {
        this['destination_ip_address'] = destinationIpAddress;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallRuleOption.prototype, "destinationIpAddress", {
        get: function () {
            return this['destination_ip_address'];
        },
        set: function (destinationIpAddress) {
            this['destination_ip_address'] = destinationIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronUpdateFirewallRuleOption.prototype.withDestinationPort = function (destinationPort) {
        this['destination_port'] = destinationPort;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallRuleOption.prototype, "destinationPort", {
        get: function () {
            return this['destination_port'];
        },
        set: function (destinationPort) {
            this['destination_port'] = destinationPort;
        },
        enumerable: false,
        configurable: true
    });
    NeutronUpdateFirewallRuleOption.prototype.withEnabled = function (enabled) {
        this['enabled'] = enabled;
        return this;
    };
    NeutronUpdateFirewallRuleOption.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallRuleOption.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    NeutronUpdateFirewallRuleOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronUpdateFirewallRuleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronUpdateFirewallRuleOption.prototype.withSourceIpAddress = function (sourceIpAddress) {
        this['source_ip_address'] = sourceIpAddress;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallRuleOption.prototype, "sourceIpAddress", {
        get: function () {
            return this['source_ip_address'];
        },
        set: function (sourceIpAddress) {
            this['source_ip_address'] = sourceIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronUpdateFirewallRuleOption.prototype.withSourcePort = function (sourcePort) {
        this['source_port'] = sourcePort;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallRuleOption.prototype, "sourcePort", {
        get: function () {
            return this['source_port'];
        },
        set: function (sourcePort) {
            this['source_port'] = sourcePort;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronUpdateFirewallRuleOption;
}());
exports.NeutronUpdateFirewallRuleOption = NeutronUpdateFirewallRuleOption;
/**
    * @export
    * @enum {string}
    */
var NeutronUpdateFirewallRuleOptionActionEnum;
(function (NeutronUpdateFirewallRuleOptionActionEnum) {
    NeutronUpdateFirewallRuleOptionActionEnum["DENY"] = "DENY";
    NeutronUpdateFirewallRuleOptionActionEnum["ALLOW"] = "ALLOW";
})(NeutronUpdateFirewallRuleOptionActionEnum = exports.NeutronUpdateFirewallRuleOptionActionEnum || (exports.NeutronUpdateFirewallRuleOptionActionEnum = {}));
