// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronSecurityGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronSecurityGroup = void 0;
var NeutronSecurityGroup = /** @class */ (function () {
    function NeutronSecurityGroup(description, id, name, securityGroupRules, tenantId, projectId, createdAt, updatedAt) {
        this['description'] = description;
        this['id'] = id;
        this['name'] = name;
        this['security_group_rules'] = securityGroupRules;
        this['tenant_id'] = tenantId;
        this['project_id'] = projectId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    NeutronSecurityGroup.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronSecurityGroup.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronSecurityGroup.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronSecurityGroup.prototype.withSecurityGroupRules = function (securityGroupRules) {
        this['security_group_rules'] = securityGroupRules;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroup.prototype, "securityGroupRules", {
        get: function () {
            return this['security_group_rules'];
        },
        set: function (securityGroupRules) {
            this['security_group_rules'] = securityGroupRules;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroup.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroup.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroup.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroup.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroup.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroup.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroup.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroup.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronSecurityGroup;
}());
exports.NeutronSecurityGroup = NeutronSecurityGroup;
