// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronFirewallRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronFirewallRuleActionEnum = exports.NeutronFirewallRule = void 0;
var NeutronFirewallRule = /** @class */ (function () {
    function NeutronFirewallRule(id, name, description, action, protocol, ipVersion, enabled, _public, destinationIpAddress, destinationPort, sourceIpAddress, sourcePort, tenantId, projectId) {
        this['id'] = id;
        this['name'] = name;
        this['description'] = description;
        this['action'] = action;
        this['protocol'] = protocol;
        this['ip_version'] = ipVersion;
        this['enabled'] = enabled;
        this['public'] = _public;
        this['destination_ip_address'] = destinationIpAddress;
        this['destination_port'] = destinationPort;
        this['source_ip_address'] = sourceIpAddress;
        this['source_port'] = sourcePort;
        this['tenant_id'] = tenantId;
        this['project_id'] = projectId;
    }
    NeutronFirewallRule.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronFirewallRule.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronFirewallRule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronFirewallRule.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    NeutronFirewallRule.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronFirewallRule.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withEnabled = function (enabled) {
        this['enabled'] = enabled;
        return this;
    };
    NeutronFirewallRule.prototype.withPublic = function (_public) {
        this['public'] = _public;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "_public", {
        get: function () {
            return this['public'];
        },
        set: function (_public) {
            this['public'] = _public;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withDestinationIpAddress = function (destinationIpAddress) {
        this['destination_ip_address'] = destinationIpAddress;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "destinationIpAddress", {
        get: function () {
            return this['destination_ip_address'];
        },
        set: function (destinationIpAddress) {
            this['destination_ip_address'] = destinationIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withDestinationPort = function (destinationPort) {
        this['destination_port'] = destinationPort;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "destinationPort", {
        get: function () {
            return this['destination_port'];
        },
        set: function (destinationPort) {
            this['destination_port'] = destinationPort;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withSourceIpAddress = function (sourceIpAddress) {
        this['source_ip_address'] = sourceIpAddress;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "sourceIpAddress", {
        get: function () {
            return this['source_ip_address'];
        },
        set: function (sourceIpAddress) {
            this['source_ip_address'] = sourceIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withSourcePort = function (sourcePort) {
        this['source_port'] = sourcePort;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "sourcePort", {
        get: function () {
            return this['source_port'];
        },
        set: function (sourcePort) {
            this['source_port'] = sourcePort;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallRule.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronFirewallRule.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronFirewallRule;
}());
exports.NeutronFirewallRule = NeutronFirewallRule;
/**
    * @export
    * @enum {string}
    */
var NeutronFirewallRuleActionEnum;
(function (NeutronFirewallRuleActionEnum) {
    NeutronFirewallRuleActionEnum["DENY"] = "DENY";
    NeutronFirewallRuleActionEnum["ALLOW"] = "ALLOW";
})(NeutronFirewallRuleActionEnum = exports.NeutronFirewallRuleActionEnum || (exports.NeutronFirewallRuleActionEnum = {}));
