// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronCreateSecurityGroupRuleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronCreateSecurityGroupRuleOptionEthertypeEnum = exports.NeutronCreateSecurityGroupRuleOptionDirectionEnum = exports.NeutronCreateSecurityGroupRuleOption = void 0;
var NeutronCreateSecurityGroupRuleOption = /** @class */ (function () {
    function NeutronCreateSecurityGroupRuleOption(direction, securityGroupId) {
        this['direction'] = direction;
        this['security_group_id'] = securityGroupId;
    }
    NeutronCreateSecurityGroupRuleOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronCreateSecurityGroupRuleOption.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    NeutronCreateSecurityGroupRuleOption.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    NeutronCreateSecurityGroupRuleOption.prototype.withPortRangeMax = function (portRangeMax) {
        this['port_range_max'] = portRangeMax;
        return this;
    };
    Object.defineProperty(NeutronCreateSecurityGroupRuleOption.prototype, "portRangeMax", {
        get: function () {
            return this['port_range_max'];
        },
        set: function (portRangeMax) {
            this['port_range_max'] = portRangeMax;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateSecurityGroupRuleOption.prototype.withPortRangeMin = function (portRangeMin) {
        this['port_range_min'] = portRangeMin;
        return this;
    };
    Object.defineProperty(NeutronCreateSecurityGroupRuleOption.prototype, "portRangeMin", {
        get: function () {
            return this['port_range_min'];
        },
        set: function (portRangeMin) {
            this['port_range_min'] = portRangeMin;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateSecurityGroupRuleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronCreateSecurityGroupRuleOption.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(NeutronCreateSecurityGroupRuleOption.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateSecurityGroupRuleOption.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(NeutronCreateSecurityGroupRuleOption.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateSecurityGroupRuleOption.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(NeutronCreateSecurityGroupRuleOption.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronCreateSecurityGroupRuleOption;
}());
exports.NeutronCreateSecurityGroupRuleOption = NeutronCreateSecurityGroupRuleOption;
/**
    * @export
    * @enum {string}
    */
var NeutronCreateSecurityGroupRuleOptionDirectionEnum;
(function (NeutronCreateSecurityGroupRuleOptionDirectionEnum) {
    NeutronCreateSecurityGroupRuleOptionDirectionEnum["INGRESS"] = "ingress";
    NeutronCreateSecurityGroupRuleOptionDirectionEnum["EGRESS"] = "egress";
})(NeutronCreateSecurityGroupRuleOptionDirectionEnum = exports.NeutronCreateSecurityGroupRuleOptionDirectionEnum || (exports.NeutronCreateSecurityGroupRuleOptionDirectionEnum = {}));
/**
    * @export
    * @enum {string}
    */
var NeutronCreateSecurityGroupRuleOptionEthertypeEnum;
(function (NeutronCreateSecurityGroupRuleOptionEthertypeEnum) {
    NeutronCreateSecurityGroupRuleOptionEthertypeEnum["IPV4"] = "IPv4";
    NeutronCreateSecurityGroupRuleOptionEthertypeEnum["IPV6"] = "IPv6";
})(NeutronCreateSecurityGroupRuleOptionEthertypeEnum = exports.NeutronCreateSecurityGroupRuleOptionEthertypeEnum || (exports.NeutronCreateSecurityGroupRuleOptionEthertypeEnum = {}));
