// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronSecurityGroupRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronSecurityGroupRuleDirectionEnum = exports.NeutronSecurityGroupRule = void 0;
var NeutronSecurityGroupRule = /** @class */ (function () {
    function NeutronSecurityGroupRule(description, direction, ethertype, id, portRangeMax, portRangeMin, protocol, remoteGroupId, remoteIpPrefix, securityGroupId, tenantId, projectId, createdAt, updatedAt) {
        this['description'] = description;
        this['direction'] = direction;
        this['ethertype'] = ethertype;
        this['id'] = id;
        this['port_range_max'] = portRangeMax;
        this['port_range_min'] = portRangeMin;
        this['protocol'] = protocol;
        this['remote_group_id'] = remoteGroupId;
        this['remote_ip_prefix'] = remoteIpPrefix;
        this['security_group_id'] = securityGroupId;
        this['tenant_id'] = tenantId;
        this['project_id'] = projectId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    NeutronSecurityGroupRule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronSecurityGroupRule.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    NeutronSecurityGroupRule.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    NeutronSecurityGroupRule.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronSecurityGroupRule.prototype.withPortRangeMax = function (portRangeMax) {
        this['port_range_max'] = portRangeMax;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "portRangeMax", {
        get: function () {
            return this['port_range_max'];
        },
        set: function (portRangeMax) {
            this['port_range_max'] = portRangeMax;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withPortRangeMin = function (portRangeMin) {
        this['port_range_min'] = portRangeMin;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "portRangeMin", {
        get: function () {
            return this['port_range_min'];
        },
        set: function (portRangeMin) {
            this['port_range_min'] = portRangeMin;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    NeutronSecurityGroupRule.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withRemoteAddressGroupId = function (remoteAddressGroupId) {
        this['remote_address_group_id'] = remoteAddressGroupId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "remoteAddressGroupId", {
        get: function () {
            return this['remote_address_group_id'];
        },
        set: function (remoteAddressGroupId) {
            this['remote_address_group_id'] = remoteAddressGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    NeutronSecurityGroupRule.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(NeutronSecurityGroupRule.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronSecurityGroupRule;
}());
exports.NeutronSecurityGroupRule = NeutronSecurityGroupRule;
/**
    * @export
    * @enum {string}
    */
var NeutronSecurityGroupRuleDirectionEnum;
(function (NeutronSecurityGroupRuleDirectionEnum) {
    NeutronSecurityGroupRuleDirectionEnum["INGRESS"] = "ingress";
    NeutronSecurityGroupRuleDirectionEnum["EGRESS"] = "egress";
})(NeutronSecurityGroupRuleDirectionEnum = exports.NeutronSecurityGroupRuleDirectionEnum || (exports.NeutronSecurityGroupRuleDirectionEnum = {}));
