// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronListFirewallGroupsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronListFirewallGroupsRequest = void 0;
var NeutronListFirewallGroupsRequest = /** @class */ (function () {
    function NeutronListFirewallGroupsRequest() {
    }
    NeutronListFirewallGroupsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NeutronListFirewallGroupsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NeutronListFirewallGroupsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronListFirewallGroupsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronListFirewallGroupsRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronListFirewallGroupsRequest.prototype.withIngressFirewallPolicyId = function (ingressFirewallPolicyId) {
        this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronListFirewallGroupsRequest.prototype, "ingressFirewallPolicyId", {
        get: function () {
            return this['ingress_firewall_policy_id'];
        },
        set: function (ingressFirewallPolicyId) {
            this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronListFirewallGroupsRequest.prototype.withEgressFirewallPolicyId = function (egressFirewallPolicyId) {
        this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronListFirewallGroupsRequest.prototype, "egressFirewallPolicyId", {
        get: function () {
            return this['egress_firewall_policy_id'];
        },
        set: function (egressFirewallPolicyId) {
            this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronListFirewallGroupsRequest;
}());
exports.NeutronListFirewallGroupsRequest = NeutronListFirewallGroupsRequest;
