// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronUpdateFirewallPolicyOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronUpdateFirewallPolicyOption = void 0;
var NeutronUpdateFirewallPolicyOption = /** @class */ (function () {
    function NeutronUpdateFirewallPolicyOption() {
    }
    NeutronUpdateFirewallPolicyOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronUpdateFirewallPolicyOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronUpdateFirewallPolicyOption.prototype.withFirewallRules = function (firewallRules) {
        this['firewall_rules'] = firewallRules;
        return this;
    };
    Object.defineProperty(NeutronUpdateFirewallPolicyOption.prototype, "firewallRules", {
        get: function () {
            return this['firewall_rules'];
        },
        set: function (firewallRules) {
            this['firewall_rules'] = firewallRules;
        },
        enumerable: false,
        configurable: true
    });
    NeutronUpdateFirewallPolicyOption.prototype.withAudited = function (audited) {
        this['audited'] = audited;
        return this;
    };
    return NeutronUpdateFirewallPolicyOption;
}());
exports.NeutronUpdateFirewallPolicyOption = NeutronUpdateFirewallPolicyOption;
