// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronFirewallGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronFirewallGroup = void 0;
var NeutronFirewallGroup = /** @class */ (function () {
    function NeutronFirewallGroup(id, name, description, adminStateUp, egressFirewallPolicyId, ingressFirewallPolicyId, ports, _public, status, tenantId, projectId, createdAt, updatedAt) {
        this['id'] = id;
        this['name'] = name;
        this['description'] = description;
        this['admin_state_up'] = adminStateUp;
        this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        this['ports'] = ports;
        this['public'] = _public;
        this['status'] = status;
        this['tenant_id'] = tenantId;
        this['project_id'] = projectId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    NeutronFirewallGroup.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronFirewallGroup.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronFirewallGroup.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronFirewallGroup.prototype.withAdminStateUp = function (adminStateUp) {
        this['admin_state_up'] = adminStateUp;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "adminStateUp", {
        get: function () {
            return this['admin_state_up'];
        },
        set: function (adminStateUp) {
            this['admin_state_up'] = adminStateUp;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withEgressFirewallPolicyId = function (egressFirewallPolicyId) {
        this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "egressFirewallPolicyId", {
        get: function () {
            return this['egress_firewall_policy_id'];
        },
        set: function (egressFirewallPolicyId) {
            this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withIngressFirewallPolicyId = function (ingressFirewallPolicyId) {
        this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "ingressFirewallPolicyId", {
        get: function () {
            return this['ingress_firewall_policy_id'];
        },
        set: function (ingressFirewallPolicyId) {
            this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withPorts = function (ports) {
        this['ports'] = ports;
        return this;
    };
    NeutronFirewallGroup.prototype.withPublic = function (_public) {
        this['public'] = _public;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "_public", {
        get: function () {
            return this['public'];
        },
        set: function (_public) {
            this['public'] = _public;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    NeutronFirewallGroup.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallGroup.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(NeutronFirewallGroup.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronFirewallGroup;
}());
exports.NeutronFirewallGroup = NeutronFirewallGroup;
