// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronListSecurityGroupsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronListSecurityGroupsRequest = void 0;
var NeutronListSecurityGroupsRequest = /** @class */ (function () {
    function NeutronListSecurityGroupsRequest() {
    }
    NeutronListSecurityGroupsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NeutronListSecurityGroupsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NeutronListSecurityGroupsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronListSecurityGroupsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronListSecurityGroupsRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronListSecurityGroupsRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronListSecurityGroupsRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronListSecurityGroupsRequest;
}());
exports.NeutronListSecurityGroupsRequest = NeutronListSecurityGroupsRequest;
