// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronAddFirewallRuleResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronAddFirewallRuleResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var NeutronAddFirewallRuleResponse = /** @class */ (function (_super) {
    __extends(NeutronAddFirewallRuleResponse, _super);
    function NeutronAddFirewallRuleResponse() {
        return _super.call(this) || this;
    }
    NeutronAddFirewallRuleResponse.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronAddFirewallRuleResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronAddFirewallRuleResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronAddFirewallRuleResponse.prototype.withFirewallRules = function (firewallRules) {
        this['firewall_rules'] = firewallRules;
        return this;
    };
    Object.defineProperty(NeutronAddFirewallRuleResponse.prototype, "firewallRules", {
        get: function () {
            return this['firewall_rules'];
        },
        set: function (firewallRules) {
            this['firewall_rules'] = firewallRules;
        },
        enumerable: false,
        configurable: true
    });
    NeutronAddFirewallRuleResponse.prototype.withAudited = function (audited) {
        this['audited'] = audited;
        return this;
    };
    NeutronAddFirewallRuleResponse.prototype.withPublic = function (_public) {
        this['public'] = _public;
        return this;
    };
    Object.defineProperty(NeutronAddFirewallRuleResponse.prototype, "_public", {
        get: function () {
            return this['public'];
        },
        set: function (_public) {
            this['public'] = _public;
        },
        enumerable: false,
        configurable: true
    });
    NeutronAddFirewallRuleResponse.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronAddFirewallRuleResponse.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronAddFirewallRuleResponse.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronAddFirewallRuleResponse.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronAddFirewallRuleResponse;
}(SdkResponse_1.SdkResponse));
exports.NeutronAddFirewallRuleResponse = NeutronAddFirewallRuleResponse;
