// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronCreateFirewallGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronCreateFirewallGroupOption = void 0;
var NeutronCreateFirewallGroupOption = /** @class */ (function () {
    function NeutronCreateFirewallGroupOption() {
    }
    NeutronCreateFirewallGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronCreateFirewallGroupOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronCreateFirewallGroupOption.prototype.withIngressFirewallPolicyId = function (ingressFirewallPolicyId) {
        this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallGroupOption.prototype, "ingressFirewallPolicyId", {
        get: function () {
            return this['ingress_firewall_policy_id'];
        },
        set: function (ingressFirewallPolicyId) {
            this['ingress_firewall_policy_id'] = ingressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallGroupOption.prototype.withEgressFirewallPolicyId = function (egressFirewallPolicyId) {
        this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallGroupOption.prototype, "egressFirewallPolicyId", {
        get: function () {
            return this['egress_firewall_policy_id'];
        },
        set: function (egressFirewallPolicyId) {
            this['egress_firewall_policy_id'] = egressFirewallPolicyId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronCreateFirewallGroupOption.prototype.withPorts = function (ports) {
        this['ports'] = ports;
        return this;
    };
    NeutronCreateFirewallGroupOption.prototype.withAdminStateUp = function (adminStateUp) {
        this['admin_state_up'] = adminStateUp;
        return this;
    };
    Object.defineProperty(NeutronCreateFirewallGroupOption.prototype, "adminStateUp", {
        get: function () {
            return this['admin_state_up'];
        },
        set: function (adminStateUp) {
            this['admin_state_up'] = adminStateUp;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronCreateFirewallGroupOption;
}());
exports.NeutronCreateFirewallGroupOption = NeutronCreateFirewallGroupOption;
