// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronInsertFirewallRuleRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronInsertFirewallRuleRequestBody = void 0;
var NeutronInsertFirewallRuleRequestBody = /** @class */ (function () {
    function NeutronInsertFirewallRuleRequestBody(firewallRuleId) {
        this['firewall_rule_id'] = firewallRuleId;
    }
    NeutronInsertFirewallRuleRequestBody.prototype.withFirewallRuleId = function (firewallRuleId) {
        this['firewall_rule_id'] = firewallRuleId;
        return this;
    };
    Object.defineProperty(NeutronInsertFirewallRuleRequestBody.prototype, "firewallRuleId", {
        get: function () {
            return this['firewall_rule_id'];
        },
        set: function (firewallRuleId) {
            this['firewall_rule_id'] = firewallRuleId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronInsertFirewallRuleRequestBody.prototype.withInsertAfter = function (insertAfter) {
        this['insert_after'] = insertAfter;
        return this;
    };
    Object.defineProperty(NeutronInsertFirewallRuleRequestBody.prototype, "insertAfter", {
        get: function () {
            return this['insert_after'];
        },
        set: function (insertAfter) {
            this['insert_after'] = insertAfter;
        },
        enumerable: false,
        configurable: true
    });
    NeutronInsertFirewallRuleRequestBody.prototype.withInsertBefore = function (insertBefore) {
        this['insert_before'] = insertBefore;
        return this;
    };
    Object.defineProperty(NeutronInsertFirewallRuleRequestBody.prototype, "insertBefore", {
        get: function () {
            return this['insert_before'];
        },
        set: function (insertBefore) {
            this['insert_before'] = insertBefore;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronInsertFirewallRuleRequestBody;
}());
exports.NeutronInsertFirewallRuleRequestBody = NeutronInsertFirewallRuleRequestBody;
