// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronListFirewallPoliciesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronListFirewallPoliciesRequest = void 0;
var NeutronListFirewallPoliciesRequest = /** @class */ (function () {
    function NeutronListFirewallPoliciesRequest() {
    }
    NeutronListFirewallPoliciesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    NeutronListFirewallPoliciesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    NeutronListFirewallPoliciesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronListFirewallPoliciesRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronListFirewallPoliciesRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronListFirewallPoliciesRequest.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronListFirewallPoliciesRequest.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronListFirewallPoliciesRequest;
}());
exports.NeutronListFirewallPoliciesRequest = NeutronListFirewallPoliciesRequest;
