// @huaweicloud/huaweicloud-sdk-vpc/v2/model/NeutronFirewallPolicy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.NeutronFirewallPolicy = void 0;
var NeutronFirewallPolicy = /** @class */ (function () {
    function NeutronFirewallPolicy(audited, description, firewallRules, id, name, _public, tenantId, projectId) {
        this['audited'] = audited;
        this['description'] = description;
        this['firewall_rules'] = firewallRules;
        this['id'] = id;
        this['name'] = name;
        this['public'] = _public;
        this['tenant_id'] = tenantId;
        this['project_id'] = projectId;
    }
    NeutronFirewallPolicy.prototype.withAudited = function (audited) {
        this['audited'] = audited;
        return this;
    };
    NeutronFirewallPolicy.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    NeutronFirewallPolicy.prototype.withFirewallRules = function (firewallRules) {
        this['firewall_rules'] = firewallRules;
        return this;
    };
    Object.defineProperty(NeutronFirewallPolicy.prototype, "firewallRules", {
        get: function () {
            return this['firewall_rules'];
        },
        set: function (firewallRules) {
            this['firewall_rules'] = firewallRules;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallPolicy.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    NeutronFirewallPolicy.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    NeutronFirewallPolicy.prototype.withPublic = function (_public) {
        this['public'] = _public;
        return this;
    };
    Object.defineProperty(NeutronFirewallPolicy.prototype, "_public", {
        get: function () {
            return this['public'];
        },
        set: function (_public) {
            this['public'] = _public;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallPolicy.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(NeutronFirewallPolicy.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    NeutronFirewallPolicy.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(NeutronFirewallPolicy.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return NeutronFirewallPolicy;
}());
exports.NeutronFirewallPolicy = NeutronFirewallPolicy;
