// @huaweicloud/huaweicloud-sdk-vpc/v2/model/Privateip.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PrivateipDeviceOwnerEnum = exports.PrivateipStatusEnum = exports.Privateip = void 0;
var Privateip = /** @class */ (function () {
    function Privateip(status, id, subnetId, tenantId, deviceOwner, ipAddress) {
        this['status'] = status;
        this['id'] = id;
        this['subnet_id'] = subnetId;
        this['tenant_id'] = tenantId;
        this['device_owner'] = deviceOwner;
        this['ip_address'] = ipAddress;
    }
    Privateip.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    Privateip.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Privateip.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(Privateip.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    Privateip.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(Privateip.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    Privateip.prototype.withDeviceOwner = function (deviceOwner) {
        this['device_owner'] = deviceOwner;
        return this;
    };
    Object.defineProperty(Privateip.prototype, "deviceOwner", {
        get: function () {
            return this['device_owner'];
        },
        set: function (deviceOwner) {
            this['device_owner'] = deviceOwner;
        },
        enumerable: false,
        configurable: true
    });
    Privateip.prototype.withIpAddress = function (ipAddress) {
        this['ip_address'] = ipAddress;
        return this;
    };
    Object.defineProperty(Privateip.prototype, "ipAddress", {
        get: function () {
            return this['ip_address'];
        },
        set: function (ipAddress) {
            this['ip_address'] = ipAddress;
        },
        enumerable: false,
        configurable: true
    });
    return Privateip;
}());
exports.Privateip = Privateip;
/**
    * @export
    * @enum {string}
    */
var PrivateipStatusEnum;
(function (PrivateipStatusEnum) {
    PrivateipStatusEnum["ACTIVE"] = "ACTIVE";
    PrivateipStatusEnum["DOWN"] = "DOWN";
})(PrivateipStatusEnum = exports.PrivateipStatusEnum || (exports.PrivateipStatusEnum = {}));
/**
    * @export
    * @enum {string}
    */
var PrivateipDeviceOwnerEnum;
(function (PrivateipDeviceOwnerEnum) {
    PrivateipDeviceOwnerEnum["NETWORKDHCP"] = "network:dhcp";
    PrivateipDeviceOwnerEnum["NETWORKROUTER_INTERFACE_DISTRIBUTED"] = "network:router_interface_distributed";
    PrivateipDeviceOwnerEnum["COMPUTEXXX"] = "compute:xxx";
})(PrivateipDeviceOwnerEnum = exports.PrivateipDeviceOwnerEnum || (exports.PrivateipDeviceOwnerEnum = {}));
