// @huaweicloud/huaweicloud-sdk-vpc/v2/model/RejectVpcPeeringResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RejectVpcPeeringResponseStatusEnum = exports.RejectVpcPeeringResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var RejectVpcPeeringResponse = /** @class */ (function (_super) {
    __extends(RejectVpcPeeringResponse, _super);
    function RejectVpcPeeringResponse() {
        return _super.call(this) || this;
    }
    RejectVpcPeeringResponse.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    RejectVpcPeeringResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    RejectVpcPeeringResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    RejectVpcPeeringResponse.prototype.withRequestVpcInfo = function (requestVpcInfo) {
        this['request_vpc_info'] = requestVpcInfo;
        return this;
    };
    Object.defineProperty(RejectVpcPeeringResponse.prototype, "requestVpcInfo", {
        get: function () {
            return this['request_vpc_info'];
        },
        set: function (requestVpcInfo) {
            this['request_vpc_info'] = requestVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    RejectVpcPeeringResponse.prototype.withAcceptVpcInfo = function (acceptVpcInfo) {
        this['accept_vpc_info'] = acceptVpcInfo;
        return this;
    };
    Object.defineProperty(RejectVpcPeeringResponse.prototype, "acceptVpcInfo", {
        get: function () {
            return this['accept_vpc_info'];
        },
        set: function (acceptVpcInfo) {
            this['accept_vpc_info'] = acceptVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    RejectVpcPeeringResponse.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(RejectVpcPeeringResponse.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    RejectVpcPeeringResponse.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(RejectVpcPeeringResponse.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    RejectVpcPeeringResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return RejectVpcPeeringResponse;
}(SdkResponse_1.SdkResponse));
exports.RejectVpcPeeringResponse = RejectVpcPeeringResponse;
/**
    * @export
    * @enum {string}
    */
var RejectVpcPeeringResponseStatusEnum;
(function (RejectVpcPeeringResponseStatusEnum) {
    RejectVpcPeeringResponseStatusEnum["PENDING_ACCEPTANCE"] = "PENDING_ACCEPTANCE";
    RejectVpcPeeringResponseStatusEnum["REJECTED"] = "REJECTED";
    RejectVpcPeeringResponseStatusEnum["EXPIRED"] = "EXPIRED";
    RejectVpcPeeringResponseStatusEnum["DELETED"] = "DELETED";
    RejectVpcPeeringResponseStatusEnum["ACTIVE"] = "ACTIVE";
})(RejectVpcPeeringResponseStatusEnum = exports.RejectVpcPeeringResponseStatusEnum || (exports.RejectVpcPeeringResponseStatusEnum = {}));
