// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ResourceResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResourceResultTypeEnum = exports.ResourceResult = void 0;
var ResourceResult = /** @class */ (function () {
    function ResourceResult(type, used, quota, min) {
        this['type'] = type;
        this['used'] = used;
        this['quota'] = quota;
        this['min'] = min;
    }
    ResourceResult.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    ResourceResult.prototype.withUsed = function (used) {
        this['used'] = used;
        return this;
    };
    ResourceResult.prototype.withQuota = function (quota) {
        this['quota'] = quota;
        return this;
    };
    ResourceResult.prototype.withMin = function (min) {
        this['min'] = min;
        return this;
    };
    return ResourceResult;
}());
exports.ResourceResult = ResourceResult;
/**
    * @export
    * @enum {string}
    */
var ResourceResultTypeEnum;
(function (ResourceResultTypeEnum) {
    ResourceResultTypeEnum["VPC"] = "vpc";
    ResourceResultTypeEnum["SUBNET"] = "subnet";
    ResourceResultTypeEnum["SECURITYGROUP"] = "securityGroup";
    ResourceResultTypeEnum["SECURITYGROUPRULE"] = "securityGroupRule";
    ResourceResultTypeEnum["PUBLICIP"] = "publicIp";
    ResourceResultTypeEnum["VPN"] = "vpn";
    ResourceResultTypeEnum["VPNGW"] = "vpngw";
    ResourceResultTypeEnum["VPCPEER"] = "vpcPeer";
    ResourceResultTypeEnum["FIREWALL"] = "firewall";
    ResourceResultTypeEnum["SHAREBANDWIDTH"] = "shareBandwidth";
    ResourceResultTypeEnum["SHAREBANDWIDTHIP"] = "shareBandwidthIP";
    ResourceResultTypeEnum["LOADBALANCER"] = "loadbalancer";
    ResourceResultTypeEnum["LISTENER"] = "listener";
    ResourceResultTypeEnum["PHYSICALCONNECT"] = "physicalConnect";
    ResourceResultTypeEnum["VIRTUALINTERFACE"] = "virtualInterface";
    ResourceResultTypeEnum["VPCCONTAINROUTETABLE"] = "vpcContainRoutetable";
    ResourceResultTypeEnum["ROUTETABLECONTAINROUTES"] = "routetableContainRoutes";
})(ResourceResultTypeEnum = exports.ResourceResultTypeEnum || (exports.ResourceResultTypeEnum = {}));
