// @huaweicloud/huaweicloud-sdk-vpc/v2/model/RouteTableResp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteTableResp = void 0;
var RouteTableResp = /** @class */ (function () {
    function RouteTableResp(id, name, _default, routes, subnets, tenantId, vpcId, description, createdAt, updatedAt) {
        this['id'] = id;
        this['name'] = name;
        this['default'] = _default;
        this['routes'] = routes;
        this['subnets'] = subnets;
        this['tenant_id'] = tenantId;
        this['vpc_id'] = vpcId;
        this['description'] = description;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    RouteTableResp.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    RouteTableResp.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    RouteTableResp.prototype.withDefault = function (_default) {
        this['default'] = _default;
        return this;
    };
    Object.defineProperty(RouteTableResp.prototype, "_default", {
        get: function () {
            return this['default'];
        },
        set: function (_default) {
            this['default'] = _default;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableResp.prototype.withRoutes = function (routes) {
        this['routes'] = routes;
        return this;
    };
    RouteTableResp.prototype.withSubnets = function (subnets) {
        this['subnets'] = subnets;
        return this;
    };
    RouteTableResp.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(RouteTableResp.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableResp.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(RouteTableResp.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableResp.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    RouteTableResp.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(RouteTableResp.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableResp.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(RouteTableResp.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return RouteTableResp;
}());
exports.RouteTableResp = RouteTableResp;
