// @huaweicloud/huaweicloud-sdk-vpc/v2/model/RouteTableListResp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RouteTableListResp = void 0;
var RouteTableListResp = /** @class */ (function () {
    function RouteTableListResp(id, name, _default, subnets, tenantId, vpcId, description, createdAt, updatedAt) {
        this['id'] = id;
        this['name'] = name;
        this['default'] = _default;
        this['subnets'] = subnets;
        this['tenant_id'] = tenantId;
        this['vpc_id'] = vpcId;
        this['description'] = description;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    RouteTableListResp.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    RouteTableListResp.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    RouteTableListResp.prototype.withDefault = function (_default) {
        this['default'] = _default;
        return this;
    };
    Object.defineProperty(RouteTableListResp.prototype, "_default", {
        get: function () {
            return this['default'];
        },
        set: function (_default) {
            this['default'] = _default;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableListResp.prototype.withSubnets = function (subnets) {
        this['subnets'] = subnets;
        return this;
    };
    RouteTableListResp.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(RouteTableListResp.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableListResp.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(RouteTableListResp.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableListResp.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    RouteTableListResp.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(RouteTableListResp.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    RouteTableListResp.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(RouteTableListResp.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return RouteTableListResp;
}());
exports.RouteTableListResp = RouteTableListResp;
