// @huaweicloud/huaweicloud-sdk-vpc/v2/model/SecurityGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroup = void 0;
var SecurityGroup = /** @class */ (function () {
    function SecurityGroup(name, id, securityGroupRules) {
        this['name'] = name;
        this['id'] = id;
        this['security_group_rules'] = securityGroupRules;
    }
    SecurityGroup.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    SecurityGroup.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    SecurityGroup.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    SecurityGroup.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(SecurityGroup.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroup.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(SecurityGroup.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroup.prototype.withSecurityGroupRules = function (securityGroupRules) {
        this['security_group_rules'] = securityGroupRules;
        return this;
    };
    Object.defineProperty(SecurityGroup.prototype, "securityGroupRules", {
        get: function () {
            return this['security_group_rules'];
        },
        set: function (securityGroupRules) {
            this['security_group_rules'] = securityGroupRules;
        },
        enumerable: false,
        configurable: true
    });
    return SecurityGroup;
}());
exports.SecurityGroup = SecurityGroup;
