// @huaweicloud/huaweicloud-sdk-vpc/v2/model/SecurityGroupRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroupRule = void 0;
var SecurityGroupRule = /** @class */ (function () {
    function SecurityGroupRule(id, description, securityGroupId, direction, ethertype, protocol, portRangeMin, portRangeMax, remoteIpPrefix, remoteGroupId, tenantId) {
        this['id'] = id;
        this['description'] = description;
        this['security_group_id'] = securityGroupId;
        this['direction'] = direction;
        this['ethertype'] = ethertype;
        this['protocol'] = protocol;
        this['port_range_min'] = portRangeMin;
        this['port_range_max'] = portRangeMax;
        this['remote_ip_prefix'] = remoteIpPrefix;
        this['remote_group_id'] = remoteGroupId;
        this['tenant_id'] = tenantId;
    }
    SecurityGroupRule.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    SecurityGroupRule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    SecurityGroupRule.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    SecurityGroupRule.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    SecurityGroupRule.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    SecurityGroupRule.prototype.withPortRangeMin = function (portRangeMin) {
        this['port_range_min'] = portRangeMin;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "portRangeMin", {
        get: function () {
            return this['port_range_min'];
        },
        set: function (portRangeMin) {
            this['port_range_min'] = portRangeMin;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withPortRangeMax = function (portRangeMax) {
        this['port_range_max'] = portRangeMax;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "portRangeMax", {
        get: function () {
            return this['port_range_max'];
        },
        set: function (portRangeMax) {
            this['port_range_max'] = portRangeMax;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withRemoteAddressGroupId = function (remoteAddressGroupId) {
        this['remote_address_group_id'] = remoteAddressGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteAddressGroupId", {
        get: function () {
            return this['remote_address_group_id'];
        },
        set: function (remoteAddressGroupId) {
            this['remote_address_group_id'] = remoteAddressGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return SecurityGroupRule;
}());
exports.SecurityGroupRule = SecurityGroupRule;
