// @huaweicloud/huaweicloud-sdk-vpc/v2/model/ShowQuotaRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowQuotaRequestTypeEnum = exports.ShowQuotaRequest = void 0;
var ShowQuotaRequest = /** @class */ (function () {
    function ShowQuotaRequest() {
    }
    ShowQuotaRequest.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    return ShowQuotaRequest;
}());
exports.ShowQuotaRequest = ShowQuotaRequest;
/**
    * @export
    * @enum {string}
    */
var ShowQuotaRequestTypeEnum;
(function (ShowQuotaRequestTypeEnum) {
    ShowQuotaRequestTypeEnum["VPC"] = "vpc";
    ShowQuotaRequestTypeEnum["SUBNET"] = "subnet";
    ShowQuotaRequestTypeEnum["SECURITYGROUP"] = "securityGroup";
    ShowQuotaRequestTypeEnum["SECURITYGROUPRULE"] = "securityGroupRule";
    ShowQuotaRequestTypeEnum["PUBLICIP"] = "publicIp";
    ShowQuotaRequestTypeEnum["VPN"] = "vpn";
    ShowQuotaRequestTypeEnum["VPNGW"] = "vpngw";
    ShowQuotaRequestTypeEnum["VPCPEER"] = "vpcPeer";
    ShowQuotaRequestTypeEnum["FIREWALL"] = "firewall";
    ShowQuotaRequestTypeEnum["SHAREBANDWIDTH"] = "shareBandwidth";
    ShowQuotaRequestTypeEnum["SHAREBANDWIDTHIP"] = "shareBandwidthIP";
    ShowQuotaRequestTypeEnum["LOADBALANCER"] = "loadbalancer";
    ShowQuotaRequestTypeEnum["LISTENER"] = "listener";
    ShowQuotaRequestTypeEnum["PHYSICALCONNECT"] = "physicalConnect";
    ShowQuotaRequestTypeEnum["VIRTUALINTERFACE"] = "virtualInterface";
    ShowQuotaRequestTypeEnum["VPCCONTAINROUTETABLE"] = "vpcContainRoutetable";
    ShowQuotaRequestTypeEnum["ROUTETABLECONTAINROUTES"] = "routetableContainRoutes";
})(ShowQuotaRequestTypeEnum = exports.ShowQuotaRequestTypeEnum || (exports.ShowQuotaRequestTypeEnum = {}));
