// @huaweicloud/huaweicloud-sdk-vpc/v2/model/UpdateSubnetRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSubnetRequest = void 0;
var UpdateSubnetRequest = /** @class */ (function () {
    function UpdateSubnetRequest(vpcId, subnetId) {
        this['vpc_id'] = vpcId;
        this['subnet_id'] = subnetId;
    }
    UpdateSubnetRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(UpdateSubnetRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetRequest.prototype.withSubnetId = function (subnetId) {
        this['subnet_id'] = subnetId;
        return this;
    };
    Object.defineProperty(UpdateSubnetRequest.prototype, "subnetId", {
        get: function () {
            return this['subnet_id'];
        },
        set: function (subnetId) {
            this['subnet_id'] = subnetId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateSubnetRequest;
}());
exports.UpdateSubnetRequest = UpdateSubnetRequest;
