// @huaweicloud/huaweicloud-sdk-vpc/v2/model/UpdateSubnetOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSubnetOption = void 0;
var UpdateSubnetOption = /** @class */ (function () {
    function UpdateSubnetOption(name) {
        this['name'] = name;
    }
    UpdateSubnetOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateSubnetOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateSubnetOption.prototype.withIpv6Enable = function (ipv6Enable) {
        this['ipv6_enable'] = ipv6Enable;
        return this;
    };
    Object.defineProperty(UpdateSubnetOption.prototype, "ipv6Enable", {
        get: function () {
            return this['ipv6_enable'];
        },
        set: function (ipv6Enable) {
            this['ipv6_enable'] = ipv6Enable;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetOption.prototype.withDhcpEnable = function (dhcpEnable) {
        this['dhcp_enable'] = dhcpEnable;
        return this;
    };
    Object.defineProperty(UpdateSubnetOption.prototype, "dhcpEnable", {
        get: function () {
            return this['dhcp_enable'];
        },
        set: function (dhcpEnable) {
            this['dhcp_enable'] = dhcpEnable;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetOption.prototype.withPrimaryDns = function (primaryDns) {
        this['primary_dns'] = primaryDns;
        return this;
    };
    Object.defineProperty(UpdateSubnetOption.prototype, "primaryDns", {
        get: function () {
            return this['primary_dns'];
        },
        set: function (primaryDns) {
            this['primary_dns'] = primaryDns;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetOption.prototype.withSecondaryDns = function (secondaryDns) {
        this['secondary_dns'] = secondaryDns;
        return this;
    };
    Object.defineProperty(UpdateSubnetOption.prototype, "secondaryDns", {
        get: function () {
            return this['secondary_dns'];
        },
        set: function (secondaryDns) {
            this['secondary_dns'] = secondaryDns;
        },
        enumerable: false,
        configurable: true
    });
    UpdateSubnetOption.prototype.withDnsList = function (dnsList) {
        this['dnsList'] = dnsList;
        return this;
    };
    UpdateSubnetOption.prototype.withExtraDhcpOpts = function (extraDhcpOpts) {
        this['extra_dhcp_opts'] = extraDhcpOpts;
        return this;
    };
    Object.defineProperty(UpdateSubnetOption.prototype, "extraDhcpOpts", {
        get: function () {
            return this['extra_dhcp_opts'];
        },
        set: function (extraDhcpOpts) {
            this['extra_dhcp_opts'] = extraDhcpOpts;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateSubnetOption;
}());
exports.UpdateSubnetOption = UpdateSubnetOption;
