// @huaweicloud/huaweicloud-sdk-vpc/v2/model/UpdatePortOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePortOption = void 0;
var UpdatePortOption = /** @class */ (function () {
    function UpdatePortOption() {
    }
    UpdatePortOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdatePortOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(UpdatePortOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    UpdatePortOption.prototype.withAllowedAddressPairs = function (allowedAddressPairs) {
        this['allowed_address_pairs'] = allowedAddressPairs;
        return this;
    };
    Object.defineProperty(UpdatePortOption.prototype, "allowedAddressPairs", {
        get: function () {
            return this['allowed_address_pairs'];
        },
        set: function (allowedAddressPairs) {
            this['allowed_address_pairs'] = allowedAddressPairs;
        },
        enumerable: false,
        configurable: true
    });
    UpdatePortOption.prototype.withExtraDhcpOpts = function (extraDhcpOpts) {
        this['extra_dhcp_opts'] = extraDhcpOpts;
        return this;
    };
    Object.defineProperty(UpdatePortOption.prototype, "extraDhcpOpts", {
        get: function () {
            return this['extra_dhcp_opts'];
        },
        set: function (extraDhcpOpts) {
            this['extra_dhcp_opts'] = extraDhcpOpts;
        },
        enumerable: false,
        configurable: true
    });
    return UpdatePortOption;
}());
exports.UpdatePortOption = UpdatePortOption;
