// @huaweicloud/huaweicloud-sdk-vpc/v2/model/Vpc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcStatusEnum = exports.Vpc = void 0;
var Vpc = /** @class */ (function () {
    function Vpc(id, name, cidr, description, routes, status, enterpriseProjectId, tenantId, createdAt, updatedAt) {
        this['id'] = id;
        this['name'] = name;
        this['cidr'] = cidr;
        this['description'] = description;
        this['routes'] = routes;
        this['status'] = status;
        this['enterprise_project_id'] = enterpriseProjectId;
        this['tenant_id'] = tenantId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
    }
    Vpc.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Vpc.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Vpc.prototype.withCidr = function (cidr) {
        this['cidr'] = cidr;
        return this;
    };
    Vpc.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    Vpc.prototype.withRoutes = function (routes) {
        this['routes'] = routes;
        return this;
    };
    Vpc.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    Vpc.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    return Vpc;
}());
exports.Vpc = Vpc;
/**
    * @export
    * @enum {string}
    */
var VpcStatusEnum;
(function (VpcStatusEnum) {
    VpcStatusEnum["CREATING"] = "CREATING";
    VpcStatusEnum["OK"] = "OK";
    VpcStatusEnum["ERROR"] = "ERROR";
})(VpcStatusEnum = exports.VpcStatusEnum || (exports.VpcStatusEnum = {}));
