// @huaweicloud/huaweicloud-sdk-vpc/v2/model/VpcPeering.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcPeeringStatusEnum = exports.VpcPeering = void 0;
var VpcPeering = /** @class */ (function () {
    function VpcPeering(id, name, status, requestVpcInfo, acceptVpcInfo, createdAt, updatedAt, description) {
        this['id'] = id;
        this['name'] = name;
        this['status'] = status;
        this['request_vpc_info'] = requestVpcInfo;
        this['accept_vpc_info'] = acceptVpcInfo;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['description'] = description;
    }
    VpcPeering.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    VpcPeering.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    VpcPeering.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    VpcPeering.prototype.withRequestVpcInfo = function (requestVpcInfo) {
        this['request_vpc_info'] = requestVpcInfo;
        return this;
    };
    Object.defineProperty(VpcPeering.prototype, "requestVpcInfo", {
        get: function () {
            return this['request_vpc_info'];
        },
        set: function (requestVpcInfo) {
            this['request_vpc_info'] = requestVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    VpcPeering.prototype.withAcceptVpcInfo = function (acceptVpcInfo) {
        this['accept_vpc_info'] = acceptVpcInfo;
        return this;
    };
    Object.defineProperty(VpcPeering.prototype, "acceptVpcInfo", {
        get: function () {
            return this['accept_vpc_info'];
        },
        set: function (acceptVpcInfo) {
            this['accept_vpc_info'] = acceptVpcInfo;
        },
        enumerable: false,
        configurable: true
    });
    VpcPeering.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(VpcPeering.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    VpcPeering.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(VpcPeering.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    VpcPeering.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return VpcPeering;
}());
exports.VpcPeering = VpcPeering;
/**
    * @export
    * @enum {string}
    */
var VpcPeeringStatusEnum;
(function (VpcPeeringStatusEnum) {
    VpcPeeringStatusEnum["PENDING_ACCEPTANCE"] = "PENDING_ACCEPTANCE";
    VpcPeeringStatusEnum["REJECTED"] = "REJECTED";
    VpcPeeringStatusEnum["EXPIRED"] = "EXPIRED";
    VpcPeeringStatusEnum["DELETED"] = "DELETED";
    VpcPeeringStatusEnum["ACTIVE"] = "ACTIVE";
})(VpcPeeringStatusEnum = exports.VpcPeeringStatusEnum || (exports.VpcPeeringStatusEnum = {}));
