// @huaweicloud/huaweicloud-sdk-vpc/v2/model/VpcRoute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VpcRouteTypeEnum = exports.VpcRoute = void 0;
var VpcRoute = /** @class */ (function () {
    function VpcRoute(id, destination, nexthop, type, vpcId, tenantId) {
        this['id'] = id;
        this['destination'] = destination;
        this['nexthop'] = nexthop;
        this['type'] = type;
        this['vpc_id'] = vpcId;
        this['tenant_id'] = tenantId;
    }
    VpcRoute.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    VpcRoute.prototype.withDestination = function (destination) {
        this['destination'] = destination;
        return this;
    };
    VpcRoute.prototype.withNexthop = function (nexthop) {
        this['nexthop'] = nexthop;
        return this;
    };
    VpcRoute.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    VpcRoute.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(VpcRoute.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    VpcRoute.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(VpcRoute.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return VpcRoute;
}());
exports.VpcRoute = VpcRoute;
/**
    * @export
    * @enum {string}
    */
var VpcRouteTypeEnum;
(function (VpcRouteTypeEnum) {
    VpcRouteTypeEnum["PEERING"] = "peering";
})(VpcRouteTypeEnum = exports.VpcRouteTypeEnum || (exports.VpcRouteTypeEnum = {}));
