// @huaweicloud/huaweicloud-sdk-vpc/v3/VpcClient.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequiredError = exports.ParamCreater = exports.VpcClient = void 0;
var ClientBuilder_1 = require("@huaweicloud/huaweicloud-sdk-core/ClientBuilder");
var AddVpcExtendCidrRequest_1 = require("./model/AddVpcExtendCidrRequest");
var BatchCreateSubNetworkInterfaceRequest_1 = require("./model/BatchCreateSubNetworkInterfaceRequest");
var CreateAddressGroupRequest_1 = require("./model/CreateAddressGroupRequest");
var CreateSecurityGroupRequest_1 = require("./model/CreateSecurityGroupRequest");
var CreateSecurityGroupRuleRequest_1 = require("./model/CreateSecurityGroupRuleRequest");
var CreateSubNetworkInterfaceRequest_1 = require("./model/CreateSubNetworkInterfaceRequest");
var CreateVpcRequest_1 = require("./model/CreateVpcRequest");
var DeleteAddressGroupRequest_1 = require("./model/DeleteAddressGroupRequest");
var DeleteIpAddressGroupForceRequest_1 = require("./model/DeleteIpAddressGroupForceRequest");
var DeleteSecurityGroupRequest_1 = require("./model/DeleteSecurityGroupRequest");
var DeleteSecurityGroupRuleRequest_1 = require("./model/DeleteSecurityGroupRuleRequest");
var DeleteSubNetworkInterfaceRequest_1 = require("./model/DeleteSubNetworkInterfaceRequest");
var DeleteVpcRequest_1 = require("./model/DeleteVpcRequest");
var ListAddressGroupRequest_1 = require("./model/ListAddressGroupRequest");
var ListSecurityGroupRulesRequest_1 = require("./model/ListSecurityGroupRulesRequest");
var ListSecurityGroupsRequest_1 = require("./model/ListSecurityGroupsRequest");
var ListSubNetworkInterfacesRequest_1 = require("./model/ListSubNetworkInterfacesRequest");
var ListVpcsRequest_1 = require("./model/ListVpcsRequest");
var MigrateSubNetworkInterfaceRequest_1 = require("./model/MigrateSubNetworkInterfaceRequest");
var RemoveVpcExtendCidrRequest_1 = require("./model/RemoveVpcExtendCidrRequest");
var ShowAddressGroupRequest_1 = require("./model/ShowAddressGroupRequest");
var ShowSecurityGroupRequest_1 = require("./model/ShowSecurityGroupRequest");
var ShowSecurityGroupRuleRequest_1 = require("./model/ShowSecurityGroupRuleRequest");
var ShowSubNetworkInterfaceRequest_1 = require("./model/ShowSubNetworkInterfaceRequest");
var ShowVpcRequest_1 = require("./model/ShowVpcRequest");
var UpdateAddressGroupRequest_1 = require("./model/UpdateAddressGroupRequest");
var UpdateSecurityGroupRequest_1 = require("./model/UpdateSecurityGroupRequest");
var UpdateSubNetworkInterfaceRequest_1 = require("./model/UpdateSubNetworkInterfaceRequest");
var UpdateVpcRequest_1 = require("./model/UpdateVpcRequest");
var VpcClient = /** @class */ (function () {
    function VpcClient(client) {
        this.hcClient = client;
    }
    VpcClient.newBuilder = function () {
        return new ClientBuilder_1.ClientBuilder(newClient);
    };
    VpcClient.prototype.getPath = function () {
        return __dirname;
    };
    /**
     * 批量创建辅助弹性网卡
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 批量创建辅助弹性网卡
     * @param {BatchCreateSubNetworkInterfaceRequestBody} batchCreateSubNetworkInterfaceRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.batchCreateSubNetworkInterface = function (batchCreateSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().batchCreateSubNetworkInterface(batchCreateSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 创建安全组
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 创建安全组
     * @param {CreateSecurityGroupRequestBody} createSecurityGroupRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.createSecurityGroup = function (createSecurityGroupRequest) {
        var options = (0, exports.ParamCreater)().createSecurityGroup(createSecurityGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 创建安全组规则
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 创建安全组规则
     * @param {CreateSecurityGroupRuleRequestBody} createSecurityGroupRuleRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.createSecurityGroupRule = function (createSecurityGroupRuleRequest) {
        var options = (0, exports.ParamCreater)().createSecurityGroupRule(createSecurityGroupRuleRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 创建辅助弹性网卡
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 创建辅助弹性网卡
     * @param {CreateSubNetworkInterfaceRequestBody} createSubNetworkInterfaceRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.createSubNetworkInterface = function (createSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().createSubNetworkInterface(createSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 删除安全组
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 删除安全组
     * @param {string} securityGroupId 安全组资源ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteSecurityGroup = function (deleteSecurityGroupRequest) {
        var options = (0, exports.ParamCreater)().deleteSecurityGroup(deleteSecurityGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 删除安全组规则
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 删除安全组规则
     * @param {string} securityGroupRuleId 安全组规则ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteSecurityGroupRule = function (deleteSecurityGroupRuleRequest) {
        var options = (0, exports.ParamCreater)().deleteSecurityGroupRule(deleteSecurityGroupRuleRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 删除辅助弹性网卡
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 删除辅助弹性网卡
     * @param {string} subNetworkInterfaceId 弹性辅助网卡唯一标识
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteSubNetworkInterface = function (deleteSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().deleteSubNetworkInterface(deleteSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询安全组规则列表
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询安全组规则列表
     * @param {number} [limit] 功能说明：每页返回个数 取值范围：0-2000
     * @param {string} [marker] 分页查询起始的资源ID，为空时查询第一页
     * @param {Array<string>} [id] 功能说明：安全组规则ID，支持多个ID过滤
     * @param {Array<string>} [securityGroupId] 功能说明：安全组规则所属安全组ID，支持多个ID过滤
     * @param {Array<string>} [protocol] 功能说明：安全组规则协议，支持多条过滤
     * @param {Array<string>} [description] 功能说明：安全组规则的描述，支持多个描述同时过滤
     * @param {Array<string>} [remoteGroupId] 功能说明：远端安全组ID，支持多ID过滤
     * @param {string} [direction] 功能说明：安全组规则方向
     * @param {string} [action] 功能说明：安全组规则生效策略
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.listSecurityGroupRules = function (listSecurityGroupRulesRequest) {
        var options = (0, exports.ParamCreater)().listSecurityGroupRules(listSecurityGroupRulesRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询某租户下的安全组列表
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询安全组列表
     * @param {number} [limit] 功能说明：每页返回的个数 取值范围：0-2000
     * @param {string} [marker] 分页查询起始的资源ID，为空时查询第一页
     * @param {Array<string>} [id] 功能说明：安全组资源ID。可以使用该字段精确过滤安全组，支持多个ID
     * @param {Array<string>} [name] 功能说明：安全组名称。可以使用该字段精确过滤满足条件的安全组，支持传入多个name过滤
     * @param {Array<string>} [description] 功能说明：安全组描述新增。可以使用该字段精确过滤安全组，支持传入多个描述进行过滤
     * @param {string} [enterpriseProjectId] 功能说明：企业项目ID。可以使用该字段过滤某个企业项目下的安全组。 取值范围：最大长度36字节，带“-”连字符的UUID格式，或者是字符串“0”。“0”表示默认企业项目。 约束：若需要查询当前用户所有有权限查看企业项目绑定的安全组，请传参all_granted_eps。
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.listSecurityGroups = function (listSecurityGroupsRequest) {
        var options = (0, exports.ParamCreater)().listSecurityGroups(listSecurityGroupsRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询辅助弹性网卡列表，单次查询最多返回2000条数据
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询租户下辅助弹性网卡列表
     * @param {number} [limit] 功能说明：每页返回的个数 取值范围：0-2000
     * @param {string} [marker] 分页查询起始的资源ID，为空时查询第一页
     * @param {Array<string>} [id] 功能说明：辅助弹性网卡ID，支持多ID过滤 使用场景：查询需要的多个辅助弹性网卡信息
     * @param {Array<string>} [virsubnetId] 功能说明：辅助弹性网卡所属虚拟子网的ID，支持多个ID过滤 使用场景：过滤需要的单个或者多个虚拟子网下的辅助弹性网卡
     * @param {Array<string>} [privateIpAddress] 功能说明：辅助弹性网卡的私有IPv4地址，支持多个地址同时过滤 使用场景：通过单个或者多个ip地址过滤查询辅助弹性网卡
     * @param {Array<string>} [macAddress] 功能说明：辅助弹性网卡的mac地址，支持多个同时过滤 使用场景：使用mac地址精确过滤辅助弹性网卡
     * @param {Array<string>} [vpcId] 功能说明：辅助弹性网卡所属的VPC_ID，支持多ID过滤 使用场景：过滤单个或多个VPC下的辅助弹性网卡信息
     * @param {Array<string>} [description] 功能说明：辅助弹性网卡的描述信息，支持多个同时过滤 使用场景：通过描述信息过滤辅助弹性网卡
     * @param {Array<string>} [parentId] 功能说明：辅助弹性网卡的宿主网卡的ID，支持多ID过滤 使用场景：过滤单个或多个宿主网卡下存在的辅助弹性网卡
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.listSubNetworkInterfaces = function (listSubNetworkInterfacesRequest) {
        var options = (0, exports.ParamCreater)().listSubNetworkInterfaces(listSubNetworkInterfacesRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 批量迁移辅助弹性网卡
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 迁移辅助弹性网卡
     * @param {MigrateSubNetworkInterfaceRequestBody} migrateSubNetworkInterfaceRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.migrateSubNetworkInterface = function (migrateSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().migrateSubNetworkInterface(migrateSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询单个安全组详情
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询安全组
     * @param {string} securityGroupId 安全组资源ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showSecurityGroup = function (showSecurityGroupRequest) {
        var options = (0, exports.ParamCreater)().showSecurityGroup(showSecurityGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询单个安全组规则
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询安全组规则
     * @param {string} securityGroupRuleId 安全组规则ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showSecurityGroupRule = function (showSecurityGroupRuleRequest) {
        var options = (0, exports.ParamCreater)().showSecurityGroupRule(showSecurityGroupRuleRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询辅助弹性网卡详情
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询租户下辅助弹性网卡
     * @param {string} subNetworkInterfaceId 辅助弹性网卡的唯一标识
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showSubNetworkInterface = function (showSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().showSubNetworkInterface(showSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询辅助弹性网卡数目
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询租户下辅助弹性网卡数目
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showSubNetworkInterfacesQuantity = function () {
        var options = (0, exports.ParamCreater)().showSubNetworkInterfacesQuantity();
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 更新安全组
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 更新安全组
     * @param {string} securityGroupId 安全组ID
     * @param {UpdateSecurityGroupRequestBody} updateSecurityGroupRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.updateSecurityGroup = function (updateSecurityGroupRequest) {
        var options = (0, exports.ParamCreater)().updateSecurityGroup(updateSecurityGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 更新辅助弹性网卡
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 更新辅助弹性网卡
     * @param {string} subNetworkInterfaceId 辅助弹性网卡的唯一标识
     * @param {UpdateSubNetworkInterfaceRequestBody} updateSubNetworkInterfaceRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.updateSubNetworkInterface = function (updateSubNetworkInterfaceRequest) {
        var options = (0, exports.ParamCreater)().updateSubNetworkInterface(updateSubNetworkInterfaceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 创建地址组
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 创建地址组
     * @param {CreateAddressGroupRequestBody} createAddressGroupRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.createAddressGroup = function (createAddressGroupRequest) {
        var options = (0, exports.ParamCreater)().createAddressGroup(createAddressGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 删除地址组，非强制删除，删除前请确保未被其他资源引用
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 删除地址组
     * @param {string} addressGroupId 地址组的唯一标识
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteAddressGroup = function (deleteAddressGroupRequest) {
        var options = (0, exports.ParamCreater)().deleteAddressGroup(deleteAddressGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 强制删除地址组，删除的地址组与安全组规则关联时，会删除地址组与关联的安全组规则。
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 强制删除地址组
     * @param {string} addressGroupId IP地址组的唯一标识，要删除的IP地址组ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteIpAddressGroupForce = function (deleteIpAddressGroupForceRequest) {
        var options = (0, exports.ParamCreater)().deleteIpAddressGroupForce(deleteIpAddressGroupForceRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询地址组列表，根据过滤条件进行过滤。
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询地址组列表
     * @param {number} [limit] 功能说明：每页返回的个数 取值范围：0~2000
     * @param {string} [marker] 分页查询起始的资源ID，为空时查询第一页
     * @param {Array<string>} [id] 地址组唯一标识，填写后接口按照id进行过滤，支持多ID同时过滤
     * @param {Array<string>} [name] 地址组名称，填写后按照名称进行过滤，支持多名称同时过滤
     * @param {number} [ipVersion] IP地址组ip版本，当前只支持ipv4，填写后按照ip版本进行过滤
     * @param {Array<string>} [description] 地址组描述信息，填写后按照地址组描述信息过滤，支持多描述同时过滤
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.listAddressGroup = function (listAddressGroupRequest) {
        var options = (0, exports.ParamCreater)().listAddressGroup(listAddressGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询地址组详情。
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询地址组
     * @param {string} addressGroupId 地址组的唯一标识
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showAddressGroup = function (showAddressGroupRequest) {
        var options = (0, exports.ParamCreater)().showAddressGroup(showAddressGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 更新地址组。
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 更新地址组
     * @param {string} addressGroupId 地址组的唯一标识
     * @param {UpdateAddressGroupRequestBody} updateAddressGroupRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.updateAddressGroup = function (updateAddressGroupRequest) {
        var options = (0, exports.ParamCreater)().updateAddressGroup(updateAddressGroupRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 添加VPC的扩展网段
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 添加VPC扩展网段
     * @param {string} vpcId VPC资源ID
     * @param {AddVpcExtendCidrRequestBody} addVpcExtendCidrRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.addVpcExtendCidr = function (addVpcExtendCidrRequest) {
        var options = (0, exports.ParamCreater)().addVpcExtendCidr(addVpcExtendCidrRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 创建虚拟私有云
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 创建VPC
     * @param {CreateVpcRequestBody} createVpcRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.createVpc = function (createVpcRequest) {
        var options = (0, exports.ParamCreater)().createVpc(createVpcRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 删除VPC
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 删除VPC
     * @param {string} vpcId VPC的资源ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.deleteVpc = function (deleteVpcRequest) {
        var options = (0, exports.ParamCreater)().deleteVpc(deleteVpcRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询vpc列表
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询VPC列表
     * @param {number} [limit] 功能说明：每页返回的个数 取值范围：0-2000
     * @param {string} [marker] 分页查询起始的资源ID，为空时查询第一页
     * @param {Array<string>} [id] VPC资源ID。可以使用该字段过滤VPC
     * @param {Array<string>} [name] VPC的name信息，可以使用该字段过滤VPC
     * @param {Array<string>} [description] VPC的描述信息。可以使用该字段过滤VPC
     * @param {Array<string>} [cidr] VPC的CIDR。可以使用该字段过滤VPC
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.listVpcs = function (listVpcsRequest) {
        var options = (0, exports.ParamCreater)().listVpcs(listVpcsRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 移除VPC扩展网段
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 移除VPC扩展网段
     * @param {string} vpcId VPC资源ID
     * @param {RemoveVpcExtendCidrRequestBody} removeVpcExtendCidrRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.removeVpcExtendCidr = function (removeVpcExtendCidrRequest) {
        var options = (0, exports.ParamCreater)().removeVpcExtendCidr(removeVpcExtendCidrRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 查询vpc详情
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 查询VPC详情
     * @param {string} vpcId VPC资源ID
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.showVpc = function (showVpcRequest) {
        var options = (0, exports.ParamCreater)().showVpc(showVpcRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    /**
     * 更新vpc
     *
     * Please refer to HUAWEI cloud API Explorer for details.
     *
     * @summary 更新VPC
     * @param {string} vpcId VPC资源ID
     * @param {UpdateVpcRequestBody} updateVpcRequestBody This is a auto create Body Object
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     */
    VpcClient.prototype.updateVpc = function (updateVpcRequest) {
        var options = (0, exports.ParamCreater)().updateVpc(updateVpcRequest);
        // @ts-ignore
        options['responseHeaders'] = [''];
        return this.hcClient.sendRequest(options);
    };
    return VpcClient;
}());
exports.VpcClient = VpcClient;
var ParamCreater = function () {
    return {
        /**
         * 批量创建辅助弹性网卡
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        batchCreateSubNetworkInterface: function (batchCreateSubNetworkInterfaceRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/batch-create",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (batchCreateSubNetworkInterfaceRequest !== null && batchCreateSubNetworkInterfaceRequest !== undefined) {
                if (batchCreateSubNetworkInterfaceRequest instanceof BatchCreateSubNetworkInterfaceRequest_1.BatchCreateSubNetworkInterfaceRequest) {
                    body = batchCreateSubNetworkInterfaceRequest.body;
                }
                else {
                    body = batchCreateSubNetworkInterfaceRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 创建安全组
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        createSecurityGroup: function (createSecurityGroupRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/security-groups",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (createSecurityGroupRequest !== null && createSecurityGroupRequest !== undefined) {
                if (createSecurityGroupRequest instanceof CreateSecurityGroupRequest_1.CreateSecurityGroupRequest) {
                    body = createSecurityGroupRequest.body;
                }
                else {
                    body = createSecurityGroupRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 创建安全组规则
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        createSecurityGroupRule: function (createSecurityGroupRuleRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/security-group-rules",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (createSecurityGroupRuleRequest !== null && createSecurityGroupRuleRequest !== undefined) {
                if (createSecurityGroupRuleRequest instanceof CreateSecurityGroupRuleRequest_1.CreateSecurityGroupRuleRequest) {
                    body = createSecurityGroupRuleRequest.body;
                }
                else {
                    body = createSecurityGroupRuleRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 创建辅助弹性网卡
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        createSubNetworkInterface: function (createSubNetworkInterfaceRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/sub-network-interfaces",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (createSubNetworkInterfaceRequest !== null && createSubNetworkInterfaceRequest !== undefined) {
                if (createSubNetworkInterfaceRequest instanceof CreateSubNetworkInterfaceRequest_1.CreateSubNetworkInterfaceRequest) {
                    body = createSubNetworkInterfaceRequest.body;
                }
                else {
                    body = createSubNetworkInterfaceRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 删除安全组
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteSecurityGroup: function (deleteSecurityGroupRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/security-groups/{security_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var securityGroupId;
            if (deleteSecurityGroupRequest !== null && deleteSecurityGroupRequest !== undefined) {
                if (deleteSecurityGroupRequest instanceof DeleteSecurityGroupRequest_1.DeleteSecurityGroupRequest) {
                    securityGroupId = deleteSecurityGroupRequest.securityGroupId;
                }
                else {
                    securityGroupId = deleteSecurityGroupRequest['security_group_id'];
                }
            }
            if (securityGroupId === null || securityGroupId === undefined) {
                throw new RequiredError('securityGroupId', 'Required parameter securityGroupId was null or undefined when calling deleteSecurityGroup.');
            }
            options.pathParams = { 'security_group_id': securityGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 删除安全组规则
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteSecurityGroupRule: function (deleteSecurityGroupRuleRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/security-group-rules/{security_group_rule_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var securityGroupRuleId;
            if (deleteSecurityGroupRuleRequest !== null && deleteSecurityGroupRuleRequest !== undefined) {
                if (deleteSecurityGroupRuleRequest instanceof DeleteSecurityGroupRuleRequest_1.DeleteSecurityGroupRuleRequest) {
                    securityGroupRuleId = deleteSecurityGroupRuleRequest.securityGroupRuleId;
                }
                else {
                    securityGroupRuleId = deleteSecurityGroupRuleRequest['security_group_rule_id'];
                }
            }
            if (securityGroupRuleId === null || securityGroupRuleId === undefined) {
                throw new RequiredError('securityGroupRuleId', 'Required parameter securityGroupRuleId was null or undefined when calling deleteSecurityGroupRule.');
            }
            options.pathParams = { 'security_group_rule_id': securityGroupRuleId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 删除辅助弹性网卡
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteSubNetworkInterface: function (deleteSubNetworkInterfaceRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/{sub_network_interface_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var subNetworkInterfaceId;
            if (deleteSubNetworkInterfaceRequest !== null && deleteSubNetworkInterfaceRequest !== undefined) {
                if (deleteSubNetworkInterfaceRequest instanceof DeleteSubNetworkInterfaceRequest_1.DeleteSubNetworkInterfaceRequest) {
                    subNetworkInterfaceId = deleteSubNetworkInterfaceRequest.subNetworkInterfaceId;
                }
                else {
                    subNetworkInterfaceId = deleteSubNetworkInterfaceRequest['sub_network_interface_id'];
                }
            }
            if (subNetworkInterfaceId === null || subNetworkInterfaceId === undefined) {
                throw new RequiredError('subNetworkInterfaceId', 'Required parameter subNetworkInterfaceId was null or undefined when calling deleteSubNetworkInterface.');
            }
            options.pathParams = { 'sub_network_interface_id': subNetworkInterfaceId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询安全组规则列表
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        listSecurityGroupRules: function (listSecurityGroupRulesRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/security-group-rules",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var localVarQueryParameter = {};
            var limit;
            var marker;
            var id;
            var securityGroupId;
            var protocol;
            var description;
            var remoteGroupId;
            var direction;
            var action;
            if (listSecurityGroupRulesRequest !== null && listSecurityGroupRulesRequest !== undefined) {
                if (listSecurityGroupRulesRequest instanceof ListSecurityGroupRulesRequest_1.ListSecurityGroupRulesRequest) {
                    limit = listSecurityGroupRulesRequest.limit;
                    marker = listSecurityGroupRulesRequest.marker;
                    id = listSecurityGroupRulesRequest.id;
                    securityGroupId = listSecurityGroupRulesRequest.securityGroupId;
                    protocol = listSecurityGroupRulesRequest.protocol;
                    description = listSecurityGroupRulesRequest.description;
                    remoteGroupId = listSecurityGroupRulesRequest.remoteGroupId;
                    direction = listSecurityGroupRulesRequest.direction;
                    action = listSecurityGroupRulesRequest.action;
                }
                else {
                    limit = listSecurityGroupRulesRequest['limit'];
                    marker = listSecurityGroupRulesRequest['marker'];
                    id = listSecurityGroupRulesRequest['id'];
                    securityGroupId = listSecurityGroupRulesRequest['security_group_id'];
                    protocol = listSecurityGroupRulesRequest['protocol'];
                    description = listSecurityGroupRulesRequest['description'];
                    remoteGroupId = listSecurityGroupRulesRequest['remote_group_id'];
                    direction = listSecurityGroupRulesRequest['direction'];
                    action = listSecurityGroupRulesRequest['action'];
                }
            }
            if (limit !== null && limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (marker !== null && marker !== undefined) {
                localVarQueryParameter['marker'] = marker;
            }
            if (id !== null && id !== undefined) {
                localVarQueryParameter['id'] = id;
            }
            if (securityGroupId !== null && securityGroupId !== undefined) {
                localVarQueryParameter['security_group_id'] = securityGroupId;
            }
            if (protocol !== null && protocol !== undefined) {
                localVarQueryParameter['protocol'] = protocol;
            }
            if (description !== null && description !== undefined) {
                localVarQueryParameter['description'] = description;
            }
            if (remoteGroupId !== null && remoteGroupId !== undefined) {
                localVarQueryParameter['remote_group_id'] = remoteGroupId;
            }
            if (direction !== null && direction !== undefined) {
                localVarQueryParameter['direction'] = direction;
            }
            if (action !== null && action !== undefined) {
                localVarQueryParameter['action'] = action;
            }
            options.queryParams = localVarQueryParameter;
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询某租户下的安全组列表
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        listSecurityGroups: function (listSecurityGroupsRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/security-groups",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var localVarQueryParameter = {};
            var limit;
            var marker;
            var id;
            var name;
            var description;
            var enterpriseProjectId;
            if (listSecurityGroupsRequest !== null && listSecurityGroupsRequest !== undefined) {
                if (listSecurityGroupsRequest instanceof ListSecurityGroupsRequest_1.ListSecurityGroupsRequest) {
                    limit = listSecurityGroupsRequest.limit;
                    marker = listSecurityGroupsRequest.marker;
                    id = listSecurityGroupsRequest.id;
                    name = listSecurityGroupsRequest.name;
                    description = listSecurityGroupsRequest.description;
                    enterpriseProjectId = listSecurityGroupsRequest.enterpriseProjectId;
                }
                else {
                    limit = listSecurityGroupsRequest['limit'];
                    marker = listSecurityGroupsRequest['marker'];
                    id = listSecurityGroupsRequest['id'];
                    name = listSecurityGroupsRequest['name'];
                    description = listSecurityGroupsRequest['description'];
                    enterpriseProjectId = listSecurityGroupsRequest['enterprise_project_id'];
                }
            }
            if (limit !== null && limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (marker !== null && marker !== undefined) {
                localVarQueryParameter['marker'] = marker;
            }
            if (id !== null && id !== undefined) {
                localVarQueryParameter['id'] = id;
            }
            if (name !== null && name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (description !== null && description !== undefined) {
                localVarQueryParameter['description'] = description;
            }
            if (enterpriseProjectId !== null && enterpriseProjectId !== undefined) {
                localVarQueryParameter['enterprise_project_id'] = enterpriseProjectId;
            }
            options.queryParams = localVarQueryParameter;
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询辅助弹性网卡列表，单次查询最多返回2000条数据
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        listSubNetworkInterfaces: function (listSubNetworkInterfacesRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/sub-network-interfaces",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var localVarQueryParameter = {};
            var limit;
            var marker;
            var id;
            var virsubnetId;
            var privateIpAddress;
            var macAddress;
            var vpcId;
            var description;
            var parentId;
            if (listSubNetworkInterfacesRequest !== null && listSubNetworkInterfacesRequest !== undefined) {
                if (listSubNetworkInterfacesRequest instanceof ListSubNetworkInterfacesRequest_1.ListSubNetworkInterfacesRequest) {
                    limit = listSubNetworkInterfacesRequest.limit;
                    marker = listSubNetworkInterfacesRequest.marker;
                    id = listSubNetworkInterfacesRequest.id;
                    virsubnetId = listSubNetworkInterfacesRequest.virsubnetId;
                    privateIpAddress = listSubNetworkInterfacesRequest.privateIpAddress;
                    macAddress = listSubNetworkInterfacesRequest.macAddress;
                    vpcId = listSubNetworkInterfacesRequest.vpcId;
                    description = listSubNetworkInterfacesRequest.description;
                    parentId = listSubNetworkInterfacesRequest.parentId;
                }
                else {
                    limit = listSubNetworkInterfacesRequest['limit'];
                    marker = listSubNetworkInterfacesRequest['marker'];
                    id = listSubNetworkInterfacesRequest['id'];
                    virsubnetId = listSubNetworkInterfacesRequest['virsubnet_id'];
                    privateIpAddress = listSubNetworkInterfacesRequest['private_ip_address'];
                    macAddress = listSubNetworkInterfacesRequest['mac_address'];
                    vpcId = listSubNetworkInterfacesRequest['vpc_id'];
                    description = listSubNetworkInterfacesRequest['description'];
                    parentId = listSubNetworkInterfacesRequest['parent_id'];
                }
            }
            if (limit !== null && limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (marker !== null && marker !== undefined) {
                localVarQueryParameter['marker'] = marker;
            }
            if (id !== null && id !== undefined) {
                localVarQueryParameter['id'] = id;
            }
            if (virsubnetId !== null && virsubnetId !== undefined) {
                localVarQueryParameter['virsubnet_id'] = virsubnetId;
            }
            if (privateIpAddress !== null && privateIpAddress !== undefined) {
                localVarQueryParameter['private_ip_address'] = privateIpAddress;
            }
            if (macAddress !== null && macAddress !== undefined) {
                localVarQueryParameter['mac_address'] = macAddress;
            }
            if (vpcId !== null && vpcId !== undefined) {
                localVarQueryParameter['vpc_id'] = vpcId;
            }
            if (description !== null && description !== undefined) {
                localVarQueryParameter['description'] = description;
            }
            if (parentId !== null && parentId !== undefined) {
                localVarQueryParameter['parent_id'] = parentId;
            }
            options.queryParams = localVarQueryParameter;
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 批量迁移辅助弹性网卡
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        migrateSubNetworkInterface: function (migrateSubNetworkInterfaceRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/migrate",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (migrateSubNetworkInterfaceRequest !== null && migrateSubNetworkInterfaceRequest !== undefined) {
                if (migrateSubNetworkInterfaceRequest instanceof MigrateSubNetworkInterfaceRequest_1.MigrateSubNetworkInterfaceRequest) {
                    body = migrateSubNetworkInterfaceRequest.body;
                }
                else {
                    body = migrateSubNetworkInterfaceRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询单个安全组详情
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showSecurityGroup: function (showSecurityGroupRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/security-groups/{security_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var securityGroupId;
            if (showSecurityGroupRequest !== null && showSecurityGroupRequest !== undefined) {
                if (showSecurityGroupRequest instanceof ShowSecurityGroupRequest_1.ShowSecurityGroupRequest) {
                    securityGroupId = showSecurityGroupRequest.securityGroupId;
                }
                else {
                    securityGroupId = showSecurityGroupRequest['security_group_id'];
                }
            }
            if (securityGroupId === null || securityGroupId === undefined) {
                throw new RequiredError('securityGroupId', 'Required parameter securityGroupId was null or undefined when calling showSecurityGroup.');
            }
            options.pathParams = { 'security_group_id': securityGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询单个安全组规则
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showSecurityGroupRule: function (showSecurityGroupRuleRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/security-group-rules/{security_group_rule_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var securityGroupRuleId;
            if (showSecurityGroupRuleRequest !== null && showSecurityGroupRuleRequest !== undefined) {
                if (showSecurityGroupRuleRequest instanceof ShowSecurityGroupRuleRequest_1.ShowSecurityGroupRuleRequest) {
                    securityGroupRuleId = showSecurityGroupRuleRequest.securityGroupRuleId;
                }
                else {
                    securityGroupRuleId = showSecurityGroupRuleRequest['security_group_rule_id'];
                }
            }
            if (securityGroupRuleId === null || securityGroupRuleId === undefined) {
                throw new RequiredError('securityGroupRuleId', 'Required parameter securityGroupRuleId was null or undefined when calling showSecurityGroupRule.');
            }
            options.pathParams = { 'security_group_rule_id': securityGroupRuleId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询辅助弹性网卡详情
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showSubNetworkInterface: function (showSubNetworkInterfaceRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/{sub_network_interface_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var subNetworkInterfaceId;
            if (showSubNetworkInterfaceRequest !== null && showSubNetworkInterfaceRequest !== undefined) {
                if (showSubNetworkInterfaceRequest instanceof ShowSubNetworkInterfaceRequest_1.ShowSubNetworkInterfaceRequest) {
                    subNetworkInterfaceId = showSubNetworkInterfaceRequest.subNetworkInterfaceId;
                }
                else {
                    subNetworkInterfaceId = showSubNetworkInterfaceRequest['sub_network_interface_id'];
                }
            }
            if (subNetworkInterfaceId === null || subNetworkInterfaceId === undefined) {
                throw new RequiredError('subNetworkInterfaceId', 'Required parameter subNetworkInterfaceId was null or undefined when calling showSubNetworkInterface.');
            }
            options.pathParams = { 'sub_network_interface_id': subNetworkInterfaceId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询辅助弹性网卡数目
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showSubNetworkInterfacesQuantity: function () {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/count",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 更新安全组
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        updateSecurityGroup: function (updateSecurityGroupRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/security-groups/{security_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var securityGroupId;
            if (updateSecurityGroupRequest !== null && updateSecurityGroupRequest !== undefined) {
                if (updateSecurityGroupRequest instanceof UpdateSecurityGroupRequest_1.UpdateSecurityGroupRequest) {
                    securityGroupId = updateSecurityGroupRequest.securityGroupId;
                    body = updateSecurityGroupRequest.body;
                }
                else {
                    securityGroupId = updateSecurityGroupRequest['security_group_id'];
                    body = updateSecurityGroupRequest['body'];
                }
            }
            if (securityGroupId === null || securityGroupId === undefined) {
                throw new RequiredError('securityGroupId', 'Required parameter securityGroupId was null or undefined when calling updateSecurityGroup.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'security_group_id': securityGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 更新辅助弹性网卡
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        updateSubNetworkInterface: function (updateSubNetworkInterfaceRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/sub-network-interfaces/{sub_network_interface_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var subNetworkInterfaceId;
            if (updateSubNetworkInterfaceRequest !== null && updateSubNetworkInterfaceRequest !== undefined) {
                if (updateSubNetworkInterfaceRequest instanceof UpdateSubNetworkInterfaceRequest_1.UpdateSubNetworkInterfaceRequest) {
                    subNetworkInterfaceId = updateSubNetworkInterfaceRequest.subNetworkInterfaceId;
                    body = updateSubNetworkInterfaceRequest.body;
                }
                else {
                    subNetworkInterfaceId = updateSubNetworkInterfaceRequest['sub_network_interface_id'];
                    body = updateSubNetworkInterfaceRequest['body'];
                }
            }
            if (subNetworkInterfaceId === null || subNetworkInterfaceId === undefined) {
                throw new RequiredError('subNetworkInterfaceId', 'Required parameter subNetworkInterfaceId was null or undefined when calling updateSubNetworkInterface.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'sub_network_interface_id': subNetworkInterfaceId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 创建地址组
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        createAddressGroup: function (createAddressGroupRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/address-groups",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (createAddressGroupRequest !== null && createAddressGroupRequest !== undefined) {
                if (createAddressGroupRequest instanceof CreateAddressGroupRequest_1.CreateAddressGroupRequest) {
                    body = createAddressGroupRequest.body;
                }
                else {
                    body = createAddressGroupRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 删除地址组，非强制删除，删除前请确保未被其他资源引用
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteAddressGroup: function (deleteAddressGroupRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/address-groups/{address_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var addressGroupId;
            if (deleteAddressGroupRequest !== null && deleteAddressGroupRequest !== undefined) {
                if (deleteAddressGroupRequest instanceof DeleteAddressGroupRequest_1.DeleteAddressGroupRequest) {
                    addressGroupId = deleteAddressGroupRequest.addressGroupId;
                }
                else {
                    addressGroupId = deleteAddressGroupRequest['address_group_id'];
                }
            }
            if (addressGroupId === null || addressGroupId === undefined) {
                throw new RequiredError('addressGroupId', 'Required parameter addressGroupId was null or undefined when calling deleteAddressGroup.');
            }
            options.pathParams = { 'address_group_id': addressGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 强制删除地址组，删除的地址组与安全组规则关联时，会删除地址组与关联的安全组规则。
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteIpAddressGroupForce: function (deleteIpAddressGroupForceRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/address-groups/{address_group_id}/force",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var addressGroupId;
            if (deleteIpAddressGroupForceRequest !== null && deleteIpAddressGroupForceRequest !== undefined) {
                if (deleteIpAddressGroupForceRequest instanceof DeleteIpAddressGroupForceRequest_1.DeleteIpAddressGroupForceRequest) {
                    addressGroupId = deleteIpAddressGroupForceRequest.addressGroupId;
                }
                else {
                    addressGroupId = deleteIpAddressGroupForceRequest['address_group_id'];
                }
            }
            if (addressGroupId === null || addressGroupId === undefined) {
                throw new RequiredError('addressGroupId', 'Required parameter addressGroupId was null or undefined when calling deleteIpAddressGroupForce.');
            }
            options.pathParams = { 'address_group_id': addressGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询地址组列表，根据过滤条件进行过滤。
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        listAddressGroup: function (listAddressGroupRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/address-groups",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var localVarQueryParameter = {};
            var limit;
            var marker;
            var id;
            var name;
            var ipVersion;
            var description;
            if (listAddressGroupRequest !== null && listAddressGroupRequest !== undefined) {
                if (listAddressGroupRequest instanceof ListAddressGroupRequest_1.ListAddressGroupRequest) {
                    limit = listAddressGroupRequest.limit;
                    marker = listAddressGroupRequest.marker;
                    id = listAddressGroupRequest.id;
                    name = listAddressGroupRequest.name;
                    ipVersion = listAddressGroupRequest.ipVersion;
                    description = listAddressGroupRequest.description;
                }
                else {
                    limit = listAddressGroupRequest['limit'];
                    marker = listAddressGroupRequest['marker'];
                    id = listAddressGroupRequest['id'];
                    name = listAddressGroupRequest['name'];
                    ipVersion = listAddressGroupRequest['ip_version'];
                    description = listAddressGroupRequest['description'];
                }
            }
            if (limit !== null && limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (marker !== null && marker !== undefined) {
                localVarQueryParameter['marker'] = marker;
            }
            if (id !== null && id !== undefined) {
                localVarQueryParameter['id'] = id;
            }
            if (name !== null && name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (ipVersion !== null && ipVersion !== undefined) {
                localVarQueryParameter['ip_version'] = ipVersion;
            }
            if (description !== null && description !== undefined) {
                localVarQueryParameter['description'] = description;
            }
            options.queryParams = localVarQueryParameter;
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询地址组详情。
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showAddressGroup: function (showAddressGroupRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/address-groups/{address_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var addressGroupId;
            if (showAddressGroupRequest !== null && showAddressGroupRequest !== undefined) {
                if (showAddressGroupRequest instanceof ShowAddressGroupRequest_1.ShowAddressGroupRequest) {
                    addressGroupId = showAddressGroupRequest.addressGroupId;
                }
                else {
                    addressGroupId = showAddressGroupRequest['address_group_id'];
                }
            }
            if (addressGroupId === null || addressGroupId === undefined) {
                throw new RequiredError('addressGroupId', 'Required parameter addressGroupId was null or undefined when calling showAddressGroup.');
            }
            options.pathParams = { 'address_group_id': addressGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 更新地址组。
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        updateAddressGroup: function (updateAddressGroupRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/address-groups/{address_group_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var addressGroupId;
            if (updateAddressGroupRequest !== null && updateAddressGroupRequest !== undefined) {
                if (updateAddressGroupRequest instanceof UpdateAddressGroupRequest_1.UpdateAddressGroupRequest) {
                    addressGroupId = updateAddressGroupRequest.addressGroupId;
                    body = updateAddressGroupRequest.body;
                }
                else {
                    addressGroupId = updateAddressGroupRequest['address_group_id'];
                    body = updateAddressGroupRequest['body'];
                }
            }
            if (addressGroupId === null || addressGroupId === undefined) {
                throw new RequiredError('addressGroupId', 'Required parameter addressGroupId was null or undefined when calling updateAddressGroup.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'address_group_id': addressGroupId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 添加VPC的扩展网段
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        addVpcExtendCidr: function (addVpcExtendCidrRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/vpcs/{vpc_id}/add-extend-cidr",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var vpcId;
            if (addVpcExtendCidrRequest !== null && addVpcExtendCidrRequest !== undefined) {
                if (addVpcExtendCidrRequest instanceof AddVpcExtendCidrRequest_1.AddVpcExtendCidrRequest) {
                    vpcId = addVpcExtendCidrRequest.vpcId;
                    body = addVpcExtendCidrRequest.body;
                }
                else {
                    vpcId = addVpcExtendCidrRequest['vpc_id'];
                    body = addVpcExtendCidrRequest['body'];
                }
            }
            if (vpcId === null || vpcId === undefined) {
                throw new RequiredError('vpcId', 'Required parameter vpcId was null or undefined when calling addVpcExtendCidr.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'vpc_id': vpcId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 创建虚拟私有云
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        createVpc: function (createVpcRequest) {
            var options = {
                method: "POST",
                url: "/v3/{project_id}/vpc/vpcs",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            if (createVpcRequest !== null && createVpcRequest !== undefined) {
                if (createVpcRequest instanceof CreateVpcRequest_1.CreateVpcRequest) {
                    body = createVpcRequest.body;
                }
                else {
                    body = createVpcRequest['body'];
                }
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 删除VPC
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        deleteVpc: function (deleteVpcRequest) {
            var options = {
                method: "DELETE",
                url: "/v3/{project_id}/vpc/vpcs/{vpc_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var vpcId;
            if (deleteVpcRequest !== null && deleteVpcRequest !== undefined) {
                if (deleteVpcRequest instanceof DeleteVpcRequest_1.DeleteVpcRequest) {
                    vpcId = deleteVpcRequest.vpcId;
                }
                else {
                    vpcId = deleteVpcRequest['vpc_id'];
                }
            }
            if (vpcId === null || vpcId === undefined) {
                throw new RequiredError('vpcId', 'Required parameter vpcId was null or undefined when calling deleteVpc.');
            }
            options.pathParams = { 'vpc_id': vpcId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询vpc列表
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        listVpcs: function (listVpcsRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/vpcs",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var localVarQueryParameter = {};
            var limit;
            var marker;
            var id;
            var name;
            var description;
            var cidr;
            if (listVpcsRequest !== null && listVpcsRequest !== undefined) {
                if (listVpcsRequest instanceof ListVpcsRequest_1.ListVpcsRequest) {
                    limit = listVpcsRequest.limit;
                    marker = listVpcsRequest.marker;
                    id = listVpcsRequest.id;
                    name = listVpcsRequest.name;
                    description = listVpcsRequest.description;
                    cidr = listVpcsRequest.cidr;
                }
                else {
                    limit = listVpcsRequest['limit'];
                    marker = listVpcsRequest['marker'];
                    id = listVpcsRequest['id'];
                    name = listVpcsRequest['name'];
                    description = listVpcsRequest['description'];
                    cidr = listVpcsRequest['cidr'];
                }
            }
            if (limit !== null && limit !== undefined) {
                localVarQueryParameter['limit'] = limit;
            }
            if (marker !== null && marker !== undefined) {
                localVarQueryParameter['marker'] = marker;
            }
            if (id !== null && id !== undefined) {
                localVarQueryParameter['id'] = id;
            }
            if (name !== null && name !== undefined) {
                localVarQueryParameter['name'] = name;
            }
            if (description !== null && description !== undefined) {
                localVarQueryParameter['description'] = description;
            }
            if (cidr !== null && cidr !== undefined) {
                localVarQueryParameter['cidr'] = cidr;
            }
            options.queryParams = localVarQueryParameter;
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 移除VPC扩展网段
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        removeVpcExtendCidr: function (removeVpcExtendCidrRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/vpcs/{vpc_id}/remove-extend-cidr",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var vpcId;
            if (removeVpcExtendCidrRequest !== null && removeVpcExtendCidrRequest !== undefined) {
                if (removeVpcExtendCidrRequest instanceof RemoveVpcExtendCidrRequest_1.RemoveVpcExtendCidrRequest) {
                    vpcId = removeVpcExtendCidrRequest.vpcId;
                    body = removeVpcExtendCidrRequest.body;
                }
                else {
                    vpcId = removeVpcExtendCidrRequest['vpc_id'];
                    body = removeVpcExtendCidrRequest['body'];
                }
            }
            if (vpcId === null || vpcId === undefined) {
                throw new RequiredError('vpcId', 'Required parameter vpcId was null or undefined when calling removeVpcExtendCidr.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'vpc_id': vpcId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 查询vpc详情
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        showVpc: function (showVpcRequest) {
            var options = {
                method: "GET",
                url: "/v3/{project_id}/vpc/vpcs/{vpc_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var vpcId;
            if (showVpcRequest !== null && showVpcRequest !== undefined) {
                if (showVpcRequest instanceof ShowVpcRequest_1.ShowVpcRequest) {
                    vpcId = showVpcRequest.vpcId;
                }
                else {
                    vpcId = showVpcRequest['vpc_id'];
                }
            }
            if (vpcId === null || vpcId === undefined) {
                throw new RequiredError('vpcId', 'Required parameter vpcId was null or undefined when calling showVpc.');
            }
            options.pathParams = { 'vpc_id': vpcId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
        /**
         * 更新vpc
         *
         * Please refer to HUAWEI cloud API Explorer for details.
         */
        updateVpc: function (updateVpcRequest) {
            var options = {
                method: "PUT",
                url: "/v3/{project_id}/vpc/vpcs/{vpc_id}",
                contentType: "application/json",
                queryParams: {},
                pathParams: {},
                headers: {},
                data: {}
            };
            var localVarHeaderParameter = {};
            var body;
            var vpcId;
            if (updateVpcRequest !== null && updateVpcRequest !== undefined) {
                if (updateVpcRequest instanceof UpdateVpcRequest_1.UpdateVpcRequest) {
                    vpcId = updateVpcRequest.vpcId;
                    body = updateVpcRequest.body;
                }
                else {
                    vpcId = updateVpcRequest['vpc_id'];
                    body = updateVpcRequest['body'];
                }
            }
            if (vpcId === null || vpcId === undefined) {
                throw new RequiredError('vpcId', 'Required parameter vpcId was null or undefined when calling updateVpc.');
            }
            if (body === null || body === undefined) {
                throw new RequiredError('body', 'Required parameter body was null or undefined when calling body.');
            }
            localVarHeaderParameter['Content-Type'] = 'application/json';
            options.data = body !== undefined ? body : {};
            options.pathParams = { 'vpc_id': vpcId, };
            options.headers = localVarHeaderParameter;
            return options;
        },
    };
};
exports.ParamCreater = ParamCreater;
function newClient(client) {
    return new VpcClient(client);
}
/**
 *
 * @export
 * @class RequiredError
 * @extends {Error}
 */
var RequiredError = /** @class */ (function (_super) {
    __extends(RequiredError, _super);
    function RequiredError(field, msg) {
        var _this = _super.call(this, msg) || this;
        _this.field = field;
        _this.name = "RequiredError";
        return _this;
    }
    return RequiredError;
}(Error));
exports.RequiredError = RequiredError;
