// @huaweicloud/huaweicloud-sdk-vpc/v3/model/AddressGroup.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddressGroup = void 0;
var AddressGroup = /** @class */ (function () {
    function AddressGroup(id, name, description, ipSet, ipVersion, createdAt, updatedAt, tenantId) {
        this['id'] = id;
        this['name'] = name;
        this['description'] = description;
        this['ip_set'] = ipSet;
        this['ip_version'] = ipVersion;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['tenant_id'] = tenantId;
    }
    AddressGroup.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    AddressGroup.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    AddressGroup.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddressGroup.prototype.withIpSet = function (ipSet) {
        this['ip_set'] = ipSet;
        return this;
    };
    Object.defineProperty(AddressGroup.prototype, "ipSet", {
        get: function () {
            return this['ip_set'];
        },
        set: function (ipSet) {
            this['ip_set'] = ipSet;
        },
        enumerable: false,
        configurable: true
    });
    AddressGroup.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(AddressGroup.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    AddressGroup.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(AddressGroup.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    AddressGroup.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(AddressGroup.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    AddressGroup.prototype.withTenantId = function (tenantId) {
        this['tenant_id'] = tenantId;
        return this;
    };
    Object.defineProperty(AddressGroup.prototype, "tenantId", {
        get: function () {
            return this['tenant_id'];
        },
        set: function (tenantId) {
            this['tenant_id'] = tenantId;
        },
        enumerable: false,
        configurable: true
    });
    return AddressGroup;
}());
exports.AddressGroup = AddressGroup;
