// @huaweicloud/huaweicloud-sdk-vpc/v3/model/BatchCreateSubNetworkInterfaceRequestBody.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateSubNetworkInterfaceRequestBody = void 0;
var BatchCreateSubNetworkInterfaceRequestBody = /** @class */ (function () {
    function BatchCreateSubNetworkInterfaceRequestBody(subNetworkInterface, count) {
        this['sub_network_interface'] = subNetworkInterface;
        this['count'] = count;
    }
    BatchCreateSubNetworkInterfaceRequestBody.prototype.withDryRun = function (dryRun) {
        this['dry_run'] = dryRun;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceRequestBody.prototype, "dryRun", {
        get: function () {
            return this['dry_run'];
        },
        set: function (dryRun) {
            this['dry_run'] = dryRun;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceRequestBody.prototype.withSubNetworkInterface = function (subNetworkInterface) {
        this['sub_network_interface'] = subNetworkInterface;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceRequestBody.prototype, "subNetworkInterface", {
        get: function () {
            return this['sub_network_interface'];
        },
        set: function (subNetworkInterface) {
            this['sub_network_interface'] = subNetworkInterface;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceRequestBody.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    return BatchCreateSubNetworkInterfaceRequestBody;
}());
exports.BatchCreateSubNetworkInterfaceRequestBody = BatchCreateSubNetworkInterfaceRequestBody;
