// @huaweicloud/huaweicloud-sdk-vpc/v3/model/BatchCreateSubNetworkInterfaceOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchCreateSubNetworkInterfaceOption = void 0;
var BatchCreateSubNetworkInterfaceOption = /** @class */ (function () {
    function BatchCreateSubNetworkInterfaceOption(virsubnetId, parentId) {
        this['virsubnet_id'] = virsubnetId;
        this['parent_id'] = parentId;
    }
    BatchCreateSubNetworkInterfaceOption.prototype.withVirsubnetId = function (virsubnetId) {
        this['virsubnet_id'] = virsubnetId;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceOption.prototype, "virsubnetId", {
        get: function () {
            return this['virsubnet_id'];
        },
        set: function (virsubnetId) {
            this['virsubnet_id'] = virsubnetId;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceOption.prototype.withParentId = function (parentId) {
        this['parent_id'] = parentId;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceOption.prototype, "parentId", {
        get: function () {
            return this['parent_id'];
        },
        set: function (parentId) {
            this['parent_id'] = parentId;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    BatchCreateSubNetworkInterfaceOption.prototype.withIpv6Enable = function (ipv6Enable) {
        this['ipv6_enable'] = ipv6Enable;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceOption.prototype, "ipv6Enable", {
        get: function () {
            return this['ipv6_enable'];
        },
        set: function (ipv6Enable) {
            this['ipv6_enable'] = ipv6Enable;
        },
        enumerable: false,
        configurable: true
    });
    BatchCreateSubNetworkInterfaceOption.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(BatchCreateSubNetworkInterfaceOption.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return BatchCreateSubNetworkInterfaceOption;
}());
exports.BatchCreateSubNetworkInterfaceOption = BatchCreateSubNetworkInterfaceOption;
