// @huaweicloud/huaweicloud-sdk-vpc/v3/model/CreateAddressGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAddressGroupOption = void 0;
var CreateAddressGroupOption = /** @class */ (function () {
    function CreateAddressGroupOption(name, ipVersion) {
        this['name'] = name;
        this['ip_version'] = ipVersion;
    }
    CreateAddressGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateAddressGroupOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateAddressGroupOption.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(CreateAddressGroupOption.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    CreateAddressGroupOption.prototype.withIpSet = function (ipSet) {
        this['ip_set'] = ipSet;
        return this;
    };
    Object.defineProperty(CreateAddressGroupOption.prototype, "ipSet", {
        get: function () {
            return this['ip_set'];
        },
        set: function (ipSet) {
            this['ip_set'] = ipSet;
        },
        enumerable: false,
        configurable: true
    });
    return CreateAddressGroupOption;
}());
exports.CreateAddressGroupOption = CreateAddressGroupOption;
