// @huaweicloud/huaweicloud-sdk-vpc/v3/model/CreateSecurityGroupRuleOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSecurityGroupRuleOption = void 0;
var CreateSecurityGroupRuleOption = /** @class */ (function () {
    function CreateSecurityGroupRuleOption(securityGroupId, direction) {
        this['security_group_id'] = securityGroupId;
        this['direction'] = direction;
    }
    CreateSecurityGroupRuleOption.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withMultiport = function (multiport) {
        this['multiport'] = multiport;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withRemoteAddressGroupId = function (remoteAddressGroupId) {
        this['remote_address_group_id'] = remoteAddressGroupId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupRuleOption.prototype, "remoteAddressGroupId", {
        get: function () {
            return this['remote_address_group_id'];
        },
        set: function (remoteAddressGroupId) {
            this['remote_address_group_id'] = remoteAddressGroupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSecurityGroupRuleOption.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    CreateSecurityGroupRuleOption.prototype.withPriority = function (priority) {
        this['priority'] = priority;
        return this;
    };
    return CreateSecurityGroupRuleOption;
}());
exports.CreateSecurityGroupRuleOption = CreateSecurityGroupRuleOption;
