// @huaweicloud/huaweicloud-sdk-vpc/v3/model/CreateSubNetworkInterfaceOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSubNetworkInterfaceOption = void 0;
var CreateSubNetworkInterfaceOption = /** @class */ (function () {
    function CreateSubNetworkInterfaceOption(virsubnetId, parentId) {
        this['virsubnet_id'] = virsubnetId;
        this['parent_id'] = parentId;
    }
    CreateSubNetworkInterfaceOption.prototype.withVirsubnetId = function (virsubnetId) {
        this['virsubnet_id'] = virsubnetId;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "virsubnetId", {
        get: function () {
            return this['virsubnet_id'];
        },
        set: function (virsubnetId) {
            this['virsubnet_id'] = virsubnetId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withVlanId = function (vlanId) {
        this['vlan_id'] = vlanId;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "vlanId", {
        get: function () {
            return this['vlan_id'];
        },
        set: function (vlanId) {
            this['vlan_id'] = vlanId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withParentId = function (parentId) {
        this['parent_id'] = parentId;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "parentId", {
        get: function () {
            return this['parent_id'];
        },
        set: function (parentId) {
            this['parent_id'] = parentId;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateSubNetworkInterfaceOption.prototype.withIpv6Enable = function (ipv6Enable) {
        this['ipv6_enable'] = ipv6Enable;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "ipv6Enable", {
        get: function () {
            return this['ipv6_enable'];
        },
        set: function (ipv6Enable) {
            this['ipv6_enable'] = ipv6Enable;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withPrivateIpAddress = function (privateIpAddress) {
        this['private_ip_address'] = privateIpAddress;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "privateIpAddress", {
        get: function () {
            return this['private_ip_address'];
        },
        set: function (privateIpAddress) {
            this['private_ip_address'] = privateIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withIpv6IpAddress = function (ipv6IpAddress) {
        this['ipv6_ip_address'] = ipv6IpAddress;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "ipv6IpAddress", {
        get: function () {
            return this['ipv6_ip_address'];
        },
        set: function (ipv6IpAddress) {
            this['ipv6_ip_address'] = ipv6IpAddress;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    CreateSubNetworkInterfaceOption.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(CreateSubNetworkInterfaceOption.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateSubNetworkInterfaceOption;
}());
exports.CreateSubNetworkInterfaceOption = CreateSubNetworkInterfaceOption;
