// @huaweicloud/huaweicloud-sdk-vpc/v3/model/CreateVpcResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateVpcResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateVpcResponse = /** @class */ (function (_super) {
    __extends(CreateVpcResponse, _super);
    function CreateVpcResponse() {
        return _super.call(this) || this;
    }
    CreateVpcResponse.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(CreateVpcResponse.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    CreateVpcResponse.prototype.withErrorMsg = function (errorMsg) {
        this['error_msg'] = errorMsg;
        return this;
    };
    Object.defineProperty(CreateVpcResponse.prototype, "errorMsg", {
        get: function () {
            return this['error_msg'];
        },
        set: function (errorMsg) {
            this['error_msg'] = errorMsg;
        },
        enumerable: false,
        configurable: true
    });
    CreateVpcResponse.prototype.withErrorCode = function (errorCode) {
        this['error_code'] = errorCode;
        return this;
    };
    Object.defineProperty(CreateVpcResponse.prototype, "errorCode", {
        get: function () {
            return this['error_code'];
        },
        set: function (errorCode) {
            this['error_code'] = errorCode;
        },
        enumerable: false,
        configurable: true
    });
    return CreateVpcResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateVpcResponse = CreateVpcResponse;
