// @huaweicloud/huaweicloud-sdk-vpc/v3/model/CreateSecurityGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateSecurityGroupOption = void 0;
var CreateSecurityGroupOption = /** @class */ (function () {
    function CreateSecurityGroupOption(name) {
        this['name'] = name;
    }
    CreateSecurityGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateSecurityGroupOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateSecurityGroupOption.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(CreateSecurityGroupOption.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateSecurityGroupOption;
}());
exports.CreateSecurityGroupOption = CreateSecurityGroupOption;
