// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListAddressGroupResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAddressGroupResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListAddressGroupResponse = /** @class */ (function (_super) {
    __extends(ListAddressGroupResponse, _super);
    function ListAddressGroupResponse() {
        return _super.call(this) || this;
    }
    ListAddressGroupResponse.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(ListAddressGroupResponse.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    ListAddressGroupResponse.prototype.withAddressGroups = function (addressGroups) {
        this['address_groups'] = addressGroups;
        return this;
    };
    Object.defineProperty(ListAddressGroupResponse.prototype, "addressGroups", {
        get: function () {
            return this['address_groups'];
        },
        set: function (addressGroups) {
            this['address_groups'] = addressGroups;
        },
        enumerable: false,
        configurable: true
    });
    ListAddressGroupResponse.prototype.withPageInfo = function (pageInfo) {
        this['page_info'] = pageInfo;
        return this;
    };
    Object.defineProperty(ListAddressGroupResponse.prototype, "pageInfo", {
        get: function () {
            return this['page_info'];
        },
        set: function (pageInfo) {
            this['page_info'] = pageInfo;
        },
        enumerable: false,
        configurable: true
    });
    return ListAddressGroupResponse;
}(SdkResponse_1.SdkResponse));
exports.ListAddressGroupResponse = ListAddressGroupResponse;
