// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListSecurityGroupRulesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityGroupRulesRequest = void 0;
var ListSecurityGroupRulesRequest = /** @class */ (function () {
    function ListSecurityGroupRulesRequest() {
    }
    ListSecurityGroupRulesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupRulesRequest.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    ListSecurityGroupRulesRequest.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupRulesRequest.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    ListSecurityGroupRulesRequest.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    ListSecurityGroupRulesRequest.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    return ListSecurityGroupRulesRequest;
}());
exports.ListSecurityGroupRulesRequest = ListSecurityGroupRulesRequest;
