// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListAddressGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListAddressGroupRequest = void 0;
var ListAddressGroupRequest = /** @class */ (function () {
    function ListAddressGroupRequest() {
    }
    ListAddressGroupRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListAddressGroupRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListAddressGroupRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListAddressGroupRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListAddressGroupRequest.prototype.withIpVersion = function (ipVersion) {
        this['ip_version'] = ipVersion;
        return this;
    };
    Object.defineProperty(ListAddressGroupRequest.prototype, "ipVersion", {
        get: function () {
            return this['ip_version'];
        },
        set: function (ipVersion) {
            this['ip_version'] = ipVersion;
        },
        enumerable: false,
        configurable: true
    });
    ListAddressGroupRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return ListAddressGroupRequest;
}());
exports.ListAddressGroupRequest = ListAddressGroupRequest;
