// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListSecurityGroupsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityGroupsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListSecurityGroupsResponse = /** @class */ (function (_super) {
    __extends(ListSecurityGroupsResponse, _super);
    function ListSecurityGroupsResponse() {
        return _super.call(this) || this;
    }
    ListSecurityGroupsResponse.prototype.withSecurityGroups = function (securityGroups) {
        this['security_groups'] = securityGroups;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsResponse.prototype, "securityGroups", {
        get: function () {
            return this['security_groups'];
        },
        set: function (securityGroups) {
            this['security_groups'] = securityGroups;
        },
        enumerable: false,
        configurable: true
    });
    ListSecurityGroupsResponse.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsResponse.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    ListSecurityGroupsResponse.prototype.withPageInfo = function (pageInfo) {
        this['page_info'] = pageInfo;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsResponse.prototype, "pageInfo", {
        get: function () {
            return this['page_info'];
        },
        set: function (pageInfo) {
            this['page_info'] = pageInfo;
        },
        enumerable: false,
        configurable: true
    });
    return ListSecurityGroupsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListSecurityGroupsResponse = ListSecurityGroupsResponse;
