// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListVpcsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListVpcsResponse = /** @class */ (function (_super) {
    __extends(ListVpcsResponse, _super);
    function ListVpcsResponse() {
        return _super.call(this) || this;
    }
    ListVpcsResponse.prototype.withRequestId = function (requestId) {
        this['request_id'] = requestId;
        return this;
    };
    Object.defineProperty(ListVpcsResponse.prototype, "requestId", {
        get: function () {
            return this['request_id'];
        },
        set: function (requestId) {
            this['request_id'] = requestId;
        },
        enumerable: false,
        configurable: true
    });
    ListVpcsResponse.prototype.withVpcs = function (vpcs) {
        this['vpcs'] = vpcs;
        return this;
    };
    ListVpcsResponse.prototype.withPageInfo = function (pageInfo) {
        this['page_info'] = pageInfo;
        return this;
    };
    Object.defineProperty(ListVpcsResponse.prototype, "pageInfo", {
        get: function () {
            return this['page_info'];
        },
        set: function (pageInfo) {
            this['page_info'] = pageInfo;
        },
        enumerable: false,
        configurable: true
    });
    return ListVpcsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListVpcsResponse = ListVpcsResponse;
