// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListSecurityGroupsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSecurityGroupsRequest = void 0;
var ListSecurityGroupsRequest = /** @class */ (function () {
    function ListSecurityGroupsRequest() {
    }
    ListSecurityGroupsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListSecurityGroupsRequest.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(ListSecurityGroupsRequest.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    return ListSecurityGroupsRequest;
}());
exports.ListSecurityGroupsRequest = ListSecurityGroupsRequest;
