// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListVpcsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListVpcsRequest = void 0;
var ListVpcsRequest = /** @class */ (function () {
    function ListVpcsRequest() {
    }
    ListVpcsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListVpcsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListVpcsRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListVpcsRequest.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ListVpcsRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListVpcsRequest.prototype.withCidr = function (cidr) {
        this['cidr'] = cidr;
        return this;
    };
    return ListVpcsRequest;
}());
exports.ListVpcsRequest = ListVpcsRequest;
