// @huaweicloud/huaweicloud-sdk-vpc/v3/model/ListSubNetworkInterfacesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListSubNetworkInterfacesRequest = void 0;
var ListSubNetworkInterfacesRequest = /** @class */ (function () {
    function ListSubNetworkInterfacesRequest() {
    }
    ListSubNetworkInterfacesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListSubNetworkInterfacesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListSubNetworkInterfacesRequest.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    ListSubNetworkInterfacesRequest.prototype.withVirsubnetId = function (virsubnetId) {
        this['virsubnet_id'] = virsubnetId;
        return this;
    };
    Object.defineProperty(ListSubNetworkInterfacesRequest.prototype, "virsubnetId", {
        get: function () {
            return this['virsubnet_id'];
        },
        set: function (virsubnetId) {
            this['virsubnet_id'] = virsubnetId;
        },
        enumerable: false,
        configurable: true
    });
    ListSubNetworkInterfacesRequest.prototype.withPrivateIpAddress = function (privateIpAddress) {
        this['private_ip_address'] = privateIpAddress;
        return this;
    };
    Object.defineProperty(ListSubNetworkInterfacesRequest.prototype, "privateIpAddress", {
        get: function () {
            return this['private_ip_address'];
        },
        set: function (privateIpAddress) {
            this['private_ip_address'] = privateIpAddress;
        },
        enumerable: false,
        configurable: true
    });
    ListSubNetworkInterfacesRequest.prototype.withMacAddress = function (macAddress) {
        this['mac_address'] = macAddress;
        return this;
    };
    Object.defineProperty(ListSubNetworkInterfacesRequest.prototype, "macAddress", {
        get: function () {
            return this['mac_address'];
        },
        set: function (macAddress) {
            this['mac_address'] = macAddress;
        },
        enumerable: false,
        configurable: true
    });
    ListSubNetworkInterfacesRequest.prototype.withVpcId = function (vpcId) {
        this['vpc_id'] = vpcId;
        return this;
    };
    Object.defineProperty(ListSubNetworkInterfacesRequest.prototype, "vpcId", {
        get: function () {
            return this['vpc_id'];
        },
        set: function (vpcId) {
            this['vpc_id'] = vpcId;
        },
        enumerable: false,
        configurable: true
    });
    ListSubNetworkInterfacesRequest.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ListSubNetworkInterfacesRequest.prototype.withParentId = function (parentId) {
        this['parent_id'] = parentId;
        return this;
    };
    Object.defineProperty(ListSubNetworkInterfacesRequest.prototype, "parentId", {
        get: function () {
            return this['parent_id'];
        },
        set: function (parentId) {
            this['parent_id'] = parentId;
        },
        enumerable: false,
        configurable: true
    });
    return ListSubNetworkInterfacesRequest;
}());
exports.ListSubNetworkInterfacesRequest = ListSubNetworkInterfacesRequest;
