// @huaweicloud/huaweicloud-sdk-vpc/v3/model/PageInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PageInfo = void 0;
var PageInfo = /** @class */ (function () {
    function PageInfo(previousMarker, currentCount) {
        this['previous_marker'] = previousMarker;
        this['current_count'] = currentCount;
    }
    PageInfo.prototype.withPreviousMarker = function (previousMarker) {
        this['previous_marker'] = previousMarker;
        return this;
    };
    Object.defineProperty(PageInfo.prototype, "previousMarker", {
        get: function () {
            return this['previous_marker'];
        },
        set: function (previousMarker) {
            this['previous_marker'] = previousMarker;
        },
        enumerable: false,
        configurable: true
    });
    PageInfo.prototype.withCurrentCount = function (currentCount) {
        this['current_count'] = currentCount;
        return this;
    };
    Object.defineProperty(PageInfo.prototype, "currentCount", {
        get: function () {
            return this['current_count'];
        },
        set: function (currentCount) {
            this['current_count'] = currentCount;
        },
        enumerable: false,
        configurable: true
    });
    PageInfo.prototype.withNextMarker = function (nextMarker) {
        this['next_marker'] = nextMarker;
        return this;
    };
    Object.defineProperty(PageInfo.prototype, "nextMarker", {
        get: function () {
            return this['next_marker'];
        },
        set: function (nextMarker) {
            this['next_marker'] = nextMarker;
        },
        enumerable: false,
        configurable: true
    });
    return PageInfo;
}());
exports.PageInfo = PageInfo;
