// @huaweicloud/huaweicloud-sdk-vpc/v3/model/SecurityGroupRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroupRule = void 0;
var SecurityGroupRule = /** @class */ (function () {
    function SecurityGroupRule(id, description, securityGroupId, direction, protocol, ethertype, multiport, action, priority, remoteGroupId, remoteIpPrefix, remoteAddressGroupId, createdAt, updatedAt, projectId) {
        this['id'] = id;
        this['description'] = description;
        this['security_group_id'] = securityGroupId;
        this['direction'] = direction;
        this['protocol'] = protocol;
        this['ethertype'] = ethertype;
        this['multiport'] = multiport;
        this['action'] = action;
        this['priority'] = priority;
        this['remote_group_id'] = remoteGroupId;
        this['remote_ip_prefix'] = remoteIpPrefix;
        this['remote_address_group_id'] = remoteAddressGroupId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['project_id'] = projectId;
    }
    SecurityGroupRule.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    SecurityGroupRule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    SecurityGroupRule.prototype.withSecurityGroupId = function (securityGroupId) {
        this['security_group_id'] = securityGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "securityGroupId", {
        get: function () {
            return this['security_group_id'];
        },
        set: function (securityGroupId) {
            this['security_group_id'] = securityGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withDirection = function (direction) {
        this['direction'] = direction;
        return this;
    };
    SecurityGroupRule.prototype.withProtocol = function (protocol) {
        this['protocol'] = protocol;
        return this;
    };
    SecurityGroupRule.prototype.withEthertype = function (ethertype) {
        this['ethertype'] = ethertype;
        return this;
    };
    SecurityGroupRule.prototype.withMultiport = function (multiport) {
        this['multiport'] = multiport;
        return this;
    };
    SecurityGroupRule.prototype.withAction = function (action) {
        this['action'] = action;
        return this;
    };
    SecurityGroupRule.prototype.withPriority = function (priority) {
        this['priority'] = priority;
        return this;
    };
    SecurityGroupRule.prototype.withRemoteGroupId = function (remoteGroupId) {
        this['remote_group_id'] = remoteGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteGroupId", {
        get: function () {
            return this['remote_group_id'];
        },
        set: function (remoteGroupId) {
            this['remote_group_id'] = remoteGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withRemoteIpPrefix = function (remoteIpPrefix) {
        this['remote_ip_prefix'] = remoteIpPrefix;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteIpPrefix", {
        get: function () {
            return this['remote_ip_prefix'];
        },
        set: function (remoteIpPrefix) {
            this['remote_ip_prefix'] = remoteIpPrefix;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withRemoteAddressGroupId = function (remoteAddressGroupId) {
        this['remote_address_group_id'] = remoteAddressGroupId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "remoteAddressGroupId", {
        get: function () {
            return this['remote_address_group_id'];
        },
        set: function (remoteAddressGroupId) {
            this['remote_address_group_id'] = remoteAddressGroupId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupRule.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(SecurityGroupRule.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    return SecurityGroupRule;
}());
exports.SecurityGroupRule = SecurityGroupRule;
