// @huaweicloud/huaweicloud-sdk-vpc/v3/model/SecurityGroupInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SecurityGroupInfo = void 0;
var SecurityGroupInfo = /** @class */ (function () {
    function SecurityGroupInfo(id, name, description, projectId, createdAt, updatedAt, enterpriseProjectId, securityGroupRules) {
        this['id'] = id;
        this['name'] = name;
        this['description'] = description;
        this['project_id'] = projectId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['enterprise_project_id'] = enterpriseProjectId;
        this['security_group_rules'] = securityGroupRules;
    }
    SecurityGroupInfo.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    SecurityGroupInfo.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    SecurityGroupInfo.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    SecurityGroupInfo.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(SecurityGroupInfo.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupInfo.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(SecurityGroupInfo.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupInfo.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(SecurityGroupInfo.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupInfo.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(SecurityGroupInfo.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    SecurityGroupInfo.prototype.withSecurityGroupRules = function (securityGroupRules) {
        this['security_group_rules'] = securityGroupRules;
        return this;
    };
    Object.defineProperty(SecurityGroupInfo.prototype, "securityGroupRules", {
        get: function () {
            return this['security_group_rules'];
        },
        set: function (securityGroupRules) {
            this['security_group_rules'] = securityGroupRules;
        },
        enumerable: false,
        configurable: true
    });
    return SecurityGroupInfo;
}());
exports.SecurityGroupInfo = SecurityGroupInfo;
