// @huaweicloud/huaweicloud-sdk-vpc/v3/model/UpdateAddressGroupOption.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateAddressGroupOption = void 0;
var UpdateAddressGroupOption = /** @class */ (function () {
    function UpdateAddressGroupOption() {
    }
    UpdateAddressGroupOption.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateAddressGroupOption.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateAddressGroupOption.prototype.withIpSet = function (ipSet) {
        this['ip_set'] = ipSet;
        return this;
    };
    Object.defineProperty(UpdateAddressGroupOption.prototype, "ipSet", {
        get: function () {
            return this['ip_set'];
        },
        set: function (ipSet) {
            this['ip_set'] = ipSet;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateAddressGroupOption;
}());
exports.UpdateAddressGroupOption = UpdateAddressGroupOption;
