// @huaweicloud/huaweicloud-sdk-vpc/v3/model/Vpc.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Vpc = void 0;
var Vpc = /** @class */ (function () {
    function Vpc(id, name, description, cidr, extendCidrs, status, projectId, enterpriseProjectId, createdAt, updatedAt, cloudResources, tags) {
        this['id'] = id;
        this['name'] = name;
        this['description'] = description;
        this['cidr'] = cidr;
        this['extend_cidrs'] = extendCidrs;
        this['status'] = status;
        this['project_id'] = projectId;
        this['enterprise_project_id'] = enterpriseProjectId;
        this['created_at'] = createdAt;
        this['updated_at'] = updatedAt;
        this['cloud_resources'] = cloudResources;
        this['tags'] = tags;
    }
    Vpc.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Vpc.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Vpc.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    Vpc.prototype.withCidr = function (cidr) {
        this['cidr'] = cidr;
        return this;
    };
    Vpc.prototype.withExtendCidrs = function (extendCidrs) {
        this['extend_cidrs'] = extendCidrs;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "extendCidrs", {
        get: function () {
            return this['extend_cidrs'];
        },
        set: function (extendCidrs) {
            this['extend_cidrs'] = extendCidrs;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    Vpc.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withEnterpriseProjectId = function (enterpriseProjectId) {
        this['enterprise_project_id'] = enterpriseProjectId;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "enterpriseProjectId", {
        get: function () {
            return this['enterprise_project_id'];
        },
        set: function (enterpriseProjectId) {
            this['enterprise_project_id'] = enterpriseProjectId;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withCreatedAt = function (createdAt) {
        this['created_at'] = createdAt;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "createdAt", {
        get: function () {
            return this['created_at'];
        },
        set: function (createdAt) {
            this['created_at'] = createdAt;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withUpdatedAt = function (updatedAt) {
        this['updated_at'] = updatedAt;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "updatedAt", {
        get: function () {
            return this['updated_at'];
        },
        set: function (updatedAt) {
            this['updated_at'] = updatedAt;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withCloudResources = function (cloudResources) {
        this['cloud_resources'] = cloudResources;
        return this;
    };
    Object.defineProperty(Vpc.prototype, "cloudResources", {
        get: function () {
            return this['cloud_resources'];
        },
        set: function (cloudResources) {
            this['cloud_resources'] = cloudResources;
        },
        enumerable: false,
        configurable: true
    });
    Vpc.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return Vpc;
}());
exports.Vpc = Vpc;
