// @huaweicloud/huaweicloud-sdk-core/internal/model/Domains.js
"use strict";
/*
 * Copyright 2021 Huawei Technologies Co.,Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Domains = void 0;
var Domains = /** @class */ (function () {
    function Domains(enabled, id, name, links, description) {
        this['enabled'] = enabled;
        this['id'] = id;
        this['name'] = name;
        this['links'] = links;
        this['description'] = description;
    }
    Domains.prototype.withEnabled = function (enabled) {
        this['enabled'] = enabled;
        return this;
    };
    Domains.prototype.withId = function (id) {
        this['id'] = id;
        return this;
    };
    Domains.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Domains.prototype.withLinks = function (links) {
        this['links'] = links;
        return this;
    };
    Domains.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return Domains;
}());
exports.Domains = Domains;
