// @huaweicloud/huaweicloud-sdk-core/internal/model/Links.js
"use strict";
/*
 * Copyright 2021 Huawei Technologies Co.,Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.Links = void 0;
var Links = /** @class */ (function () {
    function Links(self, previous, next) {
        this['self'] = self;
        this['previous'] = previous;
        this['next'] = next;
    }
    Links.prototype.withSelf = function (self) {
        this['self'] = self;
        return this;
    };
    Links.prototype.withPrevious = function (previous) {
        this['previous'] = previous;
        return this;
    };
    Links.prototype.withNext = function (next) {
        this['next'] = next;
        return this;
    };
    return Links;
}());
exports.Links = Links;
