// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ActionSmnForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionSmnForwarding = void 0;
var ActionSmnForwarding = /** @class */ (function () {
    function ActionSmnForwarding(regionName, projectId, themeName, topicUrn, messageContent, messageTitle) {
        this['region_name'] = regionName;
        this['project_id'] = projectId;
        this['theme_name'] = themeName;
        this['topic_urn'] = topicUrn;
        this['message_content'] = messageContent;
        this['message_title'] = messageTitle;
    }
    ActionSmnForwarding.prototype.withRegionName = function (regionName) {
        this['region_name'] = regionName;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "regionName", {
        get: function () {
            return this['region_name'];
        },
        set: function (regionName) {
            this['region_name'] = regionName;
        },
        enumerable: false,
        configurable: true
    });
    ActionSmnForwarding.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    ActionSmnForwarding.prototype.withThemeName = function (themeName) {
        this['theme_name'] = themeName;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "themeName", {
        get: function () {
            return this['theme_name'];
        },
        set: function (themeName) {
            this['theme_name'] = themeName;
        },
        enumerable: false,
        configurable: true
    });
    ActionSmnForwarding.prototype.withTopicUrn = function (topicUrn) {
        this['topic_urn'] = topicUrn;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "topicUrn", {
        get: function () {
            return this['topic_urn'];
        },
        set: function (topicUrn) {
            this['topic_urn'] = topicUrn;
        },
        enumerable: false,
        configurable: true
    });
    ActionSmnForwarding.prototype.withMessageContent = function (messageContent) {
        this['message_content'] = messageContent;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "messageContent", {
        get: function () {
            return this['message_content'];
        },
        set: function (messageContent) {
            this['message_content'] = messageContent;
        },
        enumerable: false,
        configurable: true
    });
    ActionSmnForwarding.prototype.withMessageTitle = function (messageTitle) {
        this['message_title'] = messageTitle;
        return this;
    };
    Object.defineProperty(ActionSmnForwarding.prototype, "messageTitle", {
        get: function () {
            return this['message_title'];
        },
        set: function (messageTitle) {
            this['message_title'] = messageTitle;
        },
        enumerable: false,
        configurable: true
    });
    return ActionSmnForwarding;
}());
exports.ActionSmnForwarding = ActionSmnForwarding;
