// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ActionDeviceAlarm.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionDeviceAlarm = void 0;
var ActionDeviceAlarm = /** @class */ (function () {
    function ActionDeviceAlarm(name, alarmStatus, severity) {
        this['name'] = name;
        this['alarm_status'] = alarmStatus;
        this['severity'] = severity;
    }
    ActionDeviceAlarm.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ActionDeviceAlarm.prototype.withAlarmStatus = function (alarmStatus) {
        this['alarm_status'] = alarmStatus;
        return this;
    };
    Object.defineProperty(ActionDeviceAlarm.prototype, "alarmStatus", {
        get: function () {
            return this['alarm_status'];
        },
        set: function (alarmStatus) {
            this['alarm_status'] = alarmStatus;
        },
        enumerable: false,
        configurable: true
    });
    ActionDeviceAlarm.prototype.withSeverity = function (severity) {
        this['severity'] = severity;
        return this;
    };
    ActionDeviceAlarm.prototype.withDimension = function (dimension) {
        this['dimension'] = dimension;
        return this;
    };
    ActionDeviceAlarm.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return ActionDeviceAlarm;
}());
exports.ActionDeviceAlarm = ActionDeviceAlarm;
