// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddActionReq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddActionReq = void 0;
var AddActionReq = /** @class */ (function () {
    function AddActionReq(ruleId, channel, channelDetail) {
        this['rule_id'] = ruleId;
        this['channel'] = channel;
        this['channel_detail'] = channelDetail;
    }
    AddActionReq.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(AddActionReq.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    AddActionReq.prototype.withChannel = function (channel) {
        this['channel'] = channel;
        return this;
    };
    AddActionReq.prototype.withChannelDetail = function (channelDetail) {
        this['channel_detail'] = channelDetail;
        return this;
    };
    Object.defineProperty(AddActionReq.prototype, "channelDetail", {
        get: function () {
            return this['channel_detail'];
        },
        set: function (channelDetail) {
            this['channel_detail'] = channelDetail;
        },
        enumerable: false,
        configurable: true
    });
    return AddActionReq;
}());
exports.AddActionReq = AddActionReq;
