// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddCertificateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddCertificateRequest = void 0;
var AddCertificateRequest = /** @class */ (function () {
    function AddCertificateRequest() {
    }
    AddCertificateRequest.prototype.withSpAuthToken = function (spAuthToken) {
        this['Sp-Auth-Token'] = spAuthToken;
        return this;
    };
    Object.defineProperty(AddCertificateRequest.prototype, "spAuthToken", {
        get: function () {
            return this['Sp-Auth-Token'];
        },
        set: function (spAuthToken) {
            this['Sp-Auth-Token'] = spAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    AddCertificateRequest.prototype.withStageAuthToken = function (stageAuthToken) {
        this['Stage-Auth-Token'] = stageAuthToken;
        return this;
    };
    Object.defineProperty(AddCertificateRequest.prototype, "stageAuthToken", {
        get: function () {
            return this['Stage-Auth-Token'];
        },
        set: function (stageAuthToken) {
            this['Stage-Auth-Token'] = stageAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    AddCertificateRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(AddCertificateRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    AddCertificateRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return AddCertificateRequest;
}());
exports.AddCertificateRequest = AddCertificateRequest;
