// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddDeviceGroupResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddDeviceGroupResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var AddDeviceGroupResponse = /** @class */ (function (_super) {
    __extends(AddDeviceGroupResponse, _super);
    function AddDeviceGroupResponse() {
        return _super.call(this) || this;
    }
    AddDeviceGroupResponse.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(AddDeviceGroupResponse.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceGroupResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    AddDeviceGroupResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddDeviceGroupResponse.prototype.withSuperGroupId = function (superGroupId) {
        this['super_group_id'] = superGroupId;
        return this;
    };
    Object.defineProperty(AddDeviceGroupResponse.prototype, "superGroupId", {
        get: function () {
            return this['super_group_id'];
        },
        set: function (superGroupId) {
            this['super_group_id'] = superGroupId;
        },
        enumerable: false,
        configurable: true
    });
    return AddDeviceGroupResponse;
}(SdkResponse_1.SdkResponse));
exports.AddDeviceGroupResponse = AddDeviceGroupResponse;
