// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddDevice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddDevice = void 0;
var AddDevice = /** @class */ (function () {
    function AddDevice(nodeId, productId) {
        this['node_id'] = nodeId;
        this['product_id'] = productId;
    }
    AddDevice.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withNodeId = function (nodeId) {
        this['node_id'] = nodeId;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "nodeId", {
        get: function () {
            return this['node_id'];
        },
        set: function (nodeId) {
            this['node_id'] = nodeId;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withAuthInfo = function (authInfo) {
        this['auth_info'] = authInfo;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "authInfo", {
        get: function () {
            return this['auth_info'];
        },
        set: function (authInfo) {
            this['auth_info'] = authInfo;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddDevice.prototype.withGatewayId = function (gatewayId) {
        this['gateway_id'] = gatewayId;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "gatewayId", {
        get: function () {
            return this['gateway_id'];
        },
        set: function (gatewayId) {
            this['gateway_id'] = gatewayId;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withExtensionInfo = function (extensionInfo) {
        this['extension_info'] = extensionInfo;
        return this;
    };
    Object.defineProperty(AddDevice.prototype, "extensionInfo", {
        get: function () {
            return this['extension_info'];
        },
        set: function (extensionInfo) {
            this['extension_info'] = extensionInfo;
        },
        enumerable: false,
        configurable: true
    });
    AddDevice.prototype.withShadow = function (shadow) {
        this['shadow'] = shadow;
        return this;
    };
    return AddDevice;
}());
exports.AddDevice = AddDevice;
