// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddDeviceResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddDeviceResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var AddDeviceResponse = /** @class */ (function (_super) {
    __extends(AddDeviceResponse, _super);
    function AddDeviceResponse() {
        return _super.call(this) || this;
    }
    AddDeviceResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withNodeId = function (nodeId) {
        this['node_id'] = nodeId;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "nodeId", {
        get: function () {
            return this['node_id'];
        },
        set: function (nodeId) {
            this['node_id'] = nodeId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withGatewayId = function (gatewayId) {
        this['gateway_id'] = gatewayId;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "gatewayId", {
        get: function () {
            return this['gateway_id'];
        },
        set: function (gatewayId) {
            this['gateway_id'] = gatewayId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withNodeType = function (nodeType) {
        this['node_type'] = nodeType;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "nodeType", {
        get: function () {
            return this['node_type'];
        },
        set: function (nodeType) {
            this['node_type'] = nodeType;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddDeviceResponse.prototype.withFwVersion = function (fwVersion) {
        this['fw_version'] = fwVersion;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "fwVersion", {
        get: function () {
            return this['fw_version'];
        },
        set: function (fwVersion) {
            this['fw_version'] = fwVersion;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withSwVersion = function (swVersion) {
        this['sw_version'] = swVersion;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "swVersion", {
        get: function () {
            return this['sw_version'];
        },
        set: function (swVersion) {
            this['sw_version'] = swVersion;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withDeviceSdkVersion = function (deviceSdkVersion) {
        this['device_sdk_version'] = deviceSdkVersion;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "deviceSdkVersion", {
        get: function () {
            return this['device_sdk_version'];
        },
        set: function (deviceSdkVersion) {
            this['device_sdk_version'] = deviceSdkVersion;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withAuthInfo = function (authInfo) {
        this['auth_info'] = authInfo;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "authInfo", {
        get: function () {
            return this['auth_info'];
        },
        set: function (authInfo) {
            this['auth_info'] = authInfo;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withProductName = function (productName) {
        this['product_name'] = productName;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "productName", {
        get: function () {
            return this['product_name'];
        },
        set: function (productName) {
            this['product_name'] = productName;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    AddDeviceResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withConnectionStatusUpdateTime = function (connectionStatusUpdateTime) {
        this['connection_status_update_time'] = connectionStatusUpdateTime;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "connectionStatusUpdateTime", {
        get: function () {
            return this['connection_status_update_time'];
        },
        set: function (connectionStatusUpdateTime) {
            this['connection_status_update_time'] = connectionStatusUpdateTime;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withActiveTime = function (activeTime) {
        this['active_time'] = activeTime;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "activeTime", {
        get: function () {
            return this['active_time'];
        },
        set: function (activeTime) {
            this['active_time'] = activeTime;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceResponse.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    AddDeviceResponse.prototype.withExtensionInfo = function (extensionInfo) {
        this['extension_info'] = extensionInfo;
        return this;
    };
    Object.defineProperty(AddDeviceResponse.prototype, "extensionInfo", {
        get: function () {
            return this['extension_info'];
        },
        set: function (extensionInfo) {
            this['extension_info'] = extensionInfo;
        },
        enumerable: false,
        configurable: true
    });
    return AddDeviceResponse;
}(SdkResponse_1.SdkResponse));
exports.AddDeviceResponse = AddDeviceResponse;
