// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddDeviceGroupDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddDeviceGroupDTO = void 0;
var AddDeviceGroupDTO = /** @class */ (function () {
    function AddDeviceGroupDTO() {
    }
    AddDeviceGroupDTO.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    AddDeviceGroupDTO.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddDeviceGroupDTO.prototype.withSuperGroupId = function (superGroupId) {
        this['super_group_id'] = superGroupId;
        return this;
    };
    Object.defineProperty(AddDeviceGroupDTO.prototype, "superGroupId", {
        get: function () {
            return this['super_group_id'];
        },
        set: function (superGroupId) {
            this['super_group_id'] = superGroupId;
        },
        enumerable: false,
        configurable: true
    });
    AddDeviceGroupDTO.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(AddDeviceGroupDTO.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return AddDeviceGroupDTO;
}());
exports.AddDeviceGroupDTO = AddDeviceGroupDTO;
