// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddProduct.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddProduct = void 0;
var AddProduct = /** @class */ (function () {
    function AddProduct(name, deviceType, protocolType, dataFormat, serviceCapabilities) {
        this['name'] = name;
        this['device_type'] = deviceType;
        this['protocol_type'] = protocolType;
        this['data_format'] = dataFormat;
        this['service_capabilities'] = serviceCapabilities;
    }
    AddProduct.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    AddProduct.prototype.withDeviceType = function (deviceType) {
        this['device_type'] = deviceType;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "deviceType", {
        get: function () {
            return this['device_type'];
        },
        set: function (deviceType) {
            this['device_type'] = deviceType;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withProtocolType = function (protocolType) {
        this['protocol_type'] = protocolType;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "protocolType", {
        get: function () {
            return this['protocol_type'];
        },
        set: function (protocolType) {
            this['protocol_type'] = protocolType;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withDataFormat = function (dataFormat) {
        this['data_format'] = dataFormat;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "dataFormat", {
        get: function () {
            return this['data_format'];
        },
        set: function (dataFormat) {
            this['data_format'] = dataFormat;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withServiceCapabilities = function (serviceCapabilities) {
        this['service_capabilities'] = serviceCapabilities;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "serviceCapabilities", {
        get: function () {
            return this['service_capabilities'];
        },
        set: function (serviceCapabilities) {
            this['service_capabilities'] = serviceCapabilities;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withManufacturerName = function (manufacturerName) {
        this['manufacturer_name'] = manufacturerName;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "manufacturerName", {
        get: function () {
            return this['manufacturer_name'];
        },
        set: function (manufacturerName) {
            this['manufacturer_name'] = manufacturerName;
        },
        enumerable: false,
        configurable: true
    });
    AddProduct.prototype.withIndustry = function (industry) {
        this['industry'] = industry;
        return this;
    };
    AddProduct.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddProduct.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(AddProduct.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return AddProduct;
}());
exports.AddProduct = AddProduct;
