// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddQueueResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddQueueResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var AddQueueResponse = /** @class */ (function (_super) {
    __extends(AddQueueResponse, _super);
    function AddQueueResponse() {
        return _super.call(this) || this;
    }
    AddQueueResponse.prototype.withQueueId = function (queueId) {
        this['queue_id'] = queueId;
        return this;
    };
    Object.defineProperty(AddQueueResponse.prototype, "queueId", {
        get: function () {
            return this['queue_id'];
        },
        set: function (queueId) {
            this['queue_id'] = queueId;
        },
        enumerable: false,
        configurable: true
    });
    AddQueueResponse.prototype.withQueueName = function (queueName) {
        this['queue_name'] = queueName;
        return this;
    };
    Object.defineProperty(AddQueueResponse.prototype, "queueName", {
        get: function () {
            return this['queue_name'];
        },
        set: function (queueName) {
            this['queue_name'] = queueName;
        },
        enumerable: false,
        configurable: true
    });
    AddQueueResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(AddQueueResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    AddQueueResponse.prototype.withLastModifyTime = function (lastModifyTime) {
        this['last_modify_time'] = lastModifyTime;
        return this;
    };
    Object.defineProperty(AddQueueResponse.prototype, "lastModifyTime", {
        get: function () {
            return this['last_modify_time'];
        },
        set: function (lastModifyTime) {
            this['last_modify_time'] = lastModifyTime;
        },
        enumerable: false,
        configurable: true
    });
    return AddQueueResponse;
}(SdkResponse_1.SdkResponse));
exports.AddQueueResponse = AddQueueResponse;
