// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AddRuleReq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AddRuleReq = void 0;
var AddRuleReq = /** @class */ (function () {
    function AddRuleReq(subject) {
        this['subject'] = subject;
    }
    AddRuleReq.prototype.withRuleName = function (ruleName) {
        this['rule_name'] = ruleName;
        return this;
    };
    Object.defineProperty(AddRuleReq.prototype, "ruleName", {
        get: function () {
            return this['rule_name'];
        },
        set: function (ruleName) {
            this['rule_name'] = ruleName;
        },
        enumerable: false,
        configurable: true
    });
    AddRuleReq.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    AddRuleReq.prototype.withSubject = function (subject) {
        this['subject'] = subject;
        return this;
    };
    AddRuleReq.prototype.withAppType = function (appType) {
        this['app_type'] = appType;
        return this;
    };
    Object.defineProperty(AddRuleReq.prototype, "appType", {
        get: function () {
            return this['app_type'];
        },
        set: function (appType) {
            this['app_type'] = appType;
        },
        enumerable: false,
        configurable: true
    });
    AddRuleReq.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(AddRuleReq.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    AddRuleReq.prototype.withSelect = function (select) {
        this['select'] = select;
        return this;
    };
    AddRuleReq.prototype.withWhere = function (where) {
        this['where'] = where;
        return this;
    };
    return AddRuleReq;
}());
exports.AddRuleReq = AddRuleReq;
