// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ApplicationDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApplicationDTO = void 0;
var ApplicationDTO = /** @class */ (function () {
    function ApplicationDTO() {
    }
    ApplicationDTO.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ApplicationDTO.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ApplicationDTO.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(ApplicationDTO.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    ApplicationDTO.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(ApplicationDTO.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    ApplicationDTO.prototype.withDefaultApp = function (defaultApp) {
        this['default_app'] = defaultApp;
        return this;
    };
    Object.defineProperty(ApplicationDTO.prototype, "defaultApp", {
        get: function () {
            return this['default_app'];
        },
        set: function (defaultApp) {
            this['default_app'] = defaultApp;
        },
        enumerable: false,
        configurable: true
    });
    return ApplicationDTO;
}());
exports.ApplicationDTO = ApplicationDTO;
