// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AsyncDeviceCommandRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AsyncDeviceCommandRequest = void 0;
var AsyncDeviceCommandRequest = /** @class */ (function () {
    function AsyncDeviceCommandRequest(paras, sendStrategy) {
        this['paras'] = paras;
        this['send_strategy'] = sendStrategy;
    }
    AsyncDeviceCommandRequest.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(AsyncDeviceCommandRequest.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    AsyncDeviceCommandRequest.prototype.withCommandName = function (commandName) {
        this['command_name'] = commandName;
        return this;
    };
    Object.defineProperty(AsyncDeviceCommandRequest.prototype, "commandName", {
        get: function () {
            return this['command_name'];
        },
        set: function (commandName) {
            this['command_name'] = commandName;
        },
        enumerable: false,
        configurable: true
    });
    AsyncDeviceCommandRequest.prototype.withParas = function (paras) {
        this['paras'] = paras;
        return this;
    };
    AsyncDeviceCommandRequest.prototype.withExpireTime = function (expireTime) {
        this['expire_time'] = expireTime;
        return this;
    };
    Object.defineProperty(AsyncDeviceCommandRequest.prototype, "expireTime", {
        get: function () {
            return this['expire_time'];
        },
        set: function (expireTime) {
            this['expire_time'] = expireTime;
        },
        enumerable: false,
        configurable: true
    });
    AsyncDeviceCommandRequest.prototype.withSendStrategy = function (sendStrategy) {
        this['send_strategy'] = sendStrategy;
        return this;
    };
    Object.defineProperty(AsyncDeviceCommandRequest.prototype, "sendStrategy", {
        get: function () {
            return this['send_strategy'];
        },
        set: function (sendStrategy) {
            this['send_strategy'] = sendStrategy;
        },
        enumerable: false,
        configurable: true
    });
    return AsyncDeviceCommandRequest;
}());
exports.AsyncDeviceCommandRequest = AsyncDeviceCommandRequest;
