// @huaweicloud/huaweicloud-sdk-iotda/v5/model/AuthInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AuthInfo = void 0;
var AuthInfo = /** @class */ (function () {
    function AuthInfo() {
    }
    AuthInfo.prototype.withAuthType = function (authType) {
        this['auth_type'] = authType;
        return this;
    };
    Object.defineProperty(AuthInfo.prototype, "authType", {
        get: function () {
            return this['auth_type'];
        },
        set: function (authType) {
            this['auth_type'] = authType;
        },
        enumerable: false,
        configurable: true
    });
    AuthInfo.prototype.withSecret = function (secret) {
        this['secret'] = secret;
        return this;
    };
    AuthInfo.prototype.withFingerprint = function (fingerprint) {
        this['fingerprint'] = fingerprint;
        return this;
    };
    AuthInfo.prototype.withSecureAccess = function (secureAccess) {
        this['secure_access'] = secureAccess;
        return this;
    };
    Object.defineProperty(AuthInfo.prototype, "secureAccess", {
        get: function () {
            return this['secure_access'];
        },
        set: function (secureAccess) {
            this['secure_access'] = secureAccess;
        },
        enumerable: false,
        configurable: true
    });
    AuthInfo.prototype.withTimeout = function (timeout) {
        this['timeout'] = timeout;
        return this;
    };
    return AuthInfo;
}());
exports.AuthInfo = AuthInfo;
