// @huaweicloud/huaweicloud-sdk-iotda/v5/model/BatchTaskFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchTaskFile = void 0;
var BatchTaskFile = /** @class */ (function () {
    function BatchTaskFile() {
    }
    BatchTaskFile.prototype.withFileId = function (fileId) {
        this['file_id'] = fileId;
        return this;
    };
    Object.defineProperty(BatchTaskFile.prototype, "fileId", {
        get: function () {
            return this['file_id'];
        },
        set: function (fileId) {
            this['file_id'] = fileId;
        },
        enumerable: false,
        configurable: true
    });
    BatchTaskFile.prototype.withFileName = function (fileName) {
        this['file_name'] = fileName;
        return this;
    };
    Object.defineProperty(BatchTaskFile.prototype, "fileName", {
        get: function () {
            return this['file_name'];
        },
        set: function (fileName) {
            this['file_name'] = fileName;
        },
        enumerable: false,
        configurable: true
    });
    BatchTaskFile.prototype.withUploadTime = function (uploadTime) {
        this['upload_time'] = uploadTime;
        return this;
    };
    Object.defineProperty(BatchTaskFile.prototype, "uploadTime", {
        get: function () {
            return this['upload_time'];
        },
        set: function (uploadTime) {
            this['upload_time'] = uploadTime;
        },
        enumerable: false,
        configurable: true
    });
    return BatchTaskFile;
}());
exports.BatchTaskFile = BatchTaskFile;
