// @huaweicloud/huaweicloud-sdk-iotda/v5/model/BatchShowQueueRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchShowQueueRequest = void 0;
var BatchShowQueueRequest = /** @class */ (function () {
    function BatchShowQueueRequest() {
    }
    BatchShowQueueRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(BatchShowQueueRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    BatchShowQueueRequest.prototype.withQueueName = function (queueName) {
        this['queue_name'] = queueName;
        return this;
    };
    Object.defineProperty(BatchShowQueueRequest.prototype, "queueName", {
        get: function () {
            return this['queue_name'];
        },
        set: function (queueName) {
            this['queue_name'] = queueName;
        },
        enumerable: false,
        configurable: true
    });
    BatchShowQueueRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    BatchShowQueueRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    BatchShowQueueRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return BatchShowQueueRequest;
}());
exports.BatchShowQueueRequest = BatchShowQueueRequest;
