// @huaweicloud/huaweicloud-sdk-iotda/v5/model/BindTagsDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BindTagsDTO = void 0;
var BindTagsDTO = /** @class */ (function () {
    function BindTagsDTO(resourceType, resourceId, tags) {
        this['resource_type'] = resourceType;
        this['resource_id'] = resourceId;
        this['tags'] = tags;
    }
    BindTagsDTO.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(BindTagsDTO.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    BindTagsDTO.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(BindTagsDTO.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    BindTagsDTO.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    return BindTagsDTO;
}());
exports.BindTagsDTO = BindTagsDTO;
