// @huaweicloud/huaweicloud-sdk-iotda/v5/model/BroadcastMessageResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.BroadcastMessageResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var BroadcastMessageResponse = /** @class */ (function (_super) {
    __extends(BroadcastMessageResponse, _super);
    function BroadcastMessageResponse() {
        return _super.call(this) || this;
    }
    BroadcastMessageResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(BroadcastMessageResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    BroadcastMessageResponse.prototype.withTopicFullName = function (topicFullName) {
        this['topic_full_name'] = topicFullName;
        return this;
    };
    Object.defineProperty(BroadcastMessageResponse.prototype, "topicFullName", {
        get: function () {
            return this['topic_full_name'];
        },
        set: function (topicFullName) {
            this['topic_full_name'] = topicFullName;
        },
        enumerable: false,
        configurable: true
    });
    BroadcastMessageResponse.prototype.withMessageId = function (messageId) {
        this['message_id'] = messageId;
        return this;
    };
    Object.defineProperty(BroadcastMessageResponse.prototype, "messageId", {
        get: function () {
            return this['message_id'];
        },
        set: function (messageId) {
            this['message_id'] = messageId;
        },
        enumerable: false,
        configurable: true
    });
    BroadcastMessageResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(BroadcastMessageResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    return BroadcastMessageResponse;
}(SdkResponse_1.SdkResponse));
exports.BroadcastMessageResponse = BroadcastMessageResponse;
