// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CertificatesRspDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificatesRspDTO = void 0;
var CertificatesRspDTO = /** @class */ (function () {
    function CertificatesRspDTO() {
    }
    CertificatesRspDTO.prototype.withCertificateId = function (certificateId) {
        this['certificate_id'] = certificateId;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "certificateId", {
        get: function () {
            return this['certificate_id'];
        },
        set: function (certificateId) {
            this['certificate_id'] = certificateId;
        },
        enumerable: false,
        configurable: true
    });
    CertificatesRspDTO.prototype.withCnName = function (cnName) {
        this['cn_name'] = cnName;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "cnName", {
        get: function () {
            return this['cn_name'];
        },
        set: function (cnName) {
            this['cn_name'] = cnName;
        },
        enumerable: false,
        configurable: true
    });
    CertificatesRspDTO.prototype.withOwner = function (owner) {
        this['owner'] = owner;
        return this;
    };
    CertificatesRspDTO.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    CertificatesRspDTO.prototype.withVerifyCode = function (verifyCode) {
        this['verify_code'] = verifyCode;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "verifyCode", {
        get: function () {
            return this['verify_code'];
        },
        set: function (verifyCode) {
            this['verify_code'] = verifyCode;
        },
        enumerable: false,
        configurable: true
    });
    CertificatesRspDTO.prototype.withCreateDate = function (createDate) {
        this['create_date'] = createDate;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "createDate", {
        get: function () {
            return this['create_date'];
        },
        set: function (createDate) {
            this['create_date'] = createDate;
        },
        enumerable: false,
        configurable: true
    });
    CertificatesRspDTO.prototype.withEffectiveDate = function (effectiveDate) {
        this['effective_date'] = effectiveDate;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "effectiveDate", {
        get: function () {
            return this['effective_date'];
        },
        set: function (effectiveDate) {
            this['effective_date'] = effectiveDate;
        },
        enumerable: false,
        configurable: true
    });
    CertificatesRspDTO.prototype.withExpiryDate = function (expiryDate) {
        this['expiry_date'] = expiryDate;
        return this;
    };
    Object.defineProperty(CertificatesRspDTO.prototype, "expiryDate", {
        get: function () {
            return this['expiry_date'];
        },
        set: function (expiryDate) {
            this['expiry_date'] = expiryDate;
        },
        enumerable: false,
        configurable: true
    });
    return CertificatesRspDTO;
}());
exports.CertificatesRspDTO = CertificatesRspDTO;
