// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ChangeRuleStatusRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChangeRuleStatusRequest = void 0;
var ChangeRuleStatusRequest = /** @class */ (function () {
    function ChangeRuleStatusRequest(ruleId) {
        this['rule_id'] = ruleId;
    }
    ChangeRuleStatusRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ChangeRuleStatusRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ChangeRuleStatusRequest.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(ChangeRuleStatusRequest.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    ChangeRuleStatusRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return ChangeRuleStatusRequest;
}());
exports.ChangeRuleStatusRequest = ChangeRuleStatusRequest;
