// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ChannelDetail.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ChannelDetail = void 0;
var ChannelDetail = /** @class */ (function () {
    function ChannelDetail() {
    }
    ChannelDetail.prototype.withHttpForwarding = function (httpForwarding) {
        this['http_forwarding'] = httpForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "httpForwarding", {
        get: function () {
            return this['http_forwarding'];
        },
        set: function (httpForwarding) {
            this['http_forwarding'] = httpForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withDisForwarding = function (disForwarding) {
        this['dis_forwarding'] = disForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "disForwarding", {
        get: function () {
            return this['dis_forwarding'];
        },
        set: function (disForwarding) {
            this['dis_forwarding'] = disForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withObsForwarding = function (obsForwarding) {
        this['obs_forwarding'] = obsForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "obsForwarding", {
        get: function () {
            return this['obs_forwarding'];
        },
        set: function (obsForwarding) {
            this['obs_forwarding'] = obsForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withAmqpForwarding = function (amqpForwarding) {
        this['amqp_forwarding'] = amqpForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "amqpForwarding", {
        get: function () {
            return this['amqp_forwarding'];
        },
        set: function (amqpForwarding) {
            this['amqp_forwarding'] = amqpForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withDmsKafkaForwarding = function (dmsKafkaForwarding) {
        this['dms_kafka_forwarding'] = dmsKafkaForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "dmsKafkaForwarding", {
        get: function () {
            return this['dms_kafka_forwarding'];
        },
        set: function (dmsKafkaForwarding) {
            this['dms_kafka_forwarding'] = dmsKafkaForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withRomaForwarding = function (romaForwarding) {
        this['roma_forwarding'] = romaForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "romaForwarding", {
        get: function () {
            return this['roma_forwarding'];
        },
        set: function (romaForwarding) {
            this['roma_forwarding'] = romaForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withMysqlForwarding = function (mysqlForwarding) {
        this['mysql_forwarding'] = mysqlForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "mysqlForwarding", {
        get: function () {
            return this['mysql_forwarding'];
        },
        set: function (mysqlForwarding) {
            this['mysql_forwarding'] = mysqlForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withInfluxdbForwarding = function (influxdbForwarding) {
        this['influxdb_forwarding'] = influxdbForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "influxdbForwarding", {
        get: function () {
            return this['influxdb_forwarding'];
        },
        set: function (influxdbForwarding) {
            this['influxdb_forwarding'] = influxdbForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withFunctiongraphForwarding = function (functiongraphForwarding) {
        this['functiongraph_forwarding'] = functiongraphForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "functiongraphForwarding", {
        get: function () {
            return this['functiongraph_forwarding'];
        },
        set: function (functiongraphForwarding) {
            this['functiongraph_forwarding'] = functiongraphForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withMrsKafkaForwarding = function (mrsKafkaForwarding) {
        this['mrs_kafka_forwarding'] = mrsKafkaForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "mrsKafkaForwarding", {
        get: function () {
            return this['mrs_kafka_forwarding'];
        },
        set: function (mrsKafkaForwarding) {
            this['mrs_kafka_forwarding'] = mrsKafkaForwarding;
        },
        enumerable: false,
        configurable: true
    });
    ChannelDetail.prototype.withDmsRocketmqForwarding = function (dmsRocketmqForwarding) {
        this['dms_rocketmq_forwarding'] = dmsRocketmqForwarding;
        return this;
    };
    Object.defineProperty(ChannelDetail.prototype, "dmsRocketmqForwarding", {
        get: function () {
            return this['dms_rocketmq_forwarding'];
        },
        set: function (dmsRocketmqForwarding) {
            this['dms_rocketmq_forwarding'] = dmsRocketmqForwarding;
        },
        enumerable: false,
        configurable: true
    });
    return ChannelDetail;
}());
exports.ChannelDetail = ChannelDetail;
