// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CheckCertificateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckCertificateRequest = void 0;
var CheckCertificateRequest = /** @class */ (function () {
    function CheckCertificateRequest(certificateId, actionId) {
        this['certificate_id'] = certificateId;
        this['action_id'] = actionId;
    }
    CheckCertificateRequest.prototype.withSpAuthToken = function (spAuthToken) {
        this['Sp-Auth-Token'] = spAuthToken;
        return this;
    };
    Object.defineProperty(CheckCertificateRequest.prototype, "spAuthToken", {
        get: function () {
            return this['Sp-Auth-Token'];
        },
        set: function (spAuthToken) {
            this['Sp-Auth-Token'] = spAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    CheckCertificateRequest.prototype.withStageAuthToken = function (stageAuthToken) {
        this['Stage-Auth-Token'] = stageAuthToken;
        return this;
    };
    Object.defineProperty(CheckCertificateRequest.prototype, "stageAuthToken", {
        get: function () {
            return this['Stage-Auth-Token'];
        },
        set: function (stageAuthToken) {
            this['Stage-Auth-Token'] = stageAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    CheckCertificateRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(CheckCertificateRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    CheckCertificateRequest.prototype.withCertificateId = function (certificateId) {
        this['certificate_id'] = certificateId;
        return this;
    };
    Object.defineProperty(CheckCertificateRequest.prototype, "certificateId", {
        get: function () {
            return this['certificate_id'];
        },
        set: function (certificateId) {
            this['certificate_id'] = certificateId;
        },
        enumerable: false,
        configurable: true
    });
    CheckCertificateRequest.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(CheckCertificateRequest.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    CheckCertificateRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return CheckCertificateRequest;
}());
exports.CheckCertificateRequest = CheckCertificateRequest;
