// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateBatchTaskResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBatchTaskResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateBatchTaskResponse = /** @class */ (function (_super) {
    __extends(CreateBatchTaskResponse, _super);
    function CreateBatchTaskResponse() {
        return _super.call(this) || this;
    }
    CreateBatchTaskResponse.prototype.withTaskId = function (taskId) {
        this['task_id'] = taskId;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "taskId", {
        get: function () {
            return this['task_id'];
        },
        set: function (taskId) {
            this['task_id'] = taskId;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withTaskName = function (taskName) {
        this['task_name'] = taskName;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "taskName", {
        get: function () {
            return this['task_name'];
        },
        set: function (taskName) {
            this['task_name'] = taskName;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withTaskType = function (taskType) {
        this['task_type'] = taskType;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "taskType", {
        get: function () {
            return this['task_type'];
        },
        set: function (taskType) {
            this['task_type'] = taskType;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withTargets = function (targets) {
        this['targets'] = targets;
        return this;
    };
    CreateBatchTaskResponse.prototype.withTargetsFilter = function (targetsFilter) {
        this['targets_filter'] = targetsFilter;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "targetsFilter", {
        get: function () {
            return this['targets_filter'];
        },
        set: function (targetsFilter) {
            this['targets_filter'] = targetsFilter;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withDocument = function (document) {
        this['document'] = document;
        return this;
    };
    CreateBatchTaskResponse.prototype.withTaskPolicy = function (taskPolicy) {
        this['task_policy'] = taskPolicy;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "taskPolicy", {
        get: function () {
            return this['task_policy'];
        },
        set: function (taskPolicy) {
            this['task_policy'] = taskPolicy;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    CreateBatchTaskResponse.prototype.withStatusDesc = function (statusDesc) {
        this['status_desc'] = statusDesc;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "statusDesc", {
        get: function () {
            return this['status_desc'];
        },
        set: function (statusDesc) {
            this['status_desc'] = statusDesc;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withTaskProgress = function (taskProgress) {
        this['task_progress'] = taskProgress;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "taskProgress", {
        get: function () {
            return this['task_progress'];
        },
        set: function (taskProgress) {
            this['task_progress'] = taskProgress;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTaskResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(CreateBatchTaskResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    return CreateBatchTaskResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateBatchTaskResponse = CreateBatchTaskResponse;
