// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateMessageRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMessageRequest = void 0;
var CreateMessageRequest = /** @class */ (function () {
    function CreateMessageRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    CreateMessageRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(CreateMessageRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateMessageRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(CreateMessageRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateMessageRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return CreateMessageRequest;
}());
exports.CreateMessageRequest = CreateMessageRequest;
