// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateAsyncCommandRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAsyncCommandRequest = void 0;
var CreateAsyncCommandRequest = /** @class */ (function () {
    function CreateAsyncCommandRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    CreateAsyncCommandRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return CreateAsyncCommandRequest;
}());
exports.CreateAsyncCommandRequest = CreateAsyncCommandRequest;
