// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateBatchTask.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateBatchTask = void 0;
var CreateBatchTask = /** @class */ (function () {
    function CreateBatchTask(taskName, taskType) {
        this['task_name'] = taskName;
        this['task_type'] = taskType;
    }
    CreateBatchTask.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTask.prototype.withTaskName = function (taskName) {
        this['task_name'] = taskName;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "taskName", {
        get: function () {
            return this['task_name'];
        },
        set: function (taskName) {
            this['task_name'] = taskName;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTask.prototype.withTaskType = function (taskType) {
        this['task_type'] = taskType;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "taskType", {
        get: function () {
            return this['task_type'];
        },
        set: function (taskType) {
            this['task_type'] = taskType;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTask.prototype.withTargets = function (targets) {
        this['targets'] = targets;
        return this;
    };
    CreateBatchTask.prototype.withTargetsFilter = function (targetsFilter) {
        this['targets_filter'] = targetsFilter;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "targetsFilter", {
        get: function () {
            return this['targets_filter'];
        },
        set: function (targetsFilter) {
            this['targets_filter'] = targetsFilter;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTask.prototype.withDocument = function (document) {
        this['document'] = document;
        return this;
    };
    CreateBatchTask.prototype.withTaskPolicy = function (taskPolicy) {
        this['task_policy'] = taskPolicy;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "taskPolicy", {
        get: function () {
            return this['task_policy'];
        },
        set: function (taskPolicy) {
            this['task_policy'] = taskPolicy;
        },
        enumerable: false,
        configurable: true
    });
    CreateBatchTask.prototype.withDocumentSource = function (documentSource) {
        this['document_source'] = documentSource;
        return this;
    };
    Object.defineProperty(CreateBatchTask.prototype, "documentSource", {
        get: function () {
            return this['document_source'];
        },
        set: function (documentSource) {
            this['document_source'] = documentSource;
        },
        enumerable: false,
        configurable: true
    });
    return CreateBatchTask;
}());
exports.CreateBatchTask = CreateBatchTask;
