// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateOrDeleteDeviceInGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOrDeleteDeviceInGroupRequest = void 0;
var CreateOrDeleteDeviceInGroupRequest = /** @class */ (function () {
    function CreateOrDeleteDeviceInGroupRequest(groupId, actionId, deviceId) {
        this['group_id'] = groupId;
        this['action_id'] = actionId;
        this['device_id'] = deviceId;
    }
    CreateOrDeleteDeviceInGroupRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(CreateOrDeleteDeviceInGroupRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateOrDeleteDeviceInGroupRequest.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(CreateOrDeleteDeviceInGroupRequest.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    CreateOrDeleteDeviceInGroupRequest.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(CreateOrDeleteDeviceInGroupRequest.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    CreateOrDeleteDeviceInGroupRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(CreateOrDeleteDeviceInGroupRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    return CreateOrDeleteDeviceInGroupRequest;
}());
exports.CreateOrDeleteDeviceInGroupRequest = CreateOrDeleteDeviceInGroupRequest;
