// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateAsyncCommandResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateAsyncCommandResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateAsyncCommandResponse = /** @class */ (function (_super) {
    __extends(CreateAsyncCommandResponse, _super);
    function CreateAsyncCommandResponse() {
        return _super.call(this) || this;
    }
    CreateAsyncCommandResponse.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withCommandId = function (commandId) {
        this['command_id'] = commandId;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "commandId", {
        get: function () {
            return this['command_id'];
        },
        set: function (commandId) {
            this['command_id'] = commandId;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withCommandName = function (commandName) {
        this['command_name'] = commandName;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "commandName", {
        get: function () {
            return this['command_name'];
        },
        set: function (commandName) {
            this['command_name'] = commandName;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withParas = function (paras) {
        this['paras'] = paras;
        return this;
    };
    CreateAsyncCommandResponse.prototype.withExpireTime = function (expireTime) {
        this['expire_time'] = expireTime;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "expireTime", {
        get: function () {
            return this['expire_time'];
        },
        set: function (expireTime) {
            this['expire_time'] = expireTime;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    CreateAsyncCommandResponse.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    CreateAsyncCommandResponse.prototype.withSendStrategy = function (sendStrategy) {
        this['send_strategy'] = sendStrategy;
        return this;
    };
    Object.defineProperty(CreateAsyncCommandResponse.prototype, "sendStrategy", {
        get: function () {
            return this['send_strategy'];
        },
        set: function (sendStrategy) {
            this['send_strategy'] = sendStrategy;
        },
        enumerable: false,
        configurable: true
    });
    return CreateAsyncCommandResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateAsyncCommandResponse = CreateAsyncCommandResponse;
