// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateRuleResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRuleResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateRuleResponse = /** @class */ (function (_super) {
    __extends(CreateRuleResponse, _super);
    function CreateRuleResponse() {
        return _super.call(this) || this;
    }
    CreateRuleResponse.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    CreateRuleResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateRuleResponse.prototype.withConditionGroup = function (conditionGroup) {
        this['condition_group'] = conditionGroup;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "conditionGroup", {
        get: function () {
            return this['condition_group'];
        },
        set: function (conditionGroup) {
            this['condition_group'] = conditionGroup;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withActions = function (actions) {
        this['actions'] = actions;
        return this;
    };
    CreateRuleResponse.prototype.withRuleType = function (ruleType) {
        this['rule_type'] = ruleType;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "ruleType", {
        get: function () {
            return this['rule_type'];
        },
        set: function (ruleType) {
            this['rule_type'] = ruleType;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    CreateRuleResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withEdgeNodeIds = function (edgeNodeIds) {
        this['edge_node_ids'] = edgeNodeIds;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "edgeNodeIds", {
        get: function () {
            return this['edge_node_ids'];
        },
        set: function (edgeNodeIds) {
            this['edge_node_ids'] = edgeNodeIds;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withLastUpdateTime = function (lastUpdateTime) {
        this['last_update_time'] = lastUpdateTime;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "lastUpdateTime", {
        get: function () {
            return this['last_update_time'];
        },
        set: function (lastUpdateTime) {
            this['last_update_time'] = lastUpdateTime;
        },
        enumerable: false,
        configurable: true
    });
    CreateRuleResponse.prototype.withDeviceSide = function (deviceSide) {
        this['device_side'] = deviceSide;
        return this;
    };
    Object.defineProperty(CreateRuleResponse.prototype, "deviceSide", {
        get: function () {
            return this['device_side'];
        },
        set: function (deviceSide) {
            this['device_side'] = deviceSide;
        },
        enumerable: false,
        configurable: true
    });
    return CreateRuleResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateRuleResponse = CreateRuleResponse;
