// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateCommandRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCommandRequest = void 0;
var CreateCommandRequest = /** @class */ (function () {
    function CreateCommandRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    CreateCommandRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(CreateCommandRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateCommandRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(CreateCommandRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    CreateCommandRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return CreateCommandRequest;
}());
exports.CreateCommandRequest = CreateCommandRequest;
