// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateRoutingRuleResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateRoutingRuleResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var CreateRoutingRuleResponse = /** @class */ (function (_super) {
    __extends(CreateRoutingRuleResponse, _super);
    function CreateRoutingRuleResponse() {
        return _super.call(this) || this;
    }
    CreateRoutingRuleResponse.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(CreateRoutingRuleResponse.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    CreateRoutingRuleResponse.prototype.withRuleName = function (ruleName) {
        this['rule_name'] = ruleName;
        return this;
    };
    Object.defineProperty(CreateRoutingRuleResponse.prototype, "ruleName", {
        get: function () {
            return this['rule_name'];
        },
        set: function (ruleName) {
            this['rule_name'] = ruleName;
        },
        enumerable: false,
        configurable: true
    });
    CreateRoutingRuleResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateRoutingRuleResponse.prototype.withSubject = function (subject) {
        this['subject'] = subject;
        return this;
    };
    CreateRoutingRuleResponse.prototype.withAppType = function (appType) {
        this['app_type'] = appType;
        return this;
    };
    Object.defineProperty(CreateRoutingRuleResponse.prototype, "appType", {
        get: function () {
            return this['app_type'];
        },
        set: function (appType) {
            this['app_type'] = appType;
        },
        enumerable: false,
        configurable: true
    });
    CreateRoutingRuleResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(CreateRoutingRuleResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    CreateRoutingRuleResponse.prototype.withSelect = function (select) {
        this['select'] = select;
        return this;
    };
    CreateRoutingRuleResponse.prototype.withWhere = function (where) {
        this['where'] = where;
        return this;
    };
    CreateRoutingRuleResponse.prototype.withActive = function (active) {
        this['active'] = active;
        return this;
    };
    return CreateRoutingRuleResponse;
}(SdkResponse_1.SdkResponse));
exports.CreateRoutingRuleResponse = CreateRoutingRuleResponse;
