// @huaweicloud/huaweicloud-sdk-iotda/v5/model/CreateOtaPackage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateOtaPackage = void 0;
var CreateOtaPackage = /** @class */ (function () {
    function CreateOtaPackage(appId, packageType, productId, version, fileLocation) {
        this['app_id'] = appId;
        this['package_type'] = packageType;
        this['product_id'] = productId;
        this['version'] = version;
        this['file_location'] = fileLocation;
    }
    CreateOtaPackage.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    CreateOtaPackage.prototype.withPackageType = function (packageType) {
        this['package_type'] = packageType;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "packageType", {
        get: function () {
            return this['package_type'];
        },
        set: function (packageType) {
            this['package_type'] = packageType;
        },
        enumerable: false,
        configurable: true
    });
    CreateOtaPackage.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    CreateOtaPackage.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    CreateOtaPackage.prototype.withSupportSourceVersions = function (supportSourceVersions) {
        this['support_source_versions'] = supportSourceVersions;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "supportSourceVersions", {
        get: function () {
            return this['support_source_versions'];
        },
        set: function (supportSourceVersions) {
            this['support_source_versions'] = supportSourceVersions;
        },
        enumerable: false,
        configurable: true
    });
    CreateOtaPackage.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    CreateOtaPackage.prototype.withCustomInfo = function (customInfo) {
        this['custom_info'] = customInfo;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "customInfo", {
        get: function () {
            return this['custom_info'];
        },
        set: function (customInfo) {
            this['custom_info'] = customInfo;
        },
        enumerable: false,
        configurable: true
    });
    CreateOtaPackage.prototype.withFileLocation = function (fileLocation) {
        this['file_location'] = fileLocation;
        return this;
    };
    Object.defineProperty(CreateOtaPackage.prototype, "fileLocation", {
        get: function () {
            return this['file_location'];
        },
        set: function (fileLocation) {
            this['file_location'] = fileLocation;
        },
        enumerable: false,
        configurable: true
    });
    return CreateOtaPackage;
}());
exports.CreateOtaPackage = CreateOtaPackage;
