// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeleteProductRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteProductRequest = void 0;
var DeleteProductRequest = /** @class */ (function () {
    function DeleteProductRequest(productId) {
        this['product_id'] = productId;
    }
    DeleteProductRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(DeleteProductRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    DeleteProductRequest.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(DeleteProductRequest.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    DeleteProductRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(DeleteProductRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return DeleteProductRequest;
}());
exports.DeleteProductRequest = DeleteProductRequest;
