// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeleteCertificateRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteCertificateRequest = void 0;
var DeleteCertificateRequest = /** @class */ (function () {
    function DeleteCertificateRequest(certificateId) {
        this['certificate_id'] = certificateId;
    }
    DeleteCertificateRequest.prototype.withSpAuthToken = function (spAuthToken) {
        this['Sp-Auth-Token'] = spAuthToken;
        return this;
    };
    Object.defineProperty(DeleteCertificateRequest.prototype, "spAuthToken", {
        get: function () {
            return this['Sp-Auth-Token'];
        },
        set: function (spAuthToken) {
            this['Sp-Auth-Token'] = spAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    DeleteCertificateRequest.prototype.withStageAuthToken = function (stageAuthToken) {
        this['Stage-Auth-Token'] = stageAuthToken;
        return this;
    };
    Object.defineProperty(DeleteCertificateRequest.prototype, "stageAuthToken", {
        get: function () {
            return this['Stage-Auth-Token'];
        },
        set: function (stageAuthToken) {
            this['Stage-Auth-Token'] = stageAuthToken;
        },
        enumerable: false,
        configurable: true
    });
    DeleteCertificateRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(DeleteCertificateRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    DeleteCertificateRequest.prototype.withCertificateId = function (certificateId) {
        this['certificate_id'] = certificateId;
        return this;
    };
    Object.defineProperty(DeleteCertificateRequest.prototype, "certificateId", {
        get: function () {
            return this['certificate_id'];
        },
        set: function (certificateId) {
            this['certificate_id'] = certificateId;
        },
        enumerable: false,
        configurable: true
    });
    return DeleteCertificateRequest;
}());
exports.DeleteCertificateRequest = DeleteCertificateRequest;
