// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceLinkageStatusCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceLinkageStatusCondition = void 0;
var DeviceLinkageStatusCondition = /** @class */ (function () {
    function DeviceLinkageStatusCondition() {
    }
    DeviceLinkageStatusCondition.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(DeviceLinkageStatusCondition.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceLinkageStatusCondition.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(DeviceLinkageStatusCondition.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceLinkageStatusCondition.prototype.withStatusList = function (statusList) {
        this['status_list'] = statusList;
        return this;
    };
    Object.defineProperty(DeviceLinkageStatusCondition.prototype, "statusList", {
        get: function () {
            return this['status_list'];
        },
        set: function (statusList) {
            this['status_list'] = statusList;
        },
        enumerable: false,
        configurable: true
    });
    DeviceLinkageStatusCondition.prototype.withDuration = function (duration) {
        this['duration'] = duration;
        return this;
    };
    return DeviceLinkageStatusCondition;
}());
exports.DeviceLinkageStatusCondition = DeviceLinkageStatusCondition;
