// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceGroupResponsSummery.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceGroupResponsSummery = void 0;
var DeviceGroupResponsSummery = /** @class */ (function () {
    function DeviceGroupResponsSummery() {
    }
    DeviceGroupResponsSummery.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(DeviceGroupResponsSummery.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceGroupResponsSummery.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    DeviceGroupResponsSummery.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    DeviceGroupResponsSummery.prototype.withSuperGroupId = function (superGroupId) {
        this['super_group_id'] = superGroupId;
        return this;
    };
    Object.defineProperty(DeviceGroupResponsSummery.prototype, "superGroupId", {
        get: function () {
            return this['super_group_id'];
        },
        set: function (superGroupId) {
            this['super_group_id'] = superGroupId;
        },
        enumerable: false,
        configurable: true
    });
    return DeviceGroupResponsSummery;
}());
exports.DeviceGroupResponsSummery = DeviceGroupResponsSummery;
