// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceMessage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceMessage = void 0;
var DeviceMessage = /** @class */ (function () {
    function DeviceMessage() {
    }
    DeviceMessage.prototype.withMessageId = function (messageId) {
        this['message_id'] = messageId;
        return this;
    };
    Object.defineProperty(DeviceMessage.prototype, "messageId", {
        get: function () {
            return this['message_id'];
        },
        set: function (messageId) {
            this['message_id'] = messageId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceMessage.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    DeviceMessage.prototype.withMessage = function (message) {
        this['message'] = message;
        return this;
    };
    DeviceMessage.prototype.withEncoding = function (encoding) {
        this['encoding'] = encoding;
        return this;
    };
    DeviceMessage.prototype.withPayloadFormat = function (payloadFormat) {
        this['payload_format'] = payloadFormat;
        return this;
    };
    Object.defineProperty(DeviceMessage.prototype, "payloadFormat", {
        get: function () {
            return this['payload_format'];
        },
        set: function (payloadFormat) {
            this['payload_format'] = payloadFormat;
        },
        enumerable: false,
        configurable: true
    });
    DeviceMessage.prototype.withTopic = function (topic) {
        this['topic'] = topic;
        return this;
    };
    DeviceMessage.prototype.withProperties = function (properties) {
        this['properties'] = properties;
        return this;
    };
    DeviceMessage.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    DeviceMessage.prototype.withErrorInfo = function (errorInfo) {
        this['error_info'] = errorInfo;
        return this;
    };
    Object.defineProperty(DeviceMessage.prototype, "errorInfo", {
        get: function () {
            return this['error_info'];
        },
        set: function (errorInfo) {
            this['error_info'] = errorInfo;
        },
        enumerable: false,
        configurable: true
    });
    DeviceMessage.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(DeviceMessage.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    DeviceMessage.prototype.withFinishedTime = function (finishedTime) {
        this['finished_time'] = finishedTime;
        return this;
    };
    Object.defineProperty(DeviceMessage.prototype, "finishedTime", {
        get: function () {
            return this['finished_time'];
        },
        set: function (finishedTime) {
            this['finished_time'] = finishedTime;
        },
        enumerable: false,
        configurable: true
    });
    return DeviceMessage;
}());
exports.DeviceMessage = DeviceMessage;
