// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceBroadcastRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceBroadcastRequest = void 0;
var DeviceBroadcastRequest = /** @class */ (function () {
    function DeviceBroadcastRequest(topicFullName, message) {
        this['topic_full_name'] = topicFullName;
        this['message'] = message;
    }
    DeviceBroadcastRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(DeviceBroadcastRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceBroadcastRequest.prototype.withTopicFullName = function (topicFullName) {
        this['topic_full_name'] = topicFullName;
        return this;
    };
    Object.defineProperty(DeviceBroadcastRequest.prototype, "topicFullName", {
        get: function () {
            return this['topic_full_name'];
        },
        set: function (topicFullName) {
            this['topic_full_name'] = topicFullName;
        },
        enumerable: false,
        configurable: true
    });
    DeviceBroadcastRequest.prototype.withMessage = function (message) {
        this['message'] = message;
        return this;
    };
    return DeviceBroadcastRequest;
}());
exports.DeviceBroadcastRequest = DeviceBroadcastRequest;
