// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceShadowData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceShadowData = void 0;
var DeviceShadowData = /** @class */ (function () {
    function DeviceShadowData(serviceId) {
        this['service_id'] = serviceId;
    }
    DeviceShadowData.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(DeviceShadowData.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceShadowData.prototype.withDesired = function (desired) {
        this['desired'] = desired;
        return this;
    };
    DeviceShadowData.prototype.withReported = function (reported) {
        this['reported'] = reported;
        return this;
    };
    DeviceShadowData.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    return DeviceShadowData;
}());
exports.DeviceShadowData = DeviceShadowData;
