// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceDataCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceDataCondition = void 0;
var DeviceDataCondition = /** @class */ (function () {
    function DeviceDataCondition() {
    }
    DeviceDataCondition.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(DeviceDataCondition.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceDataCondition.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(DeviceDataCondition.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceDataCondition.prototype.withFilters = function (filters) {
        this['filters'] = filters;
        return this;
    };
    return DeviceDataCondition;
}());
exports.DeviceDataCondition = DeviceDataCondition;
