// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DeviceCommandRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeviceCommandRequest = void 0;
var DeviceCommandRequest = /** @class */ (function () {
    function DeviceCommandRequest(paras) {
        this['paras'] = paras;
    }
    DeviceCommandRequest.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(DeviceCommandRequest.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    DeviceCommandRequest.prototype.withCommandName = function (commandName) {
        this['command_name'] = commandName;
        return this;
    };
    Object.defineProperty(DeviceCommandRequest.prototype, "commandName", {
        get: function () {
            return this['command_name'];
        },
        set: function (commandName) {
            this['command_name'] = commandName;
        },
        enumerable: false,
        configurable: true
    });
    DeviceCommandRequest.prototype.withParas = function (paras) {
        this['paras'] = paras;
        return this;
    };
    return DeviceCommandRequest;
}());
exports.DeviceCommandRequest = DeviceCommandRequest;
