// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DisForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisForwarding = void 0;
var DisForwarding = /** @class */ (function () {
    function DisForwarding(regionName, projectId) {
        this['region_name'] = regionName;
        this['project_id'] = projectId;
    }
    DisForwarding.prototype.withRegionName = function (regionName) {
        this['region_name'] = regionName;
        return this;
    };
    Object.defineProperty(DisForwarding.prototype, "regionName", {
        get: function () {
            return this['region_name'];
        },
        set: function (regionName) {
            this['region_name'] = regionName;
        },
        enumerable: false,
        configurable: true
    });
    DisForwarding.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(DisForwarding.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    DisForwarding.prototype.withStreamName = function (streamName) {
        this['stream_name'] = streamName;
        return this;
    };
    Object.defineProperty(DisForwarding.prototype, "streamName", {
        get: function () {
            return this['stream_name'];
        },
        set: function (streamName) {
            this['stream_name'] = streamName;
        },
        enumerable: false,
        configurable: true
    });
    DisForwarding.prototype.withStreamId = function (streamId) {
        this['stream_id'] = streamId;
        return this;
    };
    Object.defineProperty(DisForwarding.prototype, "streamId", {
        get: function () {
            return this['stream_id'];
        },
        set: function (streamId) {
            this['stream_id'] = streamId;
        },
        enumerable: false,
        configurable: true
    });
    return DisForwarding;
}());
exports.DisForwarding = DisForwarding;
