// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DmsKafkaForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DmsKafkaForwarding = void 0;
var DmsKafkaForwarding = /** @class */ (function () {
    function DmsKafkaForwarding(regionName, projectId, addresses, topic) {
        this['region_name'] = regionName;
        this['project_id'] = projectId;
        this['addresses'] = addresses;
        this['topic'] = topic;
    }
    DmsKafkaForwarding.prototype.withRegionName = function (regionName) {
        this['region_name'] = regionName;
        return this;
    };
    Object.defineProperty(DmsKafkaForwarding.prototype, "regionName", {
        get: function () {
            return this['region_name'];
        },
        set: function (regionName) {
            this['region_name'] = regionName;
        },
        enumerable: false,
        configurable: true
    });
    DmsKafkaForwarding.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(DmsKafkaForwarding.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    DmsKafkaForwarding.prototype.withAddresses = function (addresses) {
        this['addresses'] = addresses;
        return this;
    };
    DmsKafkaForwarding.prototype.withTopic = function (topic) {
        this['topic'] = topic;
        return this;
    };
    DmsKafkaForwarding.prototype.withUsername = function (username) {
        this['username'] = username;
        return this;
    };
    DmsKafkaForwarding.prototype.withPassword = function (password) {
        this['password'] = password;
        return this;
    };
    DmsKafkaForwarding.prototype.withMechanism = function (mechanism) {
        this['mechanism'] = mechanism;
        return this;
    };
    return DmsKafkaForwarding;
}());
exports.DmsKafkaForwarding = DmsKafkaForwarding;
