// @huaweicloud/huaweicloud-sdk-iotda/v5/model/DmsRocketMQForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DmsRocketMQForwarding = void 0;
var DmsRocketMQForwarding = /** @class */ (function () {
    function DmsRocketMQForwarding(addresses, topic, username, password) {
        this['addresses'] = addresses;
        this['topic'] = topic;
        this['username'] = username;
        this['password'] = password;
    }
    DmsRocketMQForwarding.prototype.withAddresses = function (addresses) {
        this['addresses'] = addresses;
        return this;
    };
    DmsRocketMQForwarding.prototype.withTopic = function (topic) {
        this['topic'] = topic;
        return this;
    };
    DmsRocketMQForwarding.prototype.withUsername = function (username) {
        this['username'] = username;
        return this;
    };
    DmsRocketMQForwarding.prototype.withPassword = function (password) {
        this['password'] = password;
        return this;
    };
    DmsRocketMQForwarding.prototype.withEnableSsl = function (enableSsl) {
        this['enable_ssl'] = enableSsl;
        return this;
    };
    Object.defineProperty(DmsRocketMQForwarding.prototype, "enableSsl", {
        get: function () {
            return this['enable_ssl'];
        },
        set: function (enableSsl) {
            this['enable_ssl'] = enableSsl;
        },
        enumerable: false,
        configurable: true
    });
    return DmsRocketMQForwarding;
}());
exports.DmsRocketMQForwarding = DmsRocketMQForwarding;
