// @huaweicloud/huaweicloud-sdk-iotda/v5/model/InfluxDBForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InfluxDBForwarding = void 0;
var InfluxDBForwarding = /** @class */ (function () {
    function InfluxDBForwarding(address, dbName, username, password, measurement, columnMappings) {
        this['address'] = address;
        this['db_name'] = dbName;
        this['username'] = username;
        this['password'] = password;
        this['measurement'] = measurement;
        this['column_mappings'] = columnMappings;
    }
    InfluxDBForwarding.prototype.withAddress = function (address) {
        this['address'] = address;
        return this;
    };
    InfluxDBForwarding.prototype.withDbName = function (dbName) {
        this['db_name'] = dbName;
        return this;
    };
    Object.defineProperty(InfluxDBForwarding.prototype, "dbName", {
        get: function () {
            return this['db_name'];
        },
        set: function (dbName) {
            this['db_name'] = dbName;
        },
        enumerable: false,
        configurable: true
    });
    InfluxDBForwarding.prototype.withUsername = function (username) {
        this['username'] = username;
        return this;
    };
    InfluxDBForwarding.prototype.withPassword = function (password) {
        this['password'] = password;
        return this;
    };
    InfluxDBForwarding.prototype.withMeasurement = function (measurement) {
        this['measurement'] = measurement;
        return this;
    };
    InfluxDBForwarding.prototype.withColumnMappings = function (columnMappings) {
        this['column_mappings'] = columnMappings;
        return this;
    };
    Object.defineProperty(InfluxDBForwarding.prototype, "columnMappings", {
        get: function () {
            return this['column_mappings'];
        },
        set: function (columnMappings) {
            this['column_mappings'] = columnMappings;
        },
        enumerable: false,
        configurable: true
    });
    return InfluxDBForwarding;
}());
exports.InfluxDBForwarding = InfluxDBForwarding;
