// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListDeviceGroupsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDeviceGroupsRequest = void 0;
var ListDeviceGroupsRequest = /** @class */ (function () {
    function ListDeviceGroupsRequest() {
    }
    ListDeviceGroupsRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListDeviceGroupsRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListDeviceGroupsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListDeviceGroupsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListDeviceGroupsRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListDeviceGroupsRequest.prototype.withLastModifiedTime = function (lastModifiedTime) {
        this['last_modified_time'] = lastModifiedTime;
        return this;
    };
    Object.defineProperty(ListDeviceGroupsRequest.prototype, "lastModifiedTime", {
        get: function () {
            return this['last_modified_time'];
        },
        set: function (lastModifiedTime) {
            this['last_modified_time'] = lastModifiedTime;
        },
        enumerable: false,
        configurable: true
    });
    ListDeviceGroupsRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListDeviceGroupsRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return ListDeviceGroupsRequest;
}());
exports.ListDeviceGroupsRequest = ListDeviceGroupsRequest;
