// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListRuleActionsResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRuleActionsResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListRuleActionsResponse = /** @class */ (function (_super) {
    __extends(ListRuleActionsResponse, _super);
    function ListRuleActionsResponse() {
        return _super.call(this) || this;
    }
    ListRuleActionsResponse.prototype.withActions = function (actions) {
        this['actions'] = actions;
        return this;
    };
    ListRuleActionsResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListRuleActionsResponse.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    return ListRuleActionsResponse;
}(SdkResponse_1.SdkResponse));
exports.ListRuleActionsResponse = ListRuleActionsResponse;
