// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListProductsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListProductsRequest = void 0;
var ListProductsRequest = /** @class */ (function () {
    function ListProductsRequest() {
    }
    ListProductsRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListProductsRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListProductsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListProductsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListProductsRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListProductsRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ListProductsRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ListProductsRequest;
}());
exports.ListProductsRequest = ListProductsRequest;
