// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListOtaPackageInfoRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListOtaPackageInfoRequest = void 0;
var ListOtaPackageInfoRequest = /** @class */ (function () {
    function ListOtaPackageInfoRequest(packageType) {
        this['package_type'] = packageType;
    }
    ListOtaPackageInfoRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListOtaPackageInfoRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListOtaPackageInfoRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListOtaPackageInfoRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ListOtaPackageInfoRequest.prototype.withPackageType = function (packageType) {
        this['package_type'] = packageType;
        return this;
    };
    Object.defineProperty(ListOtaPackageInfoRequest.prototype, "packageType", {
        get: function () {
            return this['package_type'];
        },
        set: function (packageType) {
            this['package_type'] = packageType;
        },
        enumerable: false,
        configurable: true
    });
    ListOtaPackageInfoRequest.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ListOtaPackageInfoRequest.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ListOtaPackageInfoRequest.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ListOtaPackageInfoRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListOtaPackageInfoRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListOtaPackageInfoRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ListOtaPackageInfoRequest;
}());
exports.ListOtaPackageInfoRequest = ListOtaPackageInfoRequest;
