// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListResourcesByTagsRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListResourcesByTagsRequest = void 0;
var ListResourcesByTagsRequest = /** @class */ (function () {
    function ListResourcesByTagsRequest() {
    }
    ListResourcesByTagsRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListResourcesByTagsRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListResourcesByTagsRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListResourcesByTagsRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListResourcesByTagsRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListResourcesByTagsRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return ListResourcesByTagsRequest;
}());
exports.ListResourcesByTagsRequest = ListResourcesByTagsRequest;
