// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListDevicesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListDevicesRequest = void 0;
var ListDevicesRequest = /** @class */ (function () {
    function ListDevicesRequest() {
    }
    ListDevicesRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withGatewayId = function (gatewayId) {
        this['gateway_id'] = gatewayId;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "gatewayId", {
        get: function () {
            return this['gateway_id'];
        },
        set: function (gatewayId) {
            this['gateway_id'] = gatewayId;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withIsCascadeQuery = function (isCascadeQuery) {
        this['is_cascade_query'] = isCascadeQuery;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "isCascadeQuery", {
        get: function () {
            return this['is_cascade_query'];
        },
        set: function (isCascadeQuery) {
            this['is_cascade_query'] = isCascadeQuery;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withNodeId = function (nodeId) {
        this['node_id'] = nodeId;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "nodeId", {
        get: function () {
            return this['node_id'];
        },
        set: function (nodeId) {
            this['node_id'] = nodeId;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListDevicesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListDevicesRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    ListDevicesRequest.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    ListDevicesRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListDevicesRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return ListDevicesRequest;
}());
exports.ListDevicesRequest = ListDevicesRequest;
