// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListPropertiesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListPropertiesRequest = void 0;
var ListPropertiesRequest = /** @class */ (function () {
    function ListPropertiesRequest(deviceId, serviceId) {
        this['device_id'] = deviceId;
        this['service_id'] = serviceId;
    }
    ListPropertiesRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ListPropertiesRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ListPropertiesRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListPropertiesRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListPropertiesRequest.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(ListPropertiesRequest.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    return ListPropertiesRequest;
}());
exports.ListPropertiesRequest = ListPropertiesRequest;
