// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListRulesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRulesRequest = void 0;
var ListRulesRequest = /** @class */ (function () {
    function ListRulesRequest() {
    }
    ListRulesRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListRulesRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListRulesRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListRulesRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ListRulesRequest.prototype.withRuleType = function (ruleType) {
        this['rule_type'] = ruleType;
        return this;
    };
    Object.defineProperty(ListRulesRequest.prototype, "ruleType", {
        get: function () {
            return this['rule_type'];
        },
        set: function (ruleType) {
            this['rule_type'] = ruleType;
        },
        enumerable: false,
        configurable: true
    });
    ListRulesRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListRulesRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListRulesRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ListRulesRequest;
}());
exports.ListRulesRequest = ListRulesRequest;
