// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListBatchTasksRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListBatchTasksRequest = void 0;
var ListBatchTasksRequest = /** @class */ (function () {
    function ListBatchTasksRequest(taskType) {
        this['task_type'] = taskType;
    }
    ListBatchTasksRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ListBatchTasksRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ListBatchTasksRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ListBatchTasksRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ListBatchTasksRequest.prototype.withTaskType = function (taskType) {
        this['task_type'] = taskType;
        return this;
    };
    Object.defineProperty(ListBatchTasksRequest.prototype, "taskType", {
        get: function () {
            return this['task_type'];
        },
        set: function (taskType) {
            this['task_type'] = taskType;
        },
        enumerable: false,
        configurable: true
    });
    ListBatchTasksRequest.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    ListBatchTasksRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ListBatchTasksRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListBatchTasksRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ListBatchTasksRequest;
}());
exports.ListBatchTasksRequest = ListBatchTasksRequest;
