// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ListRulesResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListRulesResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ListRulesResponse = /** @class */ (function (_super) {
    __extends(ListRulesResponse, _super);
    function ListRulesResponse() {
        return _super.call(this) || this;
    }
    ListRulesResponse.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ListRulesResponse.prototype.withCount = function (count) {
        this['count'] = count;
        return this;
    };
    ListRulesResponse.prototype.withRules = function (rules) {
        this['rules'] = rules;
        return this;
    };
    return ListRulesResponse;
}(SdkResponse_1.SdkResponse));
exports.ListRulesResponse = ListRulesResponse;
