// @huaweicloud/huaweicloud-sdk-iotda/v5/model/MessageResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MessageResult = void 0;
var MessageResult = /** @class */ (function () {
    function MessageResult() {
    }
    MessageResult.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    MessageResult.prototype.withCreatedTime = function (createdTime) {
        this['created_time'] = createdTime;
        return this;
    };
    Object.defineProperty(MessageResult.prototype, "createdTime", {
        get: function () {
            return this['created_time'];
        },
        set: function (createdTime) {
            this['created_time'] = createdTime;
        },
        enumerable: false,
        configurable: true
    });
    MessageResult.prototype.withFinishedTime = function (finishedTime) {
        this['finished_time'] = finishedTime;
        return this;
    };
    Object.defineProperty(MessageResult.prototype, "finishedTime", {
        get: function () {
            return this['finished_time'];
        },
        set: function (finishedTime) {
            this['finished_time'] = finishedTime;
        },
        enumerable: false,
        configurable: true
    });
    return MessageResult;
}());
exports.MessageResult = MessageResult;
