// @huaweicloud/huaweicloud-sdk-iotda/v5/model/MrsKafkaForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MrsKafkaForwarding = void 0;
var MrsKafkaForwarding = /** @class */ (function () {
    function MrsKafkaForwarding(addresses, topic) {
        this['addresses'] = addresses;
        this['topic'] = topic;
    }
    MrsKafkaForwarding.prototype.withAddresses = function (addresses) {
        this['addresses'] = addresses;
        return this;
    };
    MrsKafkaForwarding.prototype.withTopic = function (topic) {
        this['topic'] = topic;
        return this;
    };
    MrsKafkaForwarding.prototype.withKerberosAuthentication = function (kerberosAuthentication) {
        this['kerberos_authentication'] = kerberosAuthentication;
        return this;
    };
    Object.defineProperty(MrsKafkaForwarding.prototype, "kerberosAuthentication", {
        get: function () {
            return this['kerberos_authentication'];
        },
        set: function (kerberosAuthentication) {
            this['kerberos_authentication'] = kerberosAuthentication;
        },
        enumerable: false,
        configurable: true
    });
    return MrsKafkaForwarding;
}());
exports.MrsKafkaForwarding = MrsKafkaForwarding;
