// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ObsForwarding.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObsForwarding = void 0;
var ObsForwarding = /** @class */ (function () {
    function ObsForwarding(regionName, projectId, bucketName) {
        this['region_name'] = regionName;
        this['project_id'] = projectId;
        this['bucket_name'] = bucketName;
    }
    ObsForwarding.prototype.withRegionName = function (regionName) {
        this['region_name'] = regionName;
        return this;
    };
    Object.defineProperty(ObsForwarding.prototype, "regionName", {
        get: function () {
            return this['region_name'];
        },
        set: function (regionName) {
            this['region_name'] = regionName;
        },
        enumerable: false,
        configurable: true
    });
    ObsForwarding.prototype.withProjectId = function (projectId) {
        this['project_id'] = projectId;
        return this;
    };
    Object.defineProperty(ObsForwarding.prototype, "projectId", {
        get: function () {
            return this['project_id'];
        },
        set: function (projectId) {
            this['project_id'] = projectId;
        },
        enumerable: false,
        configurable: true
    });
    ObsForwarding.prototype.withBucketName = function (bucketName) {
        this['bucket_name'] = bucketName;
        return this;
    };
    Object.defineProperty(ObsForwarding.prototype, "bucketName", {
        get: function () {
            return this['bucket_name'];
        },
        set: function (bucketName) {
            this['bucket_name'] = bucketName;
        },
        enumerable: false,
        configurable: true
    });
    ObsForwarding.prototype.withLocation = function (location) {
        this['location'] = location;
        return this;
    };
    ObsForwarding.prototype.withFilePath = function (filePath) {
        this['file_path'] = filePath;
        return this;
    };
    Object.defineProperty(ObsForwarding.prototype, "filePath", {
        get: function () {
            return this['file_path'];
        },
        set: function (filePath) {
            this['file_path'] = filePath;
        },
        enumerable: false,
        configurable: true
    });
    return ObsForwarding;
}());
exports.ObsForwarding = ObsForwarding;
