// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ObsLocation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ObsLocation = void 0;
var ObsLocation = /** @class */ (function () {
    function ObsLocation(regionName, bucketName, objectKey) {
        this['region_name'] = regionName;
        this['bucket_name'] = bucketName;
        this['object_key'] = objectKey;
    }
    ObsLocation.prototype.withRegionName = function (regionName) {
        this['region_name'] = regionName;
        return this;
    };
    Object.defineProperty(ObsLocation.prototype, "regionName", {
        get: function () {
            return this['region_name'];
        },
        set: function (regionName) {
            this['region_name'] = regionName;
        },
        enumerable: false,
        configurable: true
    });
    ObsLocation.prototype.withBucketName = function (bucketName) {
        this['bucket_name'] = bucketName;
        return this;
    };
    Object.defineProperty(ObsLocation.prototype, "bucketName", {
        get: function () {
            return this['bucket_name'];
        },
        set: function (bucketName) {
            this['bucket_name'] = bucketName;
        },
        enumerable: false,
        configurable: true
    });
    ObsLocation.prototype.withObjectKey = function (objectKey) {
        this['object_key'] = objectKey;
        return this;
    };
    Object.defineProperty(ObsLocation.prototype, "objectKey", {
        get: function () {
            return this['object_key'];
        },
        set: function (objectKey) {
            this['object_key'] = objectKey;
        },
        enumerable: false,
        configurable: true
    });
    return ObsLocation;
}());
exports.ObsLocation = ObsLocation;
