// @huaweicloud/huaweicloud-sdk-iotda/v5/model/OtaPackageInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtaPackageInfo = void 0;
var OtaPackageInfo = /** @class */ (function () {
    function OtaPackageInfo() {
    }
    OtaPackageInfo.prototype.withPackageId = function (packageId) {
        this['package_id'] = packageId;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "packageId", {
        get: function () {
            return this['package_id'];
        },
        set: function (packageId) {
            this['package_id'] = packageId;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withPackageType = function (packageType) {
        this['package_type'] = packageType;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "packageType", {
        get: function () {
            return this['package_type'];
        },
        set: function (packageType) {
            this['package_type'] = packageType;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    OtaPackageInfo.prototype.withSupportSourceVersions = function (supportSourceVersions) {
        this['support_source_versions'] = supportSourceVersions;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "supportSourceVersions", {
        get: function () {
            return this['support_source_versions'];
        },
        set: function (supportSourceVersions) {
            this['support_source_versions'] = supportSourceVersions;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    OtaPackageInfo.prototype.withCustomInfo = function (customInfo) {
        this['custom_info'] = customInfo;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "customInfo", {
        get: function () {
            return this['custom_info'];
        },
        set: function (customInfo) {
            this['custom_info'] = customInfo;
        },
        enumerable: false,
        configurable: true
    });
    OtaPackageInfo.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(OtaPackageInfo.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    return OtaPackageInfo;
}());
exports.OtaPackageInfo = OtaPackageInfo;
