// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ProductSummary.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ProductSummary = void 0;
var ProductSummary = /** @class */ (function () {
    function ProductSummary() {
    }
    ProductSummary.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ProductSummary.prototype.withDeviceType = function (deviceType) {
        this['device_type'] = deviceType;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "deviceType", {
        get: function () {
            return this['device_type'];
        },
        set: function (deviceType) {
            this['device_type'] = deviceType;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withProtocolType = function (protocolType) {
        this['protocol_type'] = protocolType;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "protocolType", {
        get: function () {
            return this['protocol_type'];
        },
        set: function (protocolType) {
            this['protocol_type'] = protocolType;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withDataFormat = function (dataFormat) {
        this['data_format'] = dataFormat;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "dataFormat", {
        get: function () {
            return this['data_format'];
        },
        set: function (dataFormat) {
            this['data_format'] = dataFormat;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withManufacturerName = function (manufacturerName) {
        this['manufacturer_name'] = manufacturerName;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "manufacturerName", {
        get: function () {
            return this['manufacturer_name'];
        },
        set: function (manufacturerName) {
            this['manufacturer_name'] = manufacturerName;
        },
        enumerable: false,
        configurable: true
    });
    ProductSummary.prototype.withIndustry = function (industry) {
        this['industry'] = industry;
        return this;
    };
    ProductSummary.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ProductSummary.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(ProductSummary.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    return ProductSummary;
}());
exports.ProductSummary = ProductSummary;
