// @huaweicloud/huaweicloud-sdk-iotda/v5/model/PropertyFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyFilter = void 0;
var PropertyFilter = /** @class */ (function () {
    function PropertyFilter(path, operator) {
        this['path'] = path;
        this['operator'] = operator;
    }
    PropertyFilter.prototype.withPath = function (path) {
        this['path'] = path;
        return this;
    };
    PropertyFilter.prototype.withOperator = function (operator) {
        this['operator'] = operator;
        return this;
    };
    PropertyFilter.prototype.withValue = function (value) {
        this['value'] = value;
        return this;
    };
    PropertyFilter.prototype.withInValues = function (inValues) {
        this['in_values'] = inValues;
        return this;
    };
    Object.defineProperty(PropertyFilter.prototype, "inValues", {
        get: function () {
            return this['in_values'];
        },
        set: function (inValues) {
            this['in_values'] = inValues;
        },
        enumerable: false,
        configurable: true
    });
    PropertyFilter.prototype.withStrategy = function (strategy) {
        this['strategy'] = strategy;
        return this;
    };
    return PropertyFilter;
}());
exports.PropertyFilter = PropertyFilter;
