// @huaweicloud/huaweicloud-sdk-iotda/v5/model/PropertiesDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertiesDTO = void 0;
var PropertiesDTO = /** @class */ (function () {
    function PropertiesDTO() {
    }
    PropertiesDTO.prototype.withCorrelationData = function (correlationData) {
        this['correlation_data'] = correlationData;
        return this;
    };
    Object.defineProperty(PropertiesDTO.prototype, "correlationData", {
        get: function () {
            return this['correlation_data'];
        },
        set: function (correlationData) {
            this['correlation_data'] = correlationData;
        },
        enumerable: false,
        configurable: true
    });
    PropertiesDTO.prototype.withResponseTopic = function (responseTopic) {
        this['response_topic'] = responseTopic;
        return this;
    };
    Object.defineProperty(PropertiesDTO.prototype, "responseTopic", {
        get: function () {
            return this['response_topic'];
        },
        set: function (responseTopic) {
            this['response_topic'] = responseTopic;
        },
        enumerable: false,
        configurable: true
    });
    PropertiesDTO.prototype.withUserProperties = function (userProperties) {
        this['user_properties'] = userProperties;
        return this;
    };
    Object.defineProperty(PropertiesDTO.prototype, "userProperties", {
        get: function () {
            return this['user_properties'];
        },
        set: function (userProperties) {
            this['user_properties'] = userProperties;
        },
        enumerable: false,
        configurable: true
    });
    return PropertiesDTO;
}());
exports.PropertiesDTO = PropertiesDTO;
