// @huaweicloud/huaweicloud-sdk-iotda/v5/model/QueryQueueBase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueryQueueBase = void 0;
var QueryQueueBase = /** @class */ (function () {
    function QueryQueueBase() {
    }
    QueryQueueBase.prototype.withQueueId = function (queueId) {
        this['queue_id'] = queueId;
        return this;
    };
    Object.defineProperty(QueryQueueBase.prototype, "queueId", {
        get: function () {
            return this['queue_id'];
        },
        set: function (queueId) {
            this['queue_id'] = queueId;
        },
        enumerable: false,
        configurable: true
    });
    QueryQueueBase.prototype.withQueueName = function (queueName) {
        this['queue_name'] = queueName;
        return this;
    };
    Object.defineProperty(QueryQueueBase.prototype, "queueName", {
        get: function () {
            return this['queue_name'];
        },
        set: function (queueName) {
            this['queue_name'] = queueName;
        },
        enumerable: false,
        configurable: true
    });
    QueryQueueBase.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(QueryQueueBase.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    QueryQueueBase.prototype.withLastModifyTime = function (lastModifyTime) {
        this['last_modify_time'] = lastModifyTime;
        return this;
    };
    Object.defineProperty(QueryQueueBase.prototype, "lastModifyTime", {
        get: function () {
            return this['last_modify_time'];
        },
        set: function (lastModifyTime) {
            this['last_modify_time'] = lastModifyTime;
        },
        enumerable: false,
        configurable: true
    });
    return QueryQueueBase;
}());
exports.QueryQueueBase = QueryQueueBase;
