// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ResetDeviceSecretRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetDeviceSecretRequest = void 0;
var ResetDeviceSecretRequest = /** @class */ (function () {
    function ResetDeviceSecretRequest(deviceId, actionId) {
        this['device_id'] = deviceId;
        this['action_id'] = actionId;
    }
    ResetDeviceSecretRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ResetDeviceSecretRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ResetDeviceSecretRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ResetDeviceSecretRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ResetDeviceSecretRequest.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(ResetDeviceSecretRequest.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    ResetDeviceSecretRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return ResetDeviceSecretRequest;
}());
exports.ResetDeviceSecretRequest = ResetDeviceSecretRequest;
