// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ResetFingerprintRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ResetFingerprintRequest = void 0;
var ResetFingerprintRequest = /** @class */ (function () {
    function ResetFingerprintRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    ResetFingerprintRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ResetFingerprintRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ResetFingerprintRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ResetFingerprintRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ResetFingerprintRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return ResetFingerprintRequest;
}());
exports.ResetFingerprintRequest = ResetFingerprintRequest;
