// @huaweicloud/huaweicloud-sdk-iotda/v5/model/RoutingRuleAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingRuleAction = void 0;
var RoutingRuleAction = /** @class */ (function () {
    function RoutingRuleAction() {
    }
    RoutingRuleAction.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(RoutingRuleAction.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRuleAction.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(RoutingRuleAction.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRuleAction.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(RoutingRuleAction.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRuleAction.prototype.withChannel = function (channel) {
        this['channel'] = channel;
        return this;
    };
    RoutingRuleAction.prototype.withChannelDetail = function (channelDetail) {
        this['channel_detail'] = channelDetail;
        return this;
    };
    Object.defineProperty(RoutingRuleAction.prototype, "channelDetail", {
        get: function () {
            return this['channel_detail'];
        },
        set: function (channelDetail) {
            this['channel_detail'] = channelDetail;
        },
        enumerable: false,
        configurable: true
    });
    return RoutingRuleAction;
}());
exports.RoutingRuleAction = RoutingRuleAction;
