// @huaweicloud/huaweicloud-sdk-iotda/v5/model/RoutingRule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RoutingRule = void 0;
var RoutingRule = /** @class */ (function () {
    function RoutingRule() {
    }
    RoutingRule.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(RoutingRule.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRule.prototype.withRuleName = function (ruleName) {
        this['rule_name'] = ruleName;
        return this;
    };
    Object.defineProperty(RoutingRule.prototype, "ruleName", {
        get: function () {
            return this['rule_name'];
        },
        set: function (ruleName) {
            this['rule_name'] = ruleName;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    RoutingRule.prototype.withSubject = function (subject) {
        this['subject'] = subject;
        return this;
    };
    RoutingRule.prototype.withAppType = function (appType) {
        this['app_type'] = appType;
        return this;
    };
    Object.defineProperty(RoutingRule.prototype, "appType", {
        get: function () {
            return this['app_type'];
        },
        set: function (appType) {
            this['app_type'] = appType;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRule.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(RoutingRule.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    RoutingRule.prototype.withSelect = function (select) {
        this['select'] = select;
        return this;
    };
    RoutingRule.prototype.withWhere = function (where) {
        this['where'] = where;
        return this;
    };
    RoutingRule.prototype.withActive = function (active) {
        this['active'] = active;
        return this;
    };
    return RoutingRule;
}());
exports.RoutingRule = RoutingRule;
