// @huaweicloud/huaweicloud-sdk-iotda/v5/model/RuleAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleAction = void 0;
var RuleAction = /** @class */ (function () {
    function RuleAction(type) {
        this['type'] = type;
    }
    RuleAction.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    RuleAction.prototype.withDeviceCommand = function (deviceCommand) {
        this['device_command'] = deviceCommand;
        return this;
    };
    Object.defineProperty(RuleAction.prototype, "deviceCommand", {
        get: function () {
            return this['device_command'];
        },
        set: function (deviceCommand) {
            this['device_command'] = deviceCommand;
        },
        enumerable: false,
        configurable: true
    });
    RuleAction.prototype.withSmnForwarding = function (smnForwarding) {
        this['smn_forwarding'] = smnForwarding;
        return this;
    };
    Object.defineProperty(RuleAction.prototype, "smnForwarding", {
        get: function () {
            return this['smn_forwarding'];
        },
        set: function (smnForwarding) {
            this['smn_forwarding'] = smnForwarding;
        },
        enumerable: false,
        configurable: true
    });
    RuleAction.prototype.withDeviceAlarm = function (deviceAlarm) {
        this['device_alarm'] = deviceAlarm;
        return this;
    };
    Object.defineProperty(RuleAction.prototype, "deviceAlarm", {
        get: function () {
            return this['device_alarm'];
        },
        set: function (deviceAlarm) {
            this['device_alarm'] = deviceAlarm;
        },
        enumerable: false,
        configurable: true
    });
    return RuleAction;
}());
exports.RuleAction = RuleAction;
