// @huaweicloud/huaweicloud-sdk-iotda/v5/model/RuleResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleResponse = void 0;
var RuleResponse = /** @class */ (function () {
    function RuleResponse(name, conditionGroup, actions, ruleType) {
        this['name'] = name;
        this['condition_group'] = conditionGroup;
        this['actions'] = actions;
        this['rule_type'] = ruleType;
    }
    RuleResponse.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    RuleResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    RuleResponse.prototype.withConditionGroup = function (conditionGroup) {
        this['condition_group'] = conditionGroup;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "conditionGroup", {
        get: function () {
            return this['condition_group'];
        },
        set: function (conditionGroup) {
            this['condition_group'] = conditionGroup;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withActions = function (actions) {
        this['actions'] = actions;
        return this;
    };
    RuleResponse.prototype.withRuleType = function (ruleType) {
        this['rule_type'] = ruleType;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "ruleType", {
        get: function () {
            return this['rule_type'];
        },
        set: function (ruleType) {
            this['rule_type'] = ruleType;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    RuleResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withEdgeNodeIds = function (edgeNodeIds) {
        this['edge_node_ids'] = edgeNodeIds;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "edgeNodeIds", {
        get: function () {
            return this['edge_node_ids'];
        },
        set: function (edgeNodeIds) {
            this['edge_node_ids'] = edgeNodeIds;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withLastUpdateTime = function (lastUpdateTime) {
        this['last_update_time'] = lastUpdateTime;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "lastUpdateTime", {
        get: function () {
            return this['last_update_time'];
        },
        set: function (lastUpdateTime) {
            this['last_update_time'] = lastUpdateTime;
        },
        enumerable: false,
        configurable: true
    });
    RuleResponse.prototype.withDeviceSide = function (deviceSide) {
        this['device_side'] = deviceSide;
        return this;
    };
    Object.defineProperty(RuleResponse.prototype, "deviceSide", {
        get: function () {
            return this['device_side'];
        },
        set: function (deviceSide) {
            this['device_side'] = deviceSide;
        },
        enumerable: false,
        configurable: true
    });
    return RuleResponse;
}());
exports.RuleResponse = RuleResponse;
