// @huaweicloud/huaweicloud-sdk-iotda/v5/model/RuleCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RuleCondition = void 0;
var RuleCondition = /** @class */ (function () {
    function RuleCondition(type) {
        this['type'] = type;
    }
    RuleCondition.prototype.withType = function (type) {
        this['type'] = type;
        return this;
    };
    RuleCondition.prototype.withDevicePropertyCondition = function (devicePropertyCondition) {
        this['device_property_condition'] = devicePropertyCondition;
        return this;
    };
    Object.defineProperty(RuleCondition.prototype, "devicePropertyCondition", {
        get: function () {
            return this['device_property_condition'];
        },
        set: function (devicePropertyCondition) {
            this['device_property_condition'] = devicePropertyCondition;
        },
        enumerable: false,
        configurable: true
    });
    RuleCondition.prototype.withSimpleTimerCondition = function (simpleTimerCondition) {
        this['simple_timer_condition'] = simpleTimerCondition;
        return this;
    };
    Object.defineProperty(RuleCondition.prototype, "simpleTimerCondition", {
        get: function () {
            return this['simple_timer_condition'];
        },
        set: function (simpleTimerCondition) {
            this['simple_timer_condition'] = simpleTimerCondition;
        },
        enumerable: false,
        configurable: true
    });
    RuleCondition.prototype.withDailyTimerCondition = function (dailyTimerCondition) {
        this['daily_timer_condition'] = dailyTimerCondition;
        return this;
    };
    Object.defineProperty(RuleCondition.prototype, "dailyTimerCondition", {
        get: function () {
            return this['daily_timer_condition'];
        },
        set: function (dailyTimerCondition) {
            this['daily_timer_condition'] = dailyTimerCondition;
        },
        enumerable: false,
        configurable: true
    });
    RuleCondition.prototype.withDeviceLinkageStatusCondition = function (deviceLinkageStatusCondition) {
        this['device_linkage_status_condition'] = deviceLinkageStatusCondition;
        return this;
    };
    Object.defineProperty(RuleCondition.prototype, "deviceLinkageStatusCondition", {
        get: function () {
            return this['device_linkage_status_condition'];
        },
        set: function (deviceLinkageStatusCondition) {
            this['device_linkage_status_condition'] = deviceLinkageStatusCondition;
        },
        enumerable: false,
        configurable: true
    });
    return RuleCondition;
}());
exports.RuleCondition = RuleCondition;
