// @huaweicloud/huaweicloud-sdk-iotda/v5/model/Rule.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Rule = void 0;
var Rule = /** @class */ (function () {
    function Rule(name, conditionGroup, actions, ruleType) {
        this['name'] = name;
        this['condition_group'] = conditionGroup;
        this['actions'] = actions;
        this['rule_type'] = ruleType;
    }
    Rule.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    Rule.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    Rule.prototype.withConditionGroup = function (conditionGroup) {
        this['condition_group'] = conditionGroup;
        return this;
    };
    Object.defineProperty(Rule.prototype, "conditionGroup", {
        get: function () {
            return this['condition_group'];
        },
        set: function (conditionGroup) {
            this['condition_group'] = conditionGroup;
        },
        enumerable: false,
        configurable: true
    });
    Rule.prototype.withActions = function (actions) {
        this['actions'] = actions;
        return this;
    };
    Rule.prototype.withRuleType = function (ruleType) {
        this['rule_type'] = ruleType;
        return this;
    };
    Object.defineProperty(Rule.prototype, "ruleType", {
        get: function () {
            return this['rule_type'];
        },
        set: function (ruleType) {
            this['rule_type'] = ruleType;
        },
        enumerable: false,
        configurable: true
    });
    Rule.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    Rule.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(Rule.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    Rule.prototype.withDeviceSide = function (deviceSide) {
        this['device_side'] = deviceSide;
        return this;
    };
    Object.defineProperty(Rule.prototype, "deviceSide", {
        get: function () {
            return this['device_side'];
        },
        set: function (deviceSide) {
            this['device_side'] = deviceSide;
        },
        enumerable: false,
        configurable: true
    });
    return Rule;
}());
exports.Rule = Rule;
