// @huaweicloud/huaweicloud-sdk-iotda/v5/model/SearchDevice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchDevice = void 0;
var SearchDevice = /** @class */ (function () {
    function SearchDevice() {
    }
    SearchDevice.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withNodeId = function (nodeId) {
        this['node_id'] = nodeId;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "nodeId", {
        get: function () {
            return this['node_id'];
        },
        set: function (nodeId) {
            this['node_id'] = nodeId;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withGatewayId = function (gatewayId) {
        this['gateway_id'] = gatewayId;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "gatewayId", {
        get: function () {
            return this['gateway_id'];
        },
        set: function (gatewayId) {
            this['gateway_id'] = gatewayId;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withNodeType = function (nodeType) {
        this['node_type'] = nodeType;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "nodeType", {
        get: function () {
            return this['node_type'];
        },
        set: function (nodeType) {
            this['node_type'] = nodeType;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withFwVersion = function (fwVersion) {
        this['fw_version'] = fwVersion;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "fwVersion", {
        get: function () {
            return this['fw_version'];
        },
        set: function (fwVersion) {
            this['fw_version'] = fwVersion;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withSwVersion = function (swVersion) {
        this['sw_version'] = swVersion;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "swVersion", {
        get: function () {
            return this['sw_version'];
        },
        set: function (swVersion) {
            this['sw_version'] = swVersion;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withDeviceSdkVersion = function (deviceSdkVersion) {
        this['device_sdk_version'] = deviceSdkVersion;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "deviceSdkVersion", {
        get: function () {
            return this['device_sdk_version'];
        },
        set: function (deviceSdkVersion) {
            this['device_sdk_version'] = deviceSdkVersion;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withProductName = function (productName) {
        this['product_name'] = productName;
        return this;
    };
    Object.defineProperty(SearchDevice.prototype, "productName", {
        get: function () {
            return this['product_name'];
        },
        set: function (productName) {
            this['product_name'] = productName;
        },
        enumerable: false,
        configurable: true
    });
    SearchDevice.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    SearchDevice.prototype.withTags = function (tags) {
        this['tags'] = tags;
        return this;
    };
    SearchDevice.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    return SearchDevice;
}());
exports.SearchDevice = SearchDevice;
