// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ServiceCommandPara.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceCommandPara = void 0;
var ServiceCommandPara = /** @class */ (function () {
    function ServiceCommandPara(paraName, dataType) {
        this['para_name'] = paraName;
        this['data_type'] = dataType;
    }
    ServiceCommandPara.prototype.withParaName = function (paraName) {
        this['para_name'] = paraName;
        return this;
    };
    Object.defineProperty(ServiceCommandPara.prototype, "paraName", {
        get: function () {
            return this['para_name'];
        },
        set: function (paraName) {
            this['para_name'] = paraName;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCommandPara.prototype.withDataType = function (dataType) {
        this['data_type'] = dataType;
        return this;
    };
    Object.defineProperty(ServiceCommandPara.prototype, "dataType", {
        get: function () {
            return this['data_type'];
        },
        set: function (dataType) {
            this['data_type'] = dataType;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCommandPara.prototype.withRequired = function (required) {
        this['required'] = required;
        return this;
    };
    ServiceCommandPara.prototype.withEnumList = function (enumList) {
        this['enum_list'] = enumList;
        return this;
    };
    Object.defineProperty(ServiceCommandPara.prototype, "enumList", {
        get: function () {
            return this['enum_list'];
        },
        set: function (enumList) {
            this['enum_list'] = enumList;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCommandPara.prototype.withMin = function (min) {
        this['min'] = min;
        return this;
    };
    ServiceCommandPara.prototype.withMax = function (max) {
        this['max'] = max;
        return this;
    };
    ServiceCommandPara.prototype.withMaxLength = function (maxLength) {
        this['max_length'] = maxLength;
        return this;
    };
    Object.defineProperty(ServiceCommandPara.prototype, "maxLength", {
        get: function () {
            return this['max_length'];
        },
        set: function (maxLength) {
            this['max_length'] = maxLength;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCommandPara.prototype.withStep = function (step) {
        this['step'] = step;
        return this;
    };
    ServiceCommandPara.prototype.withUnit = function (unit) {
        this['unit'] = unit;
        return this;
    };
    ServiceCommandPara.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return ServiceCommandPara;
}());
exports.ServiceCommandPara = ServiceCommandPara;
