// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ServiceCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceCommand = void 0;
var ServiceCommand = /** @class */ (function () {
    function ServiceCommand(commandName) {
        this['command_name'] = commandName;
    }
    ServiceCommand.prototype.withCommandName = function (commandName) {
        this['command_name'] = commandName;
        return this;
    };
    Object.defineProperty(ServiceCommand.prototype, "commandName", {
        get: function () {
            return this['command_name'];
        },
        set: function (commandName) {
            this['command_name'] = commandName;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCommand.prototype.withParas = function (paras) {
        this['paras'] = paras;
        return this;
    };
    ServiceCommand.prototype.withResponses = function (responses) {
        this['responses'] = responses;
        return this;
    };
    return ServiceCommand;
}());
exports.ServiceCommand = ServiceCommand;
