// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ServiceProperty.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceProperty = void 0;
var ServiceProperty = /** @class */ (function () {
    function ServiceProperty(propertyName, dataType, method) {
        this['property_name'] = propertyName;
        this['data_type'] = dataType;
        this['method'] = method;
    }
    ServiceProperty.prototype.withPropertyName = function (propertyName) {
        this['property_name'] = propertyName;
        return this;
    };
    Object.defineProperty(ServiceProperty.prototype, "propertyName", {
        get: function () {
            return this['property_name'];
        },
        set: function (propertyName) {
            this['property_name'] = propertyName;
        },
        enumerable: false,
        configurable: true
    });
    ServiceProperty.prototype.withDataType = function (dataType) {
        this['data_type'] = dataType;
        return this;
    };
    Object.defineProperty(ServiceProperty.prototype, "dataType", {
        get: function () {
            return this['data_type'];
        },
        set: function (dataType) {
            this['data_type'] = dataType;
        },
        enumerable: false,
        configurable: true
    });
    ServiceProperty.prototype.withRequired = function (required) {
        this['required'] = required;
        return this;
    };
    ServiceProperty.prototype.withEnumList = function (enumList) {
        this['enum_list'] = enumList;
        return this;
    };
    Object.defineProperty(ServiceProperty.prototype, "enumList", {
        get: function () {
            return this['enum_list'];
        },
        set: function (enumList) {
            this['enum_list'] = enumList;
        },
        enumerable: false,
        configurable: true
    });
    ServiceProperty.prototype.withMin = function (min) {
        this['min'] = min;
        return this;
    };
    ServiceProperty.prototype.withMax = function (max) {
        this['max'] = max;
        return this;
    };
    ServiceProperty.prototype.withMaxLength = function (maxLength) {
        this['max_length'] = maxLength;
        return this;
    };
    Object.defineProperty(ServiceProperty.prototype, "maxLength", {
        get: function () {
            return this['max_length'];
        },
        set: function (maxLength) {
            this['max_length'] = maxLength;
        },
        enumerable: false,
        configurable: true
    });
    ServiceProperty.prototype.withStep = function (step) {
        this['step'] = step;
        return this;
    };
    ServiceProperty.prototype.withUnit = function (unit) {
        this['unit'] = unit;
        return this;
    };
    ServiceProperty.prototype.withMethod = function (method) {
        this['method'] = method;
        return this;
    };
    ServiceProperty.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ServiceProperty.prototype.withDefaultValue = function (defaultValue) {
        this['default_value'] = defaultValue;
        return this;
    };
    Object.defineProperty(ServiceProperty.prototype, "defaultValue", {
        get: function () {
            return this['default_value'];
        },
        set: function (defaultValue) {
            this['default_value'] = defaultValue;
        },
        enumerable: false,
        configurable: true
    });
    return ServiceProperty;
}());
exports.ServiceProperty = ServiceProperty;
