// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ServiceCapability.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServiceCapability = void 0;
var ServiceCapability = /** @class */ (function () {
    function ServiceCapability(serviceId, serviceType) {
        this['service_id'] = serviceId;
        this['service_type'] = serviceType;
    }
    ServiceCapability.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(ServiceCapability.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCapability.prototype.withServiceType = function (serviceType) {
        this['service_type'] = serviceType;
        return this;
    };
    Object.defineProperty(ServiceCapability.prototype, "serviceType", {
        get: function () {
            return this['service_type'];
        },
        set: function (serviceType) {
            this['service_type'] = serviceType;
        },
        enumerable: false,
        configurable: true
    });
    ServiceCapability.prototype.withProperties = function (properties) {
        this['properties'] = properties;
        return this;
    };
    ServiceCapability.prototype.withCommands = function (commands) {
        this['commands'] = commands;
        return this;
    };
    ServiceCapability.prototype.withEvents = function (events) {
        this['events'] = events;
        return this;
    };
    ServiceCapability.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ServiceCapability.prototype.withOption = function (option) {
        this['option'] = option;
        return this;
    };
    return ServiceCapability;
}());
exports.ServiceCapability = ServiceCapability;
