// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowProductRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowProductRequest = void 0;
var ShowProductRequest = /** @class */ (function () {
    function ShowProductRequest(productId) {
        this['product_id'] = productId;
    }
    ShowProductRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ShowProductRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductRequest.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ShowProductRequest.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductRequest.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ShowProductRequest.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    return ShowProductRequest;
}());
exports.ShowProductRequest = ShowProductRequest;
