// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowBatchTaskResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowBatchTaskResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowBatchTaskResponse = /** @class */ (function (_super) {
    __extends(ShowBatchTaskResponse, _super);
    function ShowBatchTaskResponse() {
        return _super.call(this) || this;
    }
    ShowBatchTaskResponse.prototype.withBatchtask = function (batchtask) {
        this['batchtask'] = batchtask;
        return this;
    };
    ShowBatchTaskResponse.prototype.withTaskDetails = function (taskDetails) {
        this['task_details'] = taskDetails;
        return this;
    };
    Object.defineProperty(ShowBatchTaskResponse.prototype, "taskDetails", {
        get: function () {
            return this['task_details'];
        },
        set: function (taskDetails) {
            this['task_details'] = taskDetails;
        },
        enumerable: false,
        configurable: true
    });
    ShowBatchTaskResponse.prototype.withPage = function (page) {
        this['page'] = page;
        return this;
    };
    return ShowBatchTaskResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowBatchTaskResponse = ShowBatchTaskResponse;
