// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowOtaPackageResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowOtaPackageResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowOtaPackageResponse = /** @class */ (function (_super) {
    __extends(ShowOtaPackageResponse, _super);
    function ShowOtaPackageResponse() {
        return _super.call(this) || this;
    }
    ShowOtaPackageResponse.prototype.withPackageId = function (packageId) {
        this['package_id'] = packageId;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "packageId", {
        get: function () {
            return this['package_id'];
        },
        set: function (packageId) {
            this['package_id'] = packageId;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withPackageType = function (packageType) {
        this['package_type'] = packageType;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "packageType", {
        get: function () {
            return this['package_type'];
        },
        set: function (packageType) {
            this['package_type'] = packageType;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    ShowOtaPackageResponse.prototype.withSupportSourceVersions = function (supportSourceVersions) {
        this['support_source_versions'] = supportSourceVersions;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "supportSourceVersions", {
        get: function () {
            return this['support_source_versions'];
        },
        set: function (supportSourceVersions) {
            this['support_source_versions'] = supportSourceVersions;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ShowOtaPackageResponse.prototype.withCustomInfo = function (customInfo) {
        this['custom_info'] = customInfo;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "customInfo", {
        get: function () {
            return this['custom_info'];
        },
        set: function (customInfo) {
            this['custom_info'] = customInfo;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowOtaPackageResponse.prototype.withFileLocation = function (fileLocation) {
        this['file_location'] = fileLocation;
        return this;
    };
    Object.defineProperty(ShowOtaPackageResponse.prototype, "fileLocation", {
        get: function () {
            return this['file_location'];
        },
        set: function (fileLocation) {
            this['file_location'] = fileLocation;
        },
        enumerable: false,
        configurable: true
    });
    return ShowOtaPackageResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowOtaPackageResponse = ShowOtaPackageResponse;
