// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowDeviceMessageRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowDeviceMessageRequest = void 0;
var ShowDeviceMessageRequest = /** @class */ (function () {
    function ShowDeviceMessageRequest(deviceId, messageId) {
        this['device_id'] = deviceId;
        this['message_id'] = messageId;
    }
    ShowDeviceMessageRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ShowDeviceMessageRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowDeviceMessageRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ShowDeviceMessageRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowDeviceMessageRequest.prototype.withMessageId = function (messageId) {
        this['message_id'] = messageId;
        return this;
    };
    Object.defineProperty(ShowDeviceMessageRequest.prototype, "messageId", {
        get: function () {
            return this['message_id'];
        },
        set: function (messageId) {
            this['message_id'] = messageId;
        },
        enumerable: false,
        configurable: true
    });
    return ShowDeviceMessageRequest;
}());
exports.ShowDeviceMessageRequest = ShowDeviceMessageRequest;
