// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowRuleActionResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowRuleActionResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowRuleActionResponse = /** @class */ (function (_super) {
    __extends(ShowRuleActionResponse, _super);
    function ShowRuleActionResponse() {
        return _super.call(this) || this;
    }
    ShowRuleActionResponse.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(ShowRuleActionResponse.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    ShowRuleActionResponse.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(ShowRuleActionResponse.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    ShowRuleActionResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ShowRuleActionResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ShowRuleActionResponse.prototype.withChannel = function (channel) {
        this['channel'] = channel;
        return this;
    };
    ShowRuleActionResponse.prototype.withChannelDetail = function (channelDetail) {
        this['channel_detail'] = channelDetail;
        return this;
    };
    Object.defineProperty(ShowRuleActionResponse.prototype, "channelDetail", {
        get: function () {
            return this['channel_detail'];
        },
        set: function (channelDetail) {
            this['channel_detail'] = channelDetail;
        },
        enumerable: false,
        configurable: true
    });
    return ShowRuleActionResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowRuleActionResponse = ShowRuleActionResponse;
