// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowBatchTaskRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowBatchTaskRequest = void 0;
var ShowBatchTaskRequest = /** @class */ (function () {
    function ShowBatchTaskRequest(taskId) {
        this['task_id'] = taskId;
    }
    ShowBatchTaskRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ShowBatchTaskRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowBatchTaskRequest.prototype.withTaskId = function (taskId) {
        this['task_id'] = taskId;
        return this;
    };
    Object.defineProperty(ShowBatchTaskRequest.prototype, "taskId", {
        get: function () {
            return this['task_id'];
        },
        set: function (taskId) {
            this['task_id'] = taskId;
        },
        enumerable: false,
        configurable: true
    });
    ShowBatchTaskRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ShowBatchTaskRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ShowBatchTaskRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ShowBatchTaskRequest;
}());
exports.ShowBatchTaskRequest = ShowBatchTaskRequest;
