// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowApplicationResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowApplicationResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowApplicationResponse = /** @class */ (function (_super) {
    __extends(ShowApplicationResponse, _super);
    function ShowApplicationResponse() {
        return _super.call(this) || this;
    }
    ShowApplicationResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ShowApplicationResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ShowApplicationResponse.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(ShowApplicationResponse.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    ShowApplicationResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(ShowApplicationResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    ShowApplicationResponse.prototype.withDefaultApp = function (defaultApp) {
        this['default_app'] = defaultApp;
        return this;
    };
    Object.defineProperty(ShowApplicationResponse.prototype, "defaultApp", {
        get: function () {
            return this['default_app'];
        },
        set: function (defaultApp) {
            this['default_app'] = defaultApp;
        },
        enumerable: false,
        configurable: true
    });
    return ShowApplicationResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowApplicationResponse = ShowApplicationResponse;
