// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowAsyncDeviceCommandRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowAsyncDeviceCommandRequest = void 0;
var ShowAsyncDeviceCommandRequest = /** @class */ (function () {
    function ShowAsyncDeviceCommandRequest(deviceId, commandId) {
        this['device_id'] = deviceId;
        this['command_id'] = commandId;
    }
    ShowAsyncDeviceCommandRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(ShowAsyncDeviceCommandRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowAsyncDeviceCommandRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ShowAsyncDeviceCommandRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowAsyncDeviceCommandRequest.prototype.withCommandId = function (commandId) {
        this['command_id'] = commandId;
        return this;
    };
    Object.defineProperty(ShowAsyncDeviceCommandRequest.prototype, "commandId", {
        get: function () {
            return this['command_id'];
        },
        set: function (commandId) {
            this['command_id'] = commandId;
        },
        enumerable: false,
        configurable: true
    });
    return ShowAsyncDeviceCommandRequest;
}());
exports.ShowAsyncDeviceCommandRequest = ShowAsyncDeviceCommandRequest;
