// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowProductResponse.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowProductResponse = void 0;
var SdkResponse_1 = require("@huaweicloud/huaweicloud-sdk-core/SdkResponse");
var ShowProductResponse = /** @class */ (function (_super) {
    __extends(ShowProductResponse, _super);
    function ShowProductResponse() {
        return _super.call(this) || this;
    }
    ShowProductResponse.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withAppName = function (appName) {
        this['app_name'] = appName;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "appName", {
        get: function () {
            return this['app_name'];
        },
        set: function (appName) {
            this['app_name'] = appName;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    ShowProductResponse.prototype.withDeviceType = function (deviceType) {
        this['device_type'] = deviceType;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "deviceType", {
        get: function () {
            return this['device_type'];
        },
        set: function (deviceType) {
            this['device_type'] = deviceType;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withProtocolType = function (protocolType) {
        this['protocol_type'] = protocolType;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "protocolType", {
        get: function () {
            return this['protocol_type'];
        },
        set: function (protocolType) {
            this['protocol_type'] = protocolType;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withDataFormat = function (dataFormat) {
        this['data_format'] = dataFormat;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "dataFormat", {
        get: function () {
            return this['data_format'];
        },
        set: function (dataFormat) {
            this['data_format'] = dataFormat;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withManufacturerName = function (manufacturerName) {
        this['manufacturer_name'] = manufacturerName;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "manufacturerName", {
        get: function () {
            return this['manufacturer_name'];
        },
        set: function (manufacturerName) {
            this['manufacturer_name'] = manufacturerName;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withIndustry = function (industry) {
        this['industry'] = industry;
        return this;
    };
    ShowProductResponse.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    ShowProductResponse.prototype.withServiceCapabilities = function (serviceCapabilities) {
        this['service_capabilities'] = serviceCapabilities;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "serviceCapabilities", {
        get: function () {
            return this['service_capabilities'];
        },
        set: function (serviceCapabilities) {
            this['service_capabilities'] = serviceCapabilities;
        },
        enumerable: false,
        configurable: true
    });
    ShowProductResponse.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(ShowProductResponse.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    return ShowProductResponse;
}(SdkResponse_1.SdkResponse));
exports.ShowProductResponse = ShowProductResponse;
