// @huaweicloud/huaweicloud-sdk-iotda/v5/model/ShowDevicesInGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ShowDevicesInGroupRequest = void 0;
var ShowDevicesInGroupRequest = /** @class */ (function () {
    function ShowDevicesInGroupRequest(groupId) {
        this['group_id'] = groupId;
    }
    ShowDevicesInGroupRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(ShowDevicesInGroupRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    ShowDevicesInGroupRequest.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(ShowDevicesInGroupRequest.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    ShowDevicesInGroupRequest.prototype.withLimit = function (limit) {
        this['limit'] = limit;
        return this;
    };
    ShowDevicesInGroupRequest.prototype.withMarker = function (marker) {
        this['marker'] = marker;
        return this;
    };
    ShowDevicesInGroupRequest.prototype.withOffset = function (offset) {
        this['offset'] = offset;
        return this;
    };
    return ShowDevicesInGroupRequest;
}());
exports.ShowDevicesInGroupRequest = ShowDevicesInGroupRequest;
