// @huaweicloud/huaweicloud-sdk-iotda/v5/model/SimpleTimerType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimpleTimerType = void 0;
var SimpleTimerType = /** @class */ (function () {
    function SimpleTimerType(startTime, repeatInterval, repeatCount) {
        this['start_time'] = startTime;
        this['repeat_interval'] = repeatInterval;
        this['repeat_count'] = repeatCount;
    }
    SimpleTimerType.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(SimpleTimerType.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    SimpleTimerType.prototype.withRepeatInterval = function (repeatInterval) {
        this['repeat_interval'] = repeatInterval;
        return this;
    };
    Object.defineProperty(SimpleTimerType.prototype, "repeatInterval", {
        get: function () {
            return this['repeat_interval'];
        },
        set: function (repeatInterval) {
            this['repeat_interval'] = repeatInterval;
        },
        enumerable: false,
        configurable: true
    });
    SimpleTimerType.prototype.withRepeatCount = function (repeatCount) {
        this['repeat_count'] = repeatCount;
        return this;
    };
    Object.defineProperty(SimpleTimerType.prototype, "repeatCount", {
        get: function () {
            return this['repeat_count'];
        },
        set: function (repeatCount) {
            this['repeat_count'] = repeatCount;
        },
        enumerable: false,
        configurable: true
    });
    return SimpleTimerType;
}());
exports.SimpleTimerType = SimpleTimerType;
