// @huaweicloud/huaweicloud-sdk-iotda/v5/model/TaskProgress.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskProgress = void 0;
var TaskProgress = /** @class */ (function () {
    function TaskProgress() {
    }
    TaskProgress.prototype.withTotal = function (total) {
        this['total'] = total;
        return this;
    };
    TaskProgress.prototype.withProcessing = function (processing) {
        this['processing'] = processing;
        return this;
    };
    TaskProgress.prototype.withSuccess = function (success) {
        this['success'] = success;
        return this;
    };
    TaskProgress.prototype.withFail = function (fail) {
        this['fail'] = fail;
        return this;
    };
    TaskProgress.prototype.withWaitting = function (waitting) {
        this['waitting'] = waitting;
        return this;
    };
    TaskProgress.prototype.withFailWaitRetry = function (failWaitRetry) {
        this['fail_wait_retry'] = failWaitRetry;
        return this;
    };
    Object.defineProperty(TaskProgress.prototype, "failWaitRetry", {
        get: function () {
            return this['fail_wait_retry'];
        },
        set: function (failWaitRetry) {
            this['fail_wait_retry'] = failWaitRetry;
        },
        enumerable: false,
        configurable: true
    });
    TaskProgress.prototype.withStopped = function (stopped) {
        this['stopped'] = stopped;
        return this;
    };
    return TaskProgress;
}());
exports.TaskProgress = TaskProgress;
