// @huaweicloud/huaweicloud-sdk-iotda/v5/model/TaskPolicy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TaskPolicy = void 0;
var TaskPolicy = /** @class */ (function () {
    function TaskPolicy() {
    }
    TaskPolicy.prototype.withScheduleTime = function (scheduleTime) {
        this['schedule_time'] = scheduleTime;
        return this;
    };
    Object.defineProperty(TaskPolicy.prototype, "scheduleTime", {
        get: function () {
            return this['schedule_time'];
        },
        set: function (scheduleTime) {
            this['schedule_time'] = scheduleTime;
        },
        enumerable: false,
        configurable: true
    });
    TaskPolicy.prototype.withRetryCount = function (retryCount) {
        this['retry_count'] = retryCount;
        return this;
    };
    Object.defineProperty(TaskPolicy.prototype, "retryCount", {
        get: function () {
            return this['retry_count'];
        },
        set: function (retryCount) {
            this['retry_count'] = retryCount;
        },
        enumerable: false,
        configurable: true
    });
    TaskPolicy.prototype.withRetryInterval = function (retryInterval) {
        this['retry_interval'] = retryInterval;
        return this;
    };
    Object.defineProperty(TaskPolicy.prototype, "retryInterval", {
        get: function () {
            return this['retry_interval'];
        },
        set: function (retryInterval) {
            this['retry_interval'] = retryInterval;
        },
        enumerable: false,
        configurable: true
    });
    return TaskPolicy;
}());
exports.TaskPolicy = TaskPolicy;
