// @huaweicloud/huaweicloud-sdk-iotda/v5/model/Task.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Task = void 0;
var Task = /** @class */ (function () {
    function Task() {
    }
    Task.prototype.withTaskId = function (taskId) {
        this['task_id'] = taskId;
        return this;
    };
    Object.defineProperty(Task.prototype, "taskId", {
        get: function () {
            return this['task_id'];
        },
        set: function (taskId) {
            this['task_id'] = taskId;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withTaskName = function (taskName) {
        this['task_name'] = taskName;
        return this;
    };
    Object.defineProperty(Task.prototype, "taskName", {
        get: function () {
            return this['task_name'];
        },
        set: function (taskName) {
            this['task_name'] = taskName;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withTaskType = function (taskType) {
        this['task_type'] = taskType;
        return this;
    };
    Object.defineProperty(Task.prototype, "taskType", {
        get: function () {
            return this['task_type'];
        },
        set: function (taskType) {
            this['task_type'] = taskType;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withTargets = function (targets) {
        this['targets'] = targets;
        return this;
    };
    Task.prototype.withTargetsFilter = function (targetsFilter) {
        this['targets_filter'] = targetsFilter;
        return this;
    };
    Object.defineProperty(Task.prototype, "targetsFilter", {
        get: function () {
            return this['targets_filter'];
        },
        set: function (targetsFilter) {
            this['targets_filter'] = targetsFilter;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withDocument = function (document) {
        this['document'] = document;
        return this;
    };
    Task.prototype.withTaskPolicy = function (taskPolicy) {
        this['task_policy'] = taskPolicy;
        return this;
    };
    Object.defineProperty(Task.prototype, "taskPolicy", {
        get: function () {
            return this['task_policy'];
        },
        set: function (taskPolicy) {
            this['task_policy'] = taskPolicy;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withStatus = function (status) {
        this['status'] = status;
        return this;
    };
    Task.prototype.withStatusDesc = function (statusDesc) {
        this['status_desc'] = statusDesc;
        return this;
    };
    Object.defineProperty(Task.prototype, "statusDesc", {
        get: function () {
            return this['status_desc'];
        },
        set: function (statusDesc) {
            this['status_desc'] = statusDesc;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withTaskProgress = function (taskProgress) {
        this['task_progress'] = taskProgress;
        return this;
    };
    Object.defineProperty(Task.prototype, "taskProgress", {
        get: function () {
            return this['task_progress'];
        },
        set: function (taskProgress) {
            this['task_progress'] = taskProgress;
        },
        enumerable: false,
        configurable: true
    });
    Task.prototype.withCreateTime = function (createTime) {
        this['create_time'] = createTime;
        return this;
    };
    Object.defineProperty(Task.prototype, "createTime", {
        get: function () {
            return this['create_time'];
        },
        set: function (createTime) {
            this['create_time'] = createTime;
        },
        enumerable: false,
        configurable: true
    });
    return Task;
}());
exports.Task = Task;
