// @huaweicloud/huaweicloud-sdk-iotda/v5/model/TimeRange.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TimeRange = void 0;
var TimeRange = /** @class */ (function () {
    function TimeRange(startTime, endTime) {
        this['start_time'] = startTime;
        this['end_time'] = endTime;
    }
    TimeRange.prototype.withStartTime = function (startTime) {
        this['start_time'] = startTime;
        return this;
    };
    Object.defineProperty(TimeRange.prototype, "startTime", {
        get: function () {
            return this['start_time'];
        },
        set: function (startTime) {
            this['start_time'] = startTime;
        },
        enumerable: false,
        configurable: true
    });
    TimeRange.prototype.withEndTime = function (endTime) {
        this['end_time'] = endTime;
        return this;
    };
    Object.defineProperty(TimeRange.prototype, "endTime", {
        get: function () {
            return this['end_time'];
        },
        set: function (endTime) {
            this['end_time'] = endTime;
        },
        enumerable: false,
        configurable: true
    });
    TimeRange.prototype.withDaysOfWeek = function (daysOfWeek) {
        this['days_of_week'] = daysOfWeek;
        return this;
    };
    Object.defineProperty(TimeRange.prototype, "daysOfWeek", {
        get: function () {
            return this['days_of_week'];
        },
        set: function (daysOfWeek) {
            this['days_of_week'] = daysOfWeek;
        },
        enumerable: false,
        configurable: true
    });
    return TimeRange;
}());
exports.TimeRange = TimeRange;
