// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UnbindTagsDTO.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnbindTagsDTO = void 0;
var UnbindTagsDTO = /** @class */ (function () {
    function UnbindTagsDTO(resourceType, resourceId, tagKeys) {
        this['resource_type'] = resourceType;
        this['resource_id'] = resourceId;
        this['tag_keys'] = tagKeys;
    }
    UnbindTagsDTO.prototype.withResourceType = function (resourceType) {
        this['resource_type'] = resourceType;
        return this;
    };
    Object.defineProperty(UnbindTagsDTO.prototype, "resourceType", {
        get: function () {
            return this['resource_type'];
        },
        set: function (resourceType) {
            this['resource_type'] = resourceType;
        },
        enumerable: false,
        configurable: true
    });
    UnbindTagsDTO.prototype.withResourceId = function (resourceId) {
        this['resource_id'] = resourceId;
        return this;
    };
    Object.defineProperty(UnbindTagsDTO.prototype, "resourceId", {
        get: function () {
            return this['resource_id'];
        },
        set: function (resourceId) {
            this['resource_id'] = resourceId;
        },
        enumerable: false,
        configurable: true
    });
    UnbindTagsDTO.prototype.withTagKeys = function (tagKeys) {
        this['tag_keys'] = tagKeys;
        return this;
    };
    Object.defineProperty(UnbindTagsDTO.prototype, "tagKeys", {
        get: function () {
            return this['tag_keys'];
        },
        set: function (tagKeys) {
            this['tag_keys'] = tagKeys;
        },
        enumerable: false,
        configurable: true
    });
    return UnbindTagsDTO;
}());
exports.UnbindTagsDTO = UnbindTagsDTO;
