// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateProductRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateProductRequest = void 0;
var UpdateProductRequest = /** @class */ (function () {
    function UpdateProductRequest(productId) {
        this['product_id'] = productId;
    }
    UpdateProductRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateProductRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProductRequest.prototype.withProductId = function (productId) {
        this['product_id'] = productId;
        return this;
    };
    Object.defineProperty(UpdateProductRequest.prototype, "productId", {
        get: function () {
            return this['product_id'];
        },
        set: function (productId) {
            this['product_id'] = productId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProductRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateProductRequest;
}());
exports.UpdateProductRequest = UpdateProductRequest;
