// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateRuleRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRuleRequest = void 0;
var UpdateRuleRequest = /** @class */ (function () {
    function UpdateRuleRequest(ruleId) {
        this['rule_id'] = ruleId;
    }
    UpdateRuleRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateRuleRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRuleRequest.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(UpdateRuleRequest.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRuleRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateRuleRequest;
}());
exports.UpdateRuleRequest = UpdateRuleRequest;
