// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateDevice.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDevice = void 0;
var UpdateDevice = /** @class */ (function () {
    function UpdateDevice() {
    }
    UpdateDevice.prototype.withDeviceName = function (deviceName) {
        this['device_name'] = deviceName;
        return this;
    };
    Object.defineProperty(UpdateDevice.prototype, "deviceName", {
        get: function () {
            return this['device_name'];
        },
        set: function (deviceName) {
            this['device_name'] = deviceName;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDevice.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateDevice.prototype.withExtensionInfo = function (extensionInfo) {
        this['extension_info'] = extensionInfo;
        return this;
    };
    Object.defineProperty(UpdateDevice.prototype, "extensionInfo", {
        get: function () {
            return this['extension_info'];
        },
        set: function (extensionInfo) {
            this['extension_info'] = extensionInfo;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDevice.prototype.withAuthInfo = function (authInfo) {
        this['auth_info'] = authInfo;
        return this;
    };
    Object.defineProperty(UpdateDevice.prototype, "authInfo", {
        get: function () {
            return this['auth_info'];
        },
        set: function (authInfo) {
            this['auth_info'] = authInfo;
        },
        enumerable: false,
        configurable: true
    });
    return UpdateDevice;
}());
exports.UpdateDevice = UpdateDevice;
