// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateProduct.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateProduct = void 0;
var UpdateProduct = /** @class */ (function () {
    function UpdateProduct() {
    }
    UpdateProduct.prototype.withAppId = function (appId) {
        this['app_id'] = appId;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "appId", {
        get: function () {
            return this['app_id'];
        },
        set: function (appId) {
            this['app_id'] = appId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withName = function (name) {
        this['name'] = name;
        return this;
    };
    UpdateProduct.prototype.withDeviceType = function (deviceType) {
        this['device_type'] = deviceType;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "deviceType", {
        get: function () {
            return this['device_type'];
        },
        set: function (deviceType) {
            this['device_type'] = deviceType;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withProtocolType = function (protocolType) {
        this['protocol_type'] = protocolType;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "protocolType", {
        get: function () {
            return this['protocol_type'];
        },
        set: function (protocolType) {
            this['protocol_type'] = protocolType;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withDataFormat = function (dataFormat) {
        this['data_format'] = dataFormat;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "dataFormat", {
        get: function () {
            return this['data_format'];
        },
        set: function (dataFormat) {
            this['data_format'] = dataFormat;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withServiceCapabilities = function (serviceCapabilities) {
        this['service_capabilities'] = serviceCapabilities;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "serviceCapabilities", {
        get: function () {
            return this['service_capabilities'];
        },
        set: function (serviceCapabilities) {
            this['service_capabilities'] = serviceCapabilities;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withManufacturerName = function (manufacturerName) {
        this['manufacturer_name'] = manufacturerName;
        return this;
    };
    Object.defineProperty(UpdateProduct.prototype, "manufacturerName", {
        get: function () {
            return this['manufacturer_name'];
        },
        set: function (manufacturerName) {
            this['manufacturer_name'] = manufacturerName;
        },
        enumerable: false,
        configurable: true
    });
    UpdateProduct.prototype.withIndustry = function (industry) {
        this['industry'] = industry;
        return this;
    };
    UpdateProduct.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    return UpdateProduct;
}());
exports.UpdateProduct = UpdateProduct;
