// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateDeviceGroupRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDeviceGroupRequest = void 0;
var UpdateDeviceGroupRequest = /** @class */ (function () {
    function UpdateDeviceGroupRequest(groupId) {
        this['group_id'] = groupId;
    }
    UpdateDeviceGroupRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateDeviceGroupRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDeviceGroupRequest.prototype.withGroupId = function (groupId) {
        this['group_id'] = groupId;
        return this;
    };
    Object.defineProperty(UpdateDeviceGroupRequest.prototype, "groupId", {
        get: function () {
            return this['group_id'];
        },
        set: function (groupId) {
            this['group_id'] = groupId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDeviceGroupRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateDeviceGroupRequest;
}());
exports.UpdateDeviceGroupRequest = UpdateDeviceGroupRequest;
