// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateDeviceShadowDesiredDataRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDeviceShadowDesiredDataRequest = void 0;
var UpdateDeviceShadowDesiredDataRequest = /** @class */ (function () {
    function UpdateDeviceShadowDesiredDataRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    UpdateDeviceShadowDesiredDataRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateDeviceShadowDesiredDataRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDeviceShadowDesiredDataRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(UpdateDeviceShadowDesiredDataRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDeviceShadowDesiredDataRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateDeviceShadowDesiredDataRequest;
}());
exports.UpdateDeviceShadowDesiredDataRequest = UpdateDeviceShadowDesiredDataRequest;
