// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateDesired.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateDesired = void 0;
var UpdateDesired = /** @class */ (function () {
    function UpdateDesired(serviceId, desired) {
        this['service_id'] = serviceId;
        this['desired'] = desired;
    }
    UpdateDesired.prototype.withServiceId = function (serviceId) {
        this['service_id'] = serviceId;
        return this;
    };
    Object.defineProperty(UpdateDesired.prototype, "serviceId", {
        get: function () {
            return this['service_id'];
        },
        set: function (serviceId) {
            this['service_id'] = serviceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateDesired.prototype.withDesired = function (desired) {
        this['desired'] = desired;
        return this;
    };
    UpdateDesired.prototype.withVersion = function (version) {
        this['version'] = version;
        return this;
    };
    return UpdateDesired;
}());
exports.UpdateDesired = UpdateDesired;
