// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateRuleReq.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRuleReq = void 0;
var UpdateRuleReq = /** @class */ (function () {
    function UpdateRuleReq() {
    }
    UpdateRuleReq.prototype.withRuleName = function (ruleName) {
        this['rule_name'] = ruleName;
        return this;
    };
    Object.defineProperty(UpdateRuleReq.prototype, "ruleName", {
        get: function () {
            return this['rule_name'];
        },
        set: function (ruleName) {
            this['rule_name'] = ruleName;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRuleReq.prototype.withDescription = function (description) {
        this['description'] = description;
        return this;
    };
    UpdateRuleReq.prototype.withSelect = function (select) {
        this['select'] = select;
        return this;
    };
    UpdateRuleReq.prototype.withWhere = function (where) {
        this['where'] = where;
        return this;
    };
    UpdateRuleReq.prototype.withActive = function (active) {
        this['active'] = active;
        return this;
    };
    return UpdateRuleReq;
}());
exports.UpdateRuleReq = UpdateRuleReq;
