// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateRoutingRuleRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRoutingRuleRequest = void 0;
var UpdateRoutingRuleRequest = /** @class */ (function () {
    function UpdateRoutingRuleRequest(ruleId) {
        this['rule_id'] = ruleId;
    }
    UpdateRoutingRuleRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateRoutingRuleRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRoutingRuleRequest.prototype.withRuleId = function (ruleId) {
        this['rule_id'] = ruleId;
        return this;
    };
    Object.defineProperty(UpdateRoutingRuleRequest.prototype, "ruleId", {
        get: function () {
            return this['rule_id'];
        },
        set: function (ruleId) {
            this['rule_id'] = ruleId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRoutingRuleRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateRoutingRuleRequest;
}());
exports.UpdateRoutingRuleRequest = UpdateRoutingRuleRequest;
