// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdateRuleActionRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateRuleActionRequest = void 0;
var UpdateRuleActionRequest = /** @class */ (function () {
    function UpdateRuleActionRequest(actionId) {
        this['action_id'] = actionId;
    }
    UpdateRuleActionRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdateRuleActionRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRuleActionRequest.prototype.withActionId = function (actionId) {
        this['action_id'] = actionId;
        return this;
    };
    Object.defineProperty(UpdateRuleActionRequest.prototype, "actionId", {
        get: function () {
            return this['action_id'];
        },
        set: function (actionId) {
            this['action_id'] = actionId;
        },
        enumerable: false,
        configurable: true
    });
    UpdateRuleActionRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdateRuleActionRequest;
}());
exports.UpdateRuleActionRequest = UpdateRuleActionRequest;
