// @huaweicloud/huaweicloud-sdk-iotda/v5/model/UpdatePropertiesRequest.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePropertiesRequest = void 0;
var UpdatePropertiesRequest = /** @class */ (function () {
    function UpdatePropertiesRequest(deviceId) {
        this['device_id'] = deviceId;
    }
    UpdatePropertiesRequest.prototype.withDeviceId = function (deviceId) {
        this['device_id'] = deviceId;
        return this;
    };
    Object.defineProperty(UpdatePropertiesRequest.prototype, "deviceId", {
        get: function () {
            return this['device_id'];
        },
        set: function (deviceId) {
            this['device_id'] = deviceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdatePropertiesRequest.prototype.withInstanceId = function (instanceId) {
        this['Instance-Id'] = instanceId;
        return this;
    };
    Object.defineProperty(UpdatePropertiesRequest.prototype, "instanceId", {
        get: function () {
            return this['Instance-Id'];
        },
        set: function (instanceId) {
            this['Instance-Id'] = instanceId;
        },
        enumerable: false,
        configurable: true
    });
    UpdatePropertiesRequest.prototype.withBody = function (body) {
        this['body'] = body;
        return this;
    };
    return UpdatePropertiesRequest;
}());
exports.UpdatePropertiesRequest = UpdatePropertiesRequest;
