// @huaweicloud/huaweicloud-sdk-core/logger/index.js
"use strict";
/*
 * Copyright 2021 Huawei Technologies Co.,Ltd.
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var _a;
Object.defineProperty(exports, "__esModule", { value: true });
exports.getLogger = exports.LocalLogger = exports.LoggerUtils = exports.LogLevel = void 0;
var log4jLogger_1 = require("./log4jLogger");
var LogLevel;
(function (LogLevel) {
    LogLevel["ERROR"] = "error";
    LogLevel["DEBUG"] = "debug";
    LogLevel["INFO"] = "info";
    LogLevel["WARN"] = "warn";
})(LogLevel = exports.LogLevel || (exports.LogLevel = {}));
var LoggerUtils = exports.LoggerUtils = /** @class */ (function () {
    function LoggerUtils() {
    }
    LoggerUtils.isMoreOrEqualSevere = function (a, b) {
        return LoggerUtils.severity[a] >= LoggerUtils.severity[b];
    };
    /** Map of severity as comparable numbers for each log level */
    LoggerUtils.severity = (_a = {},
        _a[LogLevel.ERROR] = 400,
        _a[LogLevel.WARN] = 300,
        _a[LogLevel.INFO] = 200,
        _a[LogLevel.DEBUG] = 100,
        _a);
    return LoggerUtils;
}());
var LocalLogger = /** @class */ (function () {
    function LocalLogger() {
        this.level = LogLevel.INFO;
    }
    LocalLogger.prototype.debug = function () {
        var msg = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            msg[_i] = arguments[_i];
        }
        if (LoggerUtils.isMoreOrEqualSevere(LogLevel.DEBUG, this.level)) {
            log4jLogger_1.Logger4jInstance.debug.apply(log4jLogger_1.Logger4jInstance, __spreadArray([this.name], msg, false));
        }
    };
    LocalLogger.prototype.info = function () {
        var msg = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            msg[_i] = arguments[_i];
        }
        if (LoggerUtils.isMoreOrEqualSevere(LogLevel.INFO, this.level)) {
            log4jLogger_1.Logger4jInstance.info.apply(log4jLogger_1.Logger4jInstance, __spreadArray([this.name], msg, false));
        }
    };
    LocalLogger.prototype.warn = function () {
        var msg = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            msg[_i] = arguments[_i];
        }
        if (LoggerUtils.isMoreOrEqualSevere(LogLevel.WARN, this.level)) {
            log4jLogger_1.Logger4jInstance.warn.apply(log4jLogger_1.Logger4jInstance, __spreadArray([this.name], msg, false));
        }
    };
    LocalLogger.prototype.error = function () {
        var msg = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            msg[_i] = arguments[_i];
        }
        if (LoggerUtils.isMoreOrEqualSevere(LogLevel.ERROR, this.level)) {
            log4jLogger_1.Logger4jInstance.error.apply(log4jLogger_1.Logger4jInstance, __spreadArray([this.name], msg, false));
        }
    };
    LocalLogger.prototype.setLevel = function (level) {
        this.level = level;
    };
    LocalLogger.prototype.getLevel = function () {
        return this.level;
    };
    LocalLogger.prototype.setName = function (name) {
        this.name = name;
    };
    return LocalLogger;
}());
exports.LocalLogger = LocalLogger;
var instanceCount = 0;
function getLogger(name, level, existingLogger) {
    var instanceId = instanceCount;
    instanceCount += 1;
    var logger = (function () {
        if (existingLogger !== undefined) {
            return existingLogger;
        }
        return new LocalLogger();
    })();
    if (level !== undefined) {
        logger.setLevel(level);
    }
    logger.setName("nodesdk:".concat(name, ":").concat(instanceId));
    if (process.env.DEBUG === "DEBUG") {
        logger.setLevel(LogLevel.DEBUG);
    }
    return logger;
}
exports.getLogger = getLogger;
